{
 "metadata": {
  "name": "",
  "signature": "sha256:282c47686dc4e1dd2f5664608f00684ff325811f804d722a24165939abc4f37a"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Ask Andy... but No common denom between mut and pat data"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%pylab inline\n",
      "import pandas as pd\n",
      "PATH = '/cellar/users/talejand/Documents/IdekerLab/CancerPapers/Genomic sequencing of meningiomas identifies oncogenic SMO and AKT1 mutations/'"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Populating the interactive namespace from numpy and matplotlib\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Genomic sequencing of meningiomas identifies oncogenic SMO and AKT1 mutations"
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Tissue: Meningiomas"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Most Frequently Mutated Gene</b> : NF2 (tumor suppressor) - disrupted in <b>50%</b> of cases\n",
      "\n",
      "* 9 nonsense\n",
      "* 9 splice-site \n",
      "* 9 frameshift indels"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Sequencing:</b>\n",
      "\n",
      "    1. Discovery Set Sequencing of Grade 1 Meningiomas: \n",
      "\n",
      "        * WGS (57X): n = 11\n",
      "\n",
      "        * WES (181X): n = 6\n",
      "\n",
      "\n",
      "    2. Validation Set Sequencing (154X): 645 known cancer-associated genes\n",
      "\n",
      "        * 30 additional grade 1 tumors\n",
      "        * 18 genes in an extrapolation set of Grade 2/3 \n",
      "            * allow for evaluation of findings for grade 1 tumors to progress to higher grades\n",
      "\n",
      "    3. Additional 48 tumors sequenced to validate somatic genetic alterations"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b> Focal NF2 Inactivation:</b> 43% of tumors\n",
      "\n",
      "    * Additional 8% had epigenetic modifier alterations\n",
      "    * A subset of meningiomas lacking  NF2 alterations harbored recurrent oncogenic mutations in  AKT1 (p.Glu17Lys) and SMO (p.Trp535Leu) and exhibited  immunohistochemical evidence of activation of these pathways. "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>SCNAS:</b> loss of chr 22 most frequent (contains NF2)\n",
      "\n",
      "* occur in all ten tumors w/ focal NF2 alteration and in 1 w/out\n",
      "* Median meningioma had SCNAs affecting only 3.3% of the genome, one rearrangement and eight nonsynonymous mutations."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Among mutations, cytosine-to-thymidine transitions within a CpG context predominated, consistent with spontaneous deamination "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Rates of alteration in meningiomas were significantly lower (~10x) than prev. determined rates from other tumors\n",
      "\n",
      "* In contrast, rates of genomic disruption were higher in the grade II and III meningiomas in the extrapolation set than for their grade I counterparts"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Mean rate of nonsynonymous mutation within the 645 sequenced genes was nearly twice as high in higher-grade tumors (3.0 versus 1.6; P < 0.02), and the proportion of the genome affected by SCNAs was markedly higher (median of 12.8% versus 0.3%; P < 0.001)."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Candidate Somatic Rearrangements:</b> 110\n",
      "\n",
      "* 93 confirmed reads spanning putative fusion site\n",
      "\n",
      "* most arrangments concentrated in 3 tumors, 1 w/ chromothripsis"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Also, Nonsynonymous mutations in 190 other genes\n",
      "\n",
      "\n",
      "* The nonsynonymous muta- tion rates of tumors with and without focal NF2 alterations did not differ significantly, nor did the spectra of their mutation subtypes"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Genes of Statistical Significance:</b> 6\n",
      "(1-4 prev. implicated)\n",
      "\n",
      "    1. NF2\n",
      "    2. KDM5C\n",
      "    3. SMO\n",
      "    4. AKT1\n",
      "    5. RGPD3 \n",
      "    6. CD300C\n",
      "\n",
      "* Isolated/Rare Mut. in 23 genes (TP53, APC, CBL, STK11, NOTCH2)"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Among genes mutated in our cohort and previously associated with cancer, three (KDM5C, KDM6A and SMARCB1) are epigenetic modi- fiers. Mutations involving these genes affected 8% of the cohort."
     ]
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "Mutation identified (encoding a p.Arg374Gln alteration) is near a mutational hotspot (p.Arg377His) described in meningiomas"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>SMO Mutation</b> in 3 tumors (5%)\n",
      "\n",
      "* None of these tumors had NF2 focal alteration"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Six samples had mutations of the <b>phosphatidylinositol 3-kinase (PI3K)-AKT-mTOR pathway</b>. \n",
      "\n",
      "* None of these had mutations of NF2 or SMO (P = 0.03)\n",
      "\n",
      "* Mutations in the Hedgehog and PI3K pathways were also associated with particular histopathological subtypes"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Meningiomas with mutated SMO and AKT1-MTOR were predomi- nantly of the meningothelial subtype \n",
      "\n",
      "* contrast to the grade I NF2-altered tumors, which were predominantly fibroblastic and/or transitional"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Validated these findings by genotyping an additional 46 grade I and 49 grade II or III meningiomas for AKT1 (p.Glu17Lys) and SMO (p.Trp535Leu and p.Leu412Phe) mutations <i>(using a mass spectrometry\u2013 based method (Sequenom hME))</i>.\n",
      "\n",
      "* One SMO mutation (p.Leu412Phe)\n",
      "* Three additional AKT1 mutations (p.Glu17Lys)"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Observed recurrent mutations in signaling pathways and epi- genetic modifiers, but data also highlights the heterogeneity of mutations in these tumors. \n",
      "\n",
      "* Among the 17 tumors comprehen- sively characterized, 3 (18%) did not have mutations in any of the significantly mutated genes. "
     ]
    },
    {
     "cell_type": "heading",
     "level": 4,
     "metadata": {},
     "source": [
      "The paucity of addi- tional genetic events in meningiomas harboring mutations of the PI3K and Hedgehog signaling pathways suggests that patients with these meningiomas may benefit from such targeted therapies already in development or use."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Accession Codes:</b> from database of Genotypes and Phenotypes (dbGaP) under accession phs000552.v1.p1."
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Hard Data"
     ]
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Patient Data:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "#Patient Data\n",
      "\n",
      "# ASK ANDY... Can I do something where I create a gene column for the dataframe and have the top columns that are genes be used to create the gene colum\n",
      "#i mean can the column NF2 be used to bring down the NF2 gene into a gene column or AKT1 depending on if the value is null\n",
      "#then you can combine the NF2, akt1, smo columns into a protein_change column\n",
      "\n",
      "#comments: i like the columns column with discov vs extrap, but prob don't have that consistently in others\n",
      "# also location vs location category\n",
      "\n",
      "#read in csv file, use specific columns\n",
      "patdata = pd.read_csv(PATH + 'ng.2526-S5.csv', usecols=['NF2','AKT1','SMO','Sample ID','Grade','Histology','Age', 'Sex', 'Location','Recurrent'])\n",
      "#rename columns to format\n",
      "patdata.rename(columns={'Sample ID':'Sample_ID','Sex':'Gender'})\n",
      "#write to csv in PatData folder\n",
      "patdata.to_csv()\n",
      "\n",
      "patdata\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Sample ID</th>\n",
        "      <th>Grade</th>\n",
        "      <th>Histology</th>\n",
        "      <th>Age</th>\n",
        "      <th>Sex</th>\n",
        "      <th>Location</th>\n",
        "      <th>Recurrent</th>\n",
        "      <th>NF2</th>\n",
        "      <th>AKT1</th>\n",
        "      <th>SMO</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0 </th>\n",
        "      <td>  0008-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td> Fibroblastic-Transitional</td>\n",
        "      <td> 65</td>\n",
        "      <td> F</td>\n",
        "      <td>                Left temporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>   p.R200_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1 </th>\n",
        "      <td>  0009-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 29</td>\n",
        "      <td> F</td>\n",
        "      <td>         Right frontotemporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td> NF2-TCF20fusion</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2 </th>\n",
        "      <td>  0011-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 55</td>\n",
        "      <td> F</td>\n",
        "      <td>              Right occipital</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td> p.E17K</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3 </th>\n",
        "      <td>  0015-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 41</td>\n",
        "      <td> F</td>\n",
        "      <td>                Right frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td> p.W535L</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4 </th>\n",
        "      <td>  0016-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 47</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>   p.Y150_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5 </th>\n",
        "      <td>  0017-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 71</td>\n",
        "      <td> F</td>\n",
        "      <td>           Right parasagittal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>        p.K227fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6 </th>\n",
        "      <td>  0018-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 69</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>         p.Y144*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7 </th>\n",
        "      <td>  0019-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 51</td>\n",
        "      <td> F</td>\n",
        "      <td>         Left Posterior fossa</td>\n",
        "      <td> NaN</td>\n",
        "      <td>        p.R160fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8 </th>\n",
        "      <td>  0020-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 48</td>\n",
        "      <td> F</td>\n",
        "      <td>               Right parietal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>         p.R341*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9 </th>\n",
        "      <td>  0021-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 31</td>\n",
        "      <td> F</td>\n",
        "      <td>                Right frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10</th>\n",
        "      <td>  0022-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td> Fibroblastic-Transitional</td>\n",
        "      <td> 61</td>\n",
        "      <td> F</td>\n",
        "      <td>           Right parasagittal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>    p.K80_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11</th>\n",
        "      <td>  MENex001</td>\n",
        "      <td>   I</td>\n",
        "      <td> Fibroblastic-Transitional</td>\n",
        "      <td> 61</td>\n",
        "      <td> F</td>\n",
        "      <td>           Right parasagittal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12</th>\n",
        "      <td>  MENex002</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 54</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>          p.W41*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>13</th>\n",
        "      <td> MENex003m</td>\n",
        "      <td>   I</td>\n",
        "      <td> Fibroblastic-Transitional</td>\n",
        "      <td> 35</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left P-fossa</td>\n",
        "      <td> NaN</td>\n",
        "      <td>   p.Y150_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>14</th>\n",
        "      <td>  MENex004</td>\n",
        "      <td>   I</td>\n",
        "      <td>   Angiomatous-Microcystic</td>\n",
        "      <td> 61</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15</th>\n",
        "      <td>  MENex005</td>\n",
        "      <td>   I</td>\n",
        "      <td>   Angiomatous-Microcystic</td>\n",
        "      <td> 46</td>\n",
        "      <td> M</td>\n",
        "      <td>                Right frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16</th>\n",
        "      <td>  MENex006</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 28</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left clinoid</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td> p.W535L</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17</th>\n",
        "      <td>  0043-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 45</td>\n",
        "      <td> M</td>\n",
        "      <td>                Right frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18</th>\n",
        "      <td>  0045-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 67</td>\n",
        "      <td> F</td>\n",
        "      <td>  Right Frontal and tentorial</td>\n",
        "      <td>   X</td>\n",
        "      <td>        p.Y144fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19</th>\n",
        "      <td>  0046-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>               Angiomatous</td>\n",
        "      <td> 62</td>\n",
        "      <td> M</td>\n",
        "      <td>           Left sphenoid wing</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>20</th>\n",
        "      <td>  0047-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 49</td>\n",
        "      <td> M</td>\n",
        "      <td>               Left occipital</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>21</th>\n",
        "      <td>  0056-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 40</td>\n",
        "      <td> F</td>\n",
        "      <td>           Right Petrous apex</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22</th>\n",
        "      <td>  0058-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Psammomatous</td>\n",
        "      <td> 60</td>\n",
        "      <td> F</td>\n",
        "      <td>           Right parasagittal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23</th>\n",
        "      <td>  0059-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 47</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>         p.Q18fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>24</th>\n",
        "      <td>  0060-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 57</td>\n",
        "      <td> F</td>\n",
        "      <td>               Right temporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>         p.R196*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25</th>\n",
        "      <td>  0062-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 69</td>\n",
        "      <td> F</td>\n",
        "      <td>                Left temporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>        p.S444fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>26</th>\n",
        "      <td>  0063-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 42</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>27</th>\n",
        "      <td>  0065-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 59</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>28</th>\n",
        "      <td>  0066-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 77</td>\n",
        "      <td> F</td>\n",
        "      <td>         Right frontotemporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>29</th>\n",
        "      <td>  0067-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 45</td>\n",
        "      <td> F</td>\n",
        "      <td>                Left temporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td> p.E17K</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>35</th>\n",
        "      <td>  0073-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 86</td>\n",
        "      <td> F</td>\n",
        "      <td> Right occipital/suboccipital</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>36</th>\n",
        "      <td>  0074-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 28</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>37</th>\n",
        "      <td>  0075-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 56</td>\n",
        "      <td> F</td>\n",
        "      <td>               Right CP angle</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>38</th>\n",
        "      <td>  0076-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Fibroblastic</td>\n",
        "      <td> 54</td>\n",
        "      <td> F</td>\n",
        "      <td>       Parasagittal-occipital</td>\n",
        "      <td> NaN</td>\n",
        "      <td>         p.Q324*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>39</th>\n",
        "      <td>  0077-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Psammomatous</td>\n",
        "      <td> 64</td>\n",
        "      <td> F</td>\n",
        "      <td>                      Falcine</td>\n",
        "      <td> NaN</td>\n",
        "      <td>   p.K149_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>40</th>\n",
        "      <td>  0078-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Psammomatous</td>\n",
        "      <td> 79</td>\n",
        "      <td> F</td>\n",
        "      <td>    Tentorial/Posterior fossa</td>\n",
        "      <td> NaN</td>\n",
        "      <td>          p.R57*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>41</th>\n",
        "      <td>  0079-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 48</td>\n",
        "      <td> F</td>\n",
        "      <td>           Planum sphenoidale</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>42</th>\n",
        "      <td>  0080-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 52</td>\n",
        "      <td> F</td>\n",
        "      <td>             Cervical (C1-C2)</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td> p.E17K</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>43</th>\n",
        "      <td>  0081-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 63</td>\n",
        "      <td> M</td>\n",
        "      <td>        Left olfactory groove</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td> p.L412F</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>44</th>\n",
        "      <td>  0082-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 63</td>\n",
        "      <td> M</td>\n",
        "      <td>         Right frontotemporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td> p.E17K</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>45</th>\n",
        "      <td>  0083-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>            Meningothelial</td>\n",
        "      <td> 56</td>\n",
        "      <td> F</td>\n",
        "      <td>         Left cavernous sinus</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>46</th>\n",
        "      <td>  0085-MEN</td>\n",
        "      <td>   I</td>\n",
        "      <td>              Transitional</td>\n",
        "      <td> 70</td>\n",
        "      <td> M</td>\n",
        "      <td>                Left Parietal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>          p.R57*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>47</th>\n",
        "      <td>  0024-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 47</td>\n",
        "      <td> F</td>\n",
        "      <td>               Right temporal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>        p.I521fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>48</th>\n",
        "      <td>  0025-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 45</td>\n",
        "      <td> F</td>\n",
        "      <td>    Right lateral ventricular</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>49</th>\n",
        "      <td>  0039-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 61</td>\n",
        "      <td> F</td>\n",
        "      <td>            Right parafalcine</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>50</th>\n",
        "      <td>  0041-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 67</td>\n",
        "      <td> F</td>\n",
        "      <td>         Right Frontotemporal</td>\n",
        "      <td>   X</td>\n",
        "      <td>        p.I417fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>51</th>\n",
        "      <td>  0042-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 79</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Parasagittal</td>\n",
        "      <td>   X</td>\n",
        "      <td>          p.E32*</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>52</th>\n",
        "      <td>  0044-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                 Atypical </td>\n",
        "      <td> 57</td>\n",
        "      <td> F</td>\n",
        "      <td>              Right occipital</td>\n",
        "      <td> NaN</td>\n",
        "      <td>    p.V81_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>53</th>\n",
        "      <td>  0050-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 84</td>\n",
        "      <td> F</td>\n",
        "      <td>          Right Sphenoid Wing</td>\n",
        "      <td> NaN</td>\n",
        "      <td>    p.V81_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>54</th>\n",
        "      <td>  0051-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 65</td>\n",
        "      <td> F</td>\n",
        "      <td>                Right Falcine</td>\n",
        "      <td> NaN</td>\n",
        "      <td>   p.K149_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>55</th>\n",
        "      <td>  0052-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 75</td>\n",
        "      <td> M</td>\n",
        "      <td>         Right frontoparietal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>   p.R200_splice</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>56</th>\n",
        "      <td>  0053-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 40</td>\n",
        "      <td> M</td>\n",
        "      <td>       Right frontal invasive</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>57</th>\n",
        "      <td>  0054-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 42</td>\n",
        "      <td> F</td>\n",
        "      <td>            Bilateral falcine</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>58</th>\n",
        "      <td>  0055-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 85</td>\n",
        "      <td> F</td>\n",
        "      <td>          Right Sphenoid Wing</td>\n",
        "      <td> NaN</td>\n",
        "      <td>        p.L214fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>59</th>\n",
        "      <td>  0061-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 60</td>\n",
        "      <td> M</td>\n",
        "      <td>         Right foramen magnum</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>60</th>\n",
        "      <td>  0064-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 66</td>\n",
        "      <td> F</td>\n",
        "      <td>                 Left Falcine</td>\n",
        "      <td> NaN</td>\n",
        "      <td>        p.E386fs</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>61</th>\n",
        "      <td>  0084-MEN</td>\n",
        "      <td>  II</td>\n",
        "      <td>                  Atypical</td>\n",
        "      <td> 40</td>\n",
        "      <td> F</td>\n",
        "      <td>         Falcine/parasagittal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>62</th>\n",
        "      <td>  0048-MEN</td>\n",
        "      <td> III</td>\n",
        "      <td>                Anaplastic</td>\n",
        "      <td> 69</td>\n",
        "      <td> F</td>\n",
        "      <td>       Parasagittal occipital</td>\n",
        "      <td>   X</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>63</th>\n",
        "      <td>  0049-MEN</td>\n",
        "      <td> III</td>\n",
        "      <td>                Anaplastic</td>\n",
        "      <td> 46</td>\n",
        "      <td> M</td>\n",
        "      <td>               Left occipital</td>\n",
        "      <td>   X</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>64</th>\n",
        "      <td>  0057-MEN</td>\n",
        "      <td> III</td>\n",
        "      <td>                Anaplastic</td>\n",
        "      <td> 58</td>\n",
        "      <td> M</td>\n",
        "      <td>                 Left Frontal</td>\n",
        "      <td> NaN</td>\n",
        "      <td>             NaN</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>65 rows \u00d7 10 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 2,
       "text": [
        "    Sample ID Grade                  Histology  Age Sex  \\\n",
        "0    0008-MEN     I  Fibroblastic-Transitional   65   F   \n",
        "1    0009-MEN     I               Fibroblastic   29   F   \n",
        "2    0011-MEN     I               Transitional   55   F   \n",
        "3    0015-MEN     I             Meningothelial   41   F   \n",
        "4    0016-MEN     I               Fibroblastic   47   F   \n",
        "5    0017-MEN     I               Transitional   71   F   \n",
        "6    0018-MEN     I               Transitional   69   F   \n",
        "7    0019-MEN     I               Fibroblastic   51   F   \n",
        "8    0020-MEN     I               Fibroblastic   48   F   \n",
        "9    0021-MEN     I               Transitional   31   F   \n",
        "10   0022-MEN     I  Fibroblastic-Transitional   61   F   \n",
        "11   MENex001     I  Fibroblastic-Transitional   61   F   \n",
        "12   MENex002     I               Fibroblastic   54   F   \n",
        "13  MENex003m     I  Fibroblastic-Transitional   35   F   \n",
        "14   MENex004     I    Angiomatous-Microcystic   61   F   \n",
        "15   MENex005     I    Angiomatous-Microcystic   46   M   \n",
        "16   MENex006     I             Meningothelial   28   F   \n",
        "17   0043-MEN     I               Transitional   45   M   \n",
        "18   0045-MEN     I               Fibroblastic   67   F   \n",
        "19   0046-MEN     I                Angiomatous   62   M   \n",
        "20   0047-MEN     I               Fibroblastic   49   M   \n",
        "21   0056-MEN     I               Transitional   40   F   \n",
        "22   0058-MEN     I               Psammomatous   60   F   \n",
        "23   0059-MEN     I               Fibroblastic   47   F   \n",
        "24   0060-MEN     I               Fibroblastic   57   F   \n",
        "25   0062-MEN     I               Transitional   69   F   \n",
        "26   0063-MEN     I               Fibroblastic   42   F   \n",
        "27   0065-MEN     I             Meningothelial   59   F   \n",
        "28   0066-MEN     I             Meningothelial   77   F   \n",
        "29   0067-MEN     I             Meningothelial   45   F   \n",
        "..        ...   ...                        ...  ...  ..   \n",
        "35   0073-MEN     I               Fibroblastic   86   F   \n",
        "36   0074-MEN     I               Fibroblastic   28   F   \n",
        "37   0075-MEN     I             Meningothelial   56   F   \n",
        "38   0076-MEN     I               Fibroblastic   54   F   \n",
        "39   0077-MEN     I               Psammomatous   64   F   \n",
        "40   0078-MEN     I               Psammomatous   79   F   \n",
        "41   0079-MEN     I               Transitional   48   F   \n",
        "42   0080-MEN     I             Meningothelial   52   F   \n",
        "43   0081-MEN     I             Meningothelial   63   M   \n",
        "44   0082-MEN     I             Meningothelial   63   M   \n",
        "45   0083-MEN     I             Meningothelial   56   F   \n",
        "46   0085-MEN     I               Transitional   70   M   \n",
        "47   0024-MEN    II                   Atypical   47   F   \n",
        "48   0025-MEN    II                   Atypical   45   F   \n",
        "49   0039-MEN    II                   Atypical   61   F   \n",
        "50   0041-MEN    II                   Atypical   67   F   \n",
        "51   0042-MEN    II                   Atypical   79   F   \n",
        "52   0044-MEN    II                  Atypical    57   F   \n",
        "53   0050-MEN    II                   Atypical   84   F   \n",
        "54   0051-MEN    II                   Atypical   65   F   \n",
        "55   0052-MEN    II                   Atypical   75   M   \n",
        "56   0053-MEN    II                   Atypical   40   M   \n",
        "57   0054-MEN    II                   Atypical   42   F   \n",
        "58   0055-MEN    II                   Atypical   85   F   \n",
        "59   0061-MEN    II                   Atypical   60   M   \n",
        "60   0064-MEN    II                   Atypical   66   F   \n",
        "61   0084-MEN    II                   Atypical   40   F   \n",
        "62   0048-MEN   III                 Anaplastic   69   F   \n",
        "63   0049-MEN   III                 Anaplastic   46   M   \n",
        "64   0057-MEN   III                 Anaplastic   58   M   \n",
        "\n",
        "                        Location Recurrent              NF2    AKT1      SMO  \n",
        "0                  Left temporal       NaN    p.R200_splice     NaN      NaN  \n",
        "1           Right frontotemporal       NaN  NF2-TCF20fusion     NaN      NaN  \n",
        "2                Right occipital       NaN              NaN  p.E17K      NaN  \n",
        "3                  Right frontal       NaN              NaN     NaN  p.W535L  \n",
        "4                   Left Frontal       NaN    p.Y150_splice     NaN      NaN  \n",
        "5             Right parasagittal       NaN         p.K227fs     NaN      NaN  \n",
        "6                   Left Frontal       NaN          p.Y144*     NaN      NaN  \n",
        "7           Left Posterior fossa       NaN         p.R160fs     NaN      NaN  \n",
        "8                 Right parietal       NaN          p.R341*     NaN      NaN  \n",
        "9                  Right frontal       NaN              NaN     NaN      NaN  \n",
        "10            Right parasagittal       NaN     p.K80_splice     NaN      NaN  \n",
        "11            Right parasagittal       NaN              NaN     NaN      NaN  \n",
        "12                  Left Frontal       NaN           p.W41*     NaN      NaN  \n",
        "13                  Left P-fossa       NaN    p.Y150_splice     NaN      NaN  \n",
        "14                  Left frontal       NaN              NaN     NaN      NaN  \n",
        "15                 Right frontal       NaN              NaN     NaN      NaN  \n",
        "16                  Left clinoid       NaN              NaN     NaN  p.W535L  \n",
        "17                 Right frontal       NaN              NaN     NaN      NaN  \n",
        "18   Right Frontal and tentorial         X         p.Y144fs     NaN      NaN  \n",
        "19            Left sphenoid wing       NaN              NaN     NaN      NaN  \n",
        "20                Left occipital       NaN              NaN     NaN      NaN  \n",
        "21            Right Petrous apex       NaN              NaN     NaN      NaN  \n",
        "22            Right parasagittal       NaN              NaN     NaN      NaN  \n",
        "23                  Left Frontal       NaN          p.Q18fs     NaN      NaN  \n",
        "24                Right temporal       NaN          p.R196*     NaN      NaN  \n",
        "25                 Left temporal       NaN         p.S444fs     NaN      NaN  \n",
        "26                  Left Frontal       NaN              NaN     NaN      NaN  \n",
        "27                  Left Frontal       NaN              NaN     NaN      NaN  \n",
        "28          Right frontotemporal       NaN              NaN     NaN      NaN  \n",
        "29                 Left temporal       NaN              NaN  p.E17K      NaN  \n",
        "..                           ...       ...              ...     ...      ...  \n",
        "35  Right occipital/suboccipital       NaN              NaN     NaN      NaN  \n",
        "36                  Left Frontal       NaN              NaN     NaN      NaN  \n",
        "37                Right CP angle       NaN              NaN     NaN      NaN  \n",
        "38        Parasagittal-occipital       NaN          p.Q324*     NaN      NaN  \n",
        "39                       Falcine       NaN    p.K149_splice     NaN      NaN  \n",
        "40     Tentorial/Posterior fossa       NaN           p.R57*     NaN      NaN  \n",
        "41            Planum sphenoidale       NaN              NaN     NaN      NaN  \n",
        "42              Cervical (C1-C2)       NaN              NaN  p.E17K      NaN  \n",
        "43         Left olfactory groove       NaN              NaN     NaN  p.L412F  \n",
        "44          Right frontotemporal       NaN              NaN  p.E17K      NaN  \n",
        "45          Left cavernous sinus       NaN              NaN     NaN      NaN  \n",
        "46                 Left Parietal       NaN           p.R57*     NaN      NaN  \n",
        "47                Right temporal       NaN         p.I521fs     NaN      NaN  \n",
        "48     Right lateral ventricular       NaN              NaN     NaN      NaN  \n",
        "49             Right parafalcine       NaN              NaN     NaN      NaN  \n",
        "50          Right Frontotemporal         X         p.I417fs     NaN      NaN  \n",
        "51                  Parasagittal         X           p.E32*     NaN      NaN  \n",
        "52               Right occipital       NaN     p.V81_splice     NaN      NaN  \n",
        "53           Right Sphenoid Wing       NaN     p.V81_splice     NaN      NaN  \n",
        "54                 Right Falcine       NaN    p.K149_splice     NaN      NaN  \n",
        "55          Right frontoparietal       NaN    p.R200_splice     NaN      NaN  \n",
        "56        Right frontal invasive       NaN              NaN     NaN      NaN  \n",
        "57             Bilateral falcine       NaN              NaN     NaN      NaN  \n",
        "58           Right Sphenoid Wing       NaN         p.L214fs     NaN      NaN  \n",
        "59          Right foramen magnum       NaN              NaN     NaN      NaN  \n",
        "60                  Left Falcine       NaN         p.E386fs     NaN      NaN  \n",
        "61          Falcine/parasagittal       NaN              NaN     NaN      NaN  \n",
        "62        Parasagittal occipital         X              NaN     NaN      NaN  \n",
        "63                Left occipital         X              NaN     NaN      NaN  \n",
        "64                  Left Frontal       NaN              NaN     NaN      NaN  \n",
        "\n",
        "[65 rows x 10 columns]"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Mutation Data:"
     ]
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Copy Number Data:"
     ]
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Other Data:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# WGS Somatic Alterations in Meningiomas\n",
      "somaticrearrange = pd.read_csv(PATH + 'ng.2526-S3.csv', index_col=0).dropna()\n",
      "somaticrearrange"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>class</th>\n",
        "      <th>chrA</th>\n",
        "      <th>strA</th>\n",
        "      <th>posA</th>\n",
        "      <th>siteA</th>\n",
        "      <th>chrB</th>\n",
        "      <th>strB</th>\n",
        "      <th>posB</th>\n",
        "      <th>siteB</th>\n",
        "      <th>fusion</th>\n",
        "      <th>quality</th>\n",
        "      <th>score</th>\n",
        "      <th>BP</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Sample</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>         long-range</td>\n",
        "      <td> chr13</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  96492194</td>\n",
        "      <td>       Intron of UGGT2(-): 3Kb before exon 37</td>\n",
        "      <td> chr13</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 102254383</td>\n",
        "      <td>         Intron of ITGBL1(+): 4Kb after exon 7</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 44.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 173968985</td>\n",
        "      <td>                      IGR: 7Kb after RC3H1(-)</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  17746466</td>\n",
        "      <td>                      IGR: 56Kb after CECR1(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 42.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr2</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 170785074</td>\n",
        "      <td>      Intron of UBR3(+): 149bp before exon 18</td>\n",
        "      <td>  chr7</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  90676069</td>\n",
        "      <td>       Intron of CDK14(+): 791bp after exon 10</td>\n",
        "      <td>      Protein fusion: out of frame (CDK14-UBR3)</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 42.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td> chr13</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  96492190</td>\n",
        "      <td>       Intron of UGGT2(-): 3Kb before exon 37</td>\n",
        "      <td> chr20</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  47589665</td>\n",
        "      <td>      Intron of ARFGEF2(+): 9bp before exon 12</td>\n",
        "      <td>   Protein fusion: out of frame (UGGT2-ARFGEF2)</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 40.84</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>           deletion</td>\n",
        "      <td> chr18</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  51130926</td>\n",
        "      <td>                       IGR: 69Kb after DCC(+)</td>\n",
        "      <td> chr18</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  51363938</td>\n",
        "      <td>                     IGR: 317Kb before MBD2(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 38.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr2</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 170785078</td>\n",
        "      <td>      Intron of UBR3(+): 195bp before exon 18</td>\n",
        "      <td>  chr7</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  90676065</td>\n",
        "      <td>       Intron of CDK14(+): 826bp after exon 10</td>\n",
        "      <td>      Protein fusion: out of frame (UBR3-CDK14)</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 36.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chr9</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  22832947</td>\n",
        "      <td>                  IGR: 386Kb before DMRTA1(+)</td>\n",
        "      <td>  chr9</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  22959505</td>\n",
        "      <td>                   IGR: 513Kb before DMRTA1(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.76</td>\n",
        "      <td> 35.91</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>         long-range</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  30031940</td>\n",
        "      <td>        Intron of NF2(+): 800bp before exon 2</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  42592999</td>\n",
        "      <td>         Intron of TCF20(-): 13Kb after exon 1</td>\n",
        "      <td>           Protein fusion: in frame (NF2-TCF20)</td>\n",
        "      <td> 0.91</td>\n",
        "      <td> 31.88</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr7</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  61559903</td>\n",
        "      <td>                 IGR: 1Mb before LOC643955(-)</td>\n",
        "      <td> chr17</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  28819003</td>\n",
        "      <td>       Intron of GOSR1(+): 693bp before exon 6</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.96</td>\n",
        "      <td> 31.60</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr7</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  52209818</td>\n",
        "      <td>               IGR: 894Kb before POM121L12(+)</td>\n",
        "      <td> chr17</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  29508909</td>\n",
        "      <td>           Intron of NF1(+): 40bp after exon 7</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 26.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr9</td>\n",
        "      <td> (+)</td>\n",
        "      <td>   2512101</td>\n",
        "      <td>                 IGR: 24Kb before FLJ35024(-)</td>\n",
        "      <td> chr17</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  42437498</td>\n",
        "      <td>     Intron of FAM171A2(-): 25bp before exon 2</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 26.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td> chr13</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 102254374</td>\n",
        "      <td>        Intron of ITGBL1(+): 4Kb after exon 7</td>\n",
        "      <td> chr20</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  47589664</td>\n",
        "      <td>     Intron of ARFGEF2(+): 16bp before exon 12</td>\n",
        "      <td>                               Antisense fusion</td>\n",
        "      <td> 0.41</td>\n",
        "      <td> 12.80</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>         long-range</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  30031943</td>\n",
        "      <td>        Intron of NF2(+): 789bp before exon 2</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  42592992</td>\n",
        "      <td>         Intron of TCF20(-): 13Kb after exon 1</td>\n",
        "      <td>           Protein fusion: in frame (TCF20-NF2)</td>\n",
        "      <td> 0.36</td>\n",
        "      <td> 10.77</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 161079067</td>\n",
        "      <td>        Intron of PFDN2(-): 7Kb before exon 2</td>\n",
        "      <td> chr17</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  32205988</td>\n",
        "      <td>        Intron of ACCN1(-): 277Kb after exon 1</td>\n",
        "      <td>         Protein fusion: in frame (ACCN1-PFDN2)</td>\n",
        "      <td> 0.70</td>\n",
        "      <td>  8.44</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0009</th>\n",
        "      <td>          inversion</td>\n",
        "      <td>  chr9</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  14565686</td>\n",
        "      <td>                  IGR: 45Kb before ZDHHC21(-)</td>\n",
        "      <td>  chr9</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  14627235</td>\n",
        "      <td>       Intron of ZDHHC21(-): 8Kb before exon 9</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td>  8.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0011</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chr2</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 141080158</td>\n",
        "      <td>     Intron of LRP1B(-): 497bp before exon 82</td>\n",
        "      <td>  chr2</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 141127821</td>\n",
        "      <td>       Intron of LRP1B(-): 455bp after exon 71</td>\n",
        "      <td>                 Deletion of 10 exons: in frame</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 54.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0015</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chr4</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 180913022</td>\n",
        "      <td>                  IGR: 2Mb before MGC45800(-)</td>\n",
        "      <td>  chr4</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 180990563</td>\n",
        "      <td>                   IGR: 2Mb before MGC45800(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 24.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  80589104</td>\n",
        "      <td>                     IGR: 1Mb before ELTD1(-)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  90870420</td>\n",
        "      <td>                   IGR: 307Kb before BARHL2(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.99</td>\n",
        "      <td> 67.65</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  68000835</td>\n",
        "      <td>                  IGR: 127Kb before SERBP1(-)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  95538220</td>\n",
        "      <td>        Intron of ALG14(-): 100bp after exon 1</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 32.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  48880153</td>\n",
        "      <td>       Intron of SPATA6(-): 1Kb before exon 4</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 171741093</td>\n",
        "      <td>                   IGR: 10Kb before METTL13(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.96</td>\n",
        "      <td> 31.71</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  48881168</td>\n",
        "      <td>       Intron of SPATA6(-): 2Kb before exon 4</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  77612749</td>\n",
        "      <td>           Intron of PIGK(-): 7Kb after exon 9</td>\n",
        "      <td>     Protein fusion: out of frame (PIGK-SPATA6)</td>\n",
        "      <td> 0.88</td>\n",
        "      <td> 31.65</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 158828097</td>\n",
        "      <td>                     IGR: 27Kb before MNDA(+)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 187549966</td>\n",
        "      <td>                  IGR: 752Kb before PLA2G4A(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 30.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 109163626</td>\n",
        "      <td>      Intron of FAM102B(+): 4Kb before exon 6</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 187277742</td>\n",
        "      <td>                  IGR: 480Kb before PLA2G4A(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.98</td>\n",
        "      <td> 29.31</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  58421895</td>\n",
        "      <td> 5'-UTR of DAB1(-): 665Kb before coding start</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 118155121</td>\n",
        "      <td> 5'-UTR of FAM46C(+): 10Kb before coding start</td>\n",
        "      <td>                Transcript fusion (DAB1-FAM46C)</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 27.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  80323486</td>\n",
        "      <td>                   IGR: 968Kb before ELTD1(-)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 100225718</td>\n",
        "      <td>  5'-UTR of FRRS1(-): 11Kb before coding start</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 27.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  95875799</td>\n",
        "      <td>                   IGR: 176Kb before RWDD3(+)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 171930057</td>\n",
        "      <td>         Intron of DNM3(+): 27Kb before exon 3</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 25.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  77604362</td>\n",
        "      <td>         Intron of PIGK(-): 16Kb after exon 9</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 171918762</td>\n",
        "      <td>          Intron of DNM3(+): 28Kb after exon 2</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 24.97</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 100228338</td>\n",
        "      <td> 5'-UTR of FRRS1(-): 14Kb before coding start</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 179526744</td>\n",
        "      <td>       Intron of NPHS2(-): 422bp before exon 5</td>\n",
        "      <td>                               Antisense fusion</td>\n",
        "      <td> 0.99</td>\n",
        "      <td> 24.82</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  80320704</td>\n",
        "      <td>                   IGR: 965Kb before ELTD1(-)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 171918676</td>\n",
        "      <td>          Intron of DNM3(+): 28Kb after exon 2</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 24.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0017</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 171933392</td>\n",
        "      <td>        Intron of DNM3(+): 23Kb before exon 3</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 179641879</td>\n",
        "      <td>         Intron of TDRD5(+): 3Kb after exon 16</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 24.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td> chr10</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  60122505</td>\n",
        "      <td>     Intron of UBE2D1(+): 867bp before exon 4</td>\n",
        "      <td> chr11</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  26974158</td>\n",
        "      <td>                     IGR: 41Kb before FIBIN(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 86.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr5</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  58568666</td>\n",
        "      <td>       Intron of PDE4D(-): 57Kb before exon 2</td>\n",
        "      <td> chr11</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  26968714</td>\n",
        "      <td>                     IGR: 47Kb before FIBIN(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 82.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 115593652</td>\n",
        "      <td>     Intron of TSPAN2(-): 474bp before exon 8</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 117389748</td>\n",
        "      <td>                    IGR: 63Kb before PTGFRN(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 79.95</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  74757871</td>\n",
        "      <td>       Intron of TNNI3K(+): 20Kb after exon 5</td>\n",
        "      <td> chr12</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  30092371</td>\n",
        "      <td>                    IGR: 439Kb before TMTC1(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 75.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td> tandem duplication</td>\n",
        "      <td>  chr2</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  28872217</td>\n",
        "      <td>                       IGR: 6Kb after PLB1(+)</td>\n",
        "      <td>  chr2</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  29013760</td>\n",
        "      <td>         Intron of PPP1CB(+): 2Kb after exon 6</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 75.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 111688954</td>\n",
        "      <td>  5'-UTR of CEPT1(+): 1Kb before coding start</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 116484929</td>\n",
        "      <td>                  IGR: 34Kb before SLC22A15(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 73.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 115594602</td>\n",
        "      <td>        Intron of TSPAN2(-): 1Kb after exon 7</td>\n",
        "      <td> chr12</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  32139256</td>\n",
        "      <td>     Intron of C12orf35(+): 383bp after exon 4</td>\n",
        "      <td> Protein fusion: out of frame (TSPAN2-C12orf35)</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 73.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  73057403</td>\n",
        "      <td>                    IGR: 309Kb after NEGR1(-)</td>\n",
        "      <td> chr12</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  30692838</td>\n",
        "      <td>                      IGR: 89Kb before IPO8(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 71.93</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 102099811</td>\n",
        "      <td>                    IGR: 54Kb before ZPLD1(+)</td>\n",
        "      <td>  chr8</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  39701830</td>\n",
        "      <td>                       IGR: 6Kb after ADAM2(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 70.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>           deletion</td>\n",
        "      <td> chr12</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  30209562</td>\n",
        "      <td>                    IGR: 272Kb after TMTC1(-)</td>\n",
        "      <td> chr12</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  30672468</td>\n",
        "      <td>                     IGR: 109Kb before IPO8(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 68.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  72435934</td>\n",
        "      <td>       Intron of NEGR1(-): 35Kb before exon 2</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 101900278</td>\n",
        "      <td>                IGR: 241Kb before LOC152225(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.99</td>\n",
        "      <td> 64.30</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  74905605</td>\n",
        "      <td>     Intron of TNNI3K(+): 309bp after exon 20</td>\n",
        "      <td>  chr5</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  58568785</td>\n",
        "      <td>        Intron of PDE4D(-): 57Kb before exon 2</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.97</td>\n",
        "      <td> 63.87</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  73057529</td>\n",
        "      <td>                    IGR: 309Kb after NEGR1(-)</td>\n",
        "      <td> chr11</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  26940286</td>\n",
        "      <td>                     IGR: 75Kb before FIBIN(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 59.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  74246366</td>\n",
        "      <td>                  IGR: 245Kb before LRRIQ3(-)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 116485762</td>\n",
        "      <td>                  IGR: 33Kb before SLC22A15(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 51.88</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>           deletion</td>\n",
        "      <td> chr10</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  60093216</td>\n",
        "      <td>     Promoter of UBE2D1(+): 2Kb from tx start</td>\n",
        "      <td> chr10</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  60122801</td>\n",
        "      <td>      Intron of UBE2D1(+): 570bp before exon 4</td>\n",
        "      <td>                     Deletion within transcript</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 51.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 138548203</td>\n",
        "      <td>                   IGR: 115Kb before FOXL2(-)</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 138552125</td>\n",
        "      <td>                    IGR: 111Kb before FOXL2(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 49.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>           deletion</td>\n",
        "      <td> chr19</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  12465971</td>\n",
        "      <td>       Intron of ZNF442(-): 2Kb before exon 4</td>\n",
        "      <td> chr19</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  12527943</td>\n",
        "      <td>                    IGR: 13Kb before ZNF443(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.75</td>\n",
        "      <td> 23.31</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 111687750</td>\n",
        "      <td>  5'-UTR of CEPT1(+): 3Kb before coding start</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 117387940</td>\n",
        "      <td>                    IGR: 65Kb before PTGFRN(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.98</td>\n",
        "      <td> 18.71</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 117795848</td>\n",
        "      <td>                     IGR: 42Kb after VTCN1(-)</td>\n",
        "      <td> chr12</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  31639393</td>\n",
        "      <td>       Intron of DENND5B(-): 6Kb before exon 3</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 18.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 117387942</td>\n",
        "      <td>                   IGR: 65Kb before PTGFRN(+)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 118487439</td>\n",
        "      <td>         Intron of WDR3(+): 1Kb before exon 12</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 17.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 111695748</td>\n",
        "      <td>         Intron of CEPT1(+): 5Kb after exon 2</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 117796055</td>\n",
        "      <td>                      IGR: 43Kb after VTCN1(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 16.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0018</th>\n",
        "      <td>          inversion</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 116486182</td>\n",
        "      <td>                 IGR: 33Kb before SLC22A15(+)</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 116620887</td>\n",
        "      <td>                  IGR: 33Kb before C1orf161(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 0.97</td>\n",
        "      <td> 10.66</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0020</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chrX</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  17772190</td>\n",
        "      <td>  3'-UTR of SCML1(+): 680bp after coding stop</td>\n",
        "      <td>  chrX</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  18230617</td>\n",
        "      <td>         Intron of BEND2(-): 64bp after exon 4</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 67.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0020</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  95447521</td>\n",
        "      <td>               IGR: 790Kb before LOC255025(+)</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  96696128</td>\n",
        "      <td>        Intron of EPHA6(+): 10Kb before exon 3</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 15.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0021</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chr4</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 181080822</td>\n",
        "      <td>                  IGR: 2Mb before MGC45800(-)</td>\n",
        "      <td>  chr4</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 181088399</td>\n",
        "      <td>                   IGR: 2Mb before MGC45800(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 71.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0021</th>\n",
        "      <td>           deletion</td>\n",
        "      <td> chr13</td>\n",
        "      <td> (+)</td>\n",
        "      <td>  85400254</td>\n",
        "      <td>                 IGR: 949Kb before SLITRK1(-)</td>\n",
        "      <td> chr13</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  85411894</td>\n",
        "      <td>                  IGR: 955Kb before SLITRK6(-)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 70.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0021</th>\n",
        "      <td>  inter-chromosomal</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 156102230</td>\n",
        "      <td>          Intron of LMNA(+): 2Kb after exon 2</td>\n",
        "      <td> chr19</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  45973114</td>\n",
        "      <td>        Intron of FOSB(+): 772bp before exon 2</td>\n",
        "      <td>           Protein fusion: in frame (LMNA-FOSB)</td>\n",
        "      <td> 0.82</td>\n",
        "      <td> 42.60</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0022</th>\n",
        "      <td>           deletion</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 110928019</td>\n",
        "      <td>                   IGR: 137Kb before PVRL3(+)</td>\n",
        "      <td>  chr3</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 110932451</td>\n",
        "      <td>                    IGR: 142Kb before PVRL3(+)</td>\n",
        "      <td>                                              -</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 46.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0041</th>\n",
        "      <td>         long-range</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (+)</td>\n",
        "      <td> 156503947</td>\n",
        "      <td>     Intron of IQGAP3(-): 18bp before exon 30</td>\n",
        "      <td>  chr1</td>\n",
        "      <td> (-)</td>\n",
        "      <td> 157559121</td>\n",
        "      <td>                            Exon 3 of FCRL4(-)</td>\n",
        "      <td>        Protein fusion: mid-exon (FCRL4-IQGAP3)</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 17.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEN0048</th>\n",
        "      <td>          inversion</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  29000292</td>\n",
        "      <td>        Intron of TTC28(-): 25Kb after exon 2</td>\n",
        "      <td> chr22</td>\n",
        "      <td> (-)</td>\n",
        "      <td>  29091654</td>\n",
        "      <td>        Intron of CHEK2(-): 48bp after exon 11</td>\n",
        "      <td>                               Antisense fusion</td>\n",
        "      <td> 1.00</td>\n",
        "      <td> 14.00</td>\n",
        "      <td> +</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>95 rows \u00d7 13 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 2,
       "text": [
        "                      class   chrA strA       posA  \\\n",
        "Sample                                               \n",
        "MEN0009          long-range  chr13  (+)   96492194   \n",
        "MEN0009   inter-chromosomal   chr1  (+)  173968985   \n",
        "MEN0009   inter-chromosomal   chr2  (-)  170785074   \n",
        "MEN0009   inter-chromosomal  chr13  (-)   96492190   \n",
        "MEN0009            deletion  chr18  (+)   51130926   \n",
        "MEN0009   inter-chromosomal   chr2  (+)  170785078   \n",
        "MEN0009            deletion   chr9  (+)   22832947   \n",
        "MEN0009          long-range  chr22  (+)   30031940   \n",
        "MEN0009   inter-chromosomal   chr7  (+)   61559903   \n",
        "MEN0009   inter-chromosomal   chr7  (-)   52209818   \n",
        "MEN0009   inter-chromosomal   chr9  (+)    2512101   \n",
        "MEN0009   inter-chromosomal  chr13  (+)  102254374   \n",
        "MEN0009          long-range  chr22  (-)   30031943   \n",
        "MEN0009   inter-chromosomal   chr1  (+)  161079067   \n",
        "MEN0009           inversion   chr9  (-)   14565686   \n",
        "MEN0011            deletion   chr2  (+)  141080158   \n",
        "MEN0015            deletion   chr4  (+)  180913022   \n",
        "MEN0017          long-range   chr1  (+)   80589104   \n",
        "MEN0017          long-range   chr1  (-)   68000835   \n",
        "MEN0017          long-range   chr1  (-)   48880153   \n",
        "MEN0017          long-range   chr1  (+)   48881168   \n",
        "MEN0017          long-range   chr1  (+)  158828097   \n",
        "MEN0017          long-range   chr1  (-)  109163626   \n",
        "MEN0017          long-range   chr1  (-)   58421895   \n",
        "MEN0017          long-range   chr1  (-)   80323486   \n",
        "MEN0017          long-range   chr1  (+)   95875799   \n",
        "MEN0017          long-range   chr1  (+)   77604362   \n",
        "MEN0017          long-range   chr1  (-)  100228338   \n",
        "MEN0017          long-range   chr1  (-)   80320704   \n",
        "MEN0017          long-range   chr1  (-)  171933392   \n",
        "...                     ...    ...  ...        ...   \n",
        "MEN0018   inter-chromosomal  chr10  (+)   60122505   \n",
        "MEN0018   inter-chromosomal   chr5  (-)   58568666   \n",
        "MEN0018          long-range   chr1  (-)  115593652   \n",
        "MEN0018   inter-chromosomal   chr1  (-)   74757871   \n",
        "MEN0018  tandem duplication   chr2  (-)   28872217   \n",
        "MEN0018          long-range   chr1  (-)  111688954   \n",
        "MEN0018   inter-chromosomal   chr1  (-)  115594602   \n",
        "MEN0018   inter-chromosomal   chr1  (-)   73057403   \n",
        "MEN0018   inter-chromosomal   chr3  (+)  102099811   \n",
        "MEN0018            deletion  chr12  (+)   30209562   \n",
        "MEN0018   inter-chromosomal   chr1  (+)   72435934   \n",
        "MEN0018   inter-chromosomal   chr1  (-)   74905605   \n",
        "MEN0018   inter-chromosomal   chr1  (+)   73057529   \n",
        "MEN0018          long-range   chr1  (+)   74246366   \n",
        "MEN0018            deletion  chr10  (+)   60093216   \n",
        "MEN0018            deletion   chr3  (+)  138548203   \n",
        "MEN0018            deletion  chr19  (+)   12465971   \n",
        "MEN0018          long-range   chr1  (-)  111687750   \n",
        "MEN0018   inter-chromosomal   chr1  (-)  117795848   \n",
        "MEN0018          long-range   chr1  (-)  117387942   \n",
        "MEN0018          long-range   chr1  (+)  111695748   \n",
        "MEN0018           inversion   chr1  (+)  116486182   \n",
        "MEN0020            deletion   chrX  (+)   17772190   \n",
        "MEN0020          long-range   chr3  (-)   95447521   \n",
        "MEN0021            deletion   chr4  (+)  181080822   \n",
        "MEN0021            deletion  chr13  (+)   85400254   \n",
        "MEN0021   inter-chromosomal   chr1  (+)  156102230   \n",
        "MEN0022            deletion   chr3  (+)  110928019   \n",
        "MEN0041          long-range   chr1  (+)  156503947   \n",
        "MEN0048           inversion  chr22  (-)   29000292   \n",
        "\n",
        "                                                siteA   chrB strB       posB  \\\n",
        "Sample                                                                         \n",
        "MEN0009        Intron of UGGT2(-): 3Kb before exon 37  chr13  (-)  102254383   \n",
        "MEN0009                       IGR: 7Kb after RC3H1(-)  chr22  (+)   17746466   \n",
        "MEN0009       Intron of UBR3(+): 149bp before exon 18   chr7  (+)   90676069   \n",
        "MEN0009        Intron of UGGT2(-): 3Kb before exon 37  chr20  (-)   47589665   \n",
        "MEN0009                        IGR: 69Kb after DCC(+)  chr18  (-)   51363938   \n",
        "MEN0009       Intron of UBR3(+): 195bp before exon 18   chr7  (-)   90676065   \n",
        "MEN0009                   IGR: 386Kb before DMRTA1(+)   chr9  (-)   22959505   \n",
        "MEN0009         Intron of NF2(+): 800bp before exon 2  chr22  (+)   42592999   \n",
        "MEN0009                  IGR: 1Mb before LOC643955(-)  chr17  (+)   28819003   \n",
        "MEN0009                IGR: 894Kb before POM121L12(+)  chr17  (+)   29508909   \n",
        "MEN0009                  IGR: 24Kb before FLJ35024(-)  chr17  (-)   42437498   \n",
        "MEN0009         Intron of ITGBL1(+): 4Kb after exon 7  chr20  (+)   47589664   \n",
        "MEN0009         Intron of NF2(+): 789bp before exon 2  chr22  (-)   42592992   \n",
        "MEN0009         Intron of PFDN2(-): 7Kb before exon 2  chr17  (-)   32205988   \n",
        "MEN0009                   IGR: 45Kb before ZDHHC21(-)   chr9  (-)   14627235   \n",
        "MEN0011      Intron of LRP1B(-): 497bp before exon 82   chr2  (-)  141127821   \n",
        "MEN0015                   IGR: 2Mb before MGC45800(-)   chr4  (-)  180990563   \n",
        "MEN0017                      IGR: 1Mb before ELTD1(-)   chr1  (-)   90870420   \n",
        "MEN0017                   IGR: 127Kb before SERBP1(-)   chr1  (+)   95538220   \n",
        "MEN0017        Intron of SPATA6(-): 1Kb before exon 4   chr1  (+)  171741093   \n",
        "MEN0017        Intron of SPATA6(-): 2Kb before exon 4   chr1  (-)   77612749   \n",
        "MEN0017                      IGR: 27Kb before MNDA(+)   chr1  (-)  187549966   \n",
        "MEN0017       Intron of FAM102B(+): 4Kb before exon 6   chr1  (+)  187277742   \n",
        "MEN0017  5'-UTR of DAB1(-): 665Kb before coding start   chr1  (-)  118155121   \n",
        "MEN0017                    IGR: 968Kb before ELTD1(-)   chr1  (-)  100225718   \n",
        "MEN0017                    IGR: 176Kb before RWDD3(+)   chr1  (+)  171930057   \n",
        "MEN0017          Intron of PIGK(-): 16Kb after exon 9   chr1  (-)  171918762   \n",
        "MEN0017  5'-UTR of FRRS1(-): 14Kb before coding start   chr1  (-)  179526744   \n",
        "MEN0017                    IGR: 965Kb before ELTD1(-)   chr1  (+)  171918676   \n",
        "MEN0017         Intron of DNM3(+): 23Kb before exon 3   chr1  (-)  179641879   \n",
        "...                                               ...    ...  ...        ...   \n",
        "MEN0018      Intron of UBE2D1(+): 867bp before exon 4  chr11  (+)   26974158   \n",
        "MEN0018        Intron of PDE4D(-): 57Kb before exon 2  chr11  (-)   26968714   \n",
        "MEN0018      Intron of TSPAN2(-): 474bp before exon 8   chr1  (+)  117389748   \n",
        "MEN0018        Intron of TNNI3K(+): 20Kb after exon 5  chr12  (-)   30092371   \n",
        "MEN0018                        IGR: 6Kb after PLB1(+)   chr2  (+)   29013760   \n",
        "MEN0018   5'-UTR of CEPT1(+): 1Kb before coding start   chr1  (+)  116484929   \n",
        "MEN0018         Intron of TSPAN2(-): 1Kb after exon 7  chr12  (-)   32139256   \n",
        "MEN0018                     IGR: 309Kb after NEGR1(-)  chr12  (-)   30692838   \n",
        "MEN0018                     IGR: 54Kb before ZPLD1(+)   chr8  (-)   39701830   \n",
        "MEN0018                     IGR: 272Kb after TMTC1(-)  chr12  (-)   30672468   \n",
        "MEN0018        Intron of NEGR1(-): 35Kb before exon 2   chr3  (-)  101900278   \n",
        "MEN0018      Intron of TNNI3K(+): 309bp after exon 20   chr5  (+)   58568785   \n",
        "MEN0018                     IGR: 309Kb after NEGR1(-)  chr11  (+)   26940286   \n",
        "MEN0018                   IGR: 245Kb before LRRIQ3(-)   chr1  (-)  116485762   \n",
        "MEN0018      Promoter of UBE2D1(+): 2Kb from tx start  chr10  (-)   60122801   \n",
        "MEN0018                    IGR: 115Kb before FOXL2(-)   chr3  (-)  138552125   \n",
        "MEN0018        Intron of ZNF442(-): 2Kb before exon 4  chr19  (-)   12527943   \n",
        "MEN0018   5'-UTR of CEPT1(+): 3Kb before coding start   chr1  (-)  117387940   \n",
        "MEN0018                      IGR: 42Kb after VTCN1(-)  chr12  (+)   31639393   \n",
        "MEN0018                    IGR: 65Kb before PTGFRN(+)   chr1  (-)  118487439   \n",
        "MEN0018          Intron of CEPT1(+): 5Kb after exon 2   chr1  (+)  117796055   \n",
        "MEN0018                  IGR: 33Kb before SLC22A15(+)   chr1  (+)  116620887   \n",
        "MEN0020   3'-UTR of SCML1(+): 680bp after coding stop   chrX  (-)   18230617   \n",
        "MEN0020                IGR: 790Kb before LOC255025(+)   chr3  (+)   96696128   \n",
        "MEN0021                   IGR: 2Mb before MGC45800(-)   chr4  (-)  181088399   \n",
        "MEN0021                  IGR: 949Kb before SLITRK1(-)  chr13  (-)   85411894   \n",
        "MEN0021           Intron of LMNA(+): 2Kb after exon 2  chr19  (-)   45973114   \n",
        "MEN0022                    IGR: 137Kb before PVRL3(+)   chr3  (-)  110932451   \n",
        "MEN0041      Intron of IQGAP3(-): 18bp before exon 30   chr1  (-)  157559121   \n",
        "MEN0048         Intron of TTC28(-): 25Kb after exon 2  chr22  (-)   29091654   \n",
        "\n",
        "                                                 siteB  \\\n",
        "Sample                                                   \n",
        "MEN0009          Intron of ITGBL1(+): 4Kb after exon 7   \n",
        "MEN0009                       IGR: 56Kb after CECR1(-)   \n",
        "MEN0009        Intron of CDK14(+): 791bp after exon 10   \n",
        "MEN0009       Intron of ARFGEF2(+): 9bp before exon 12   \n",
        "MEN0009                      IGR: 317Kb before MBD2(-)   \n",
        "MEN0009        Intron of CDK14(+): 826bp after exon 10   \n",
        "MEN0009                    IGR: 513Kb before DMRTA1(+)   \n",
        "MEN0009          Intron of TCF20(-): 13Kb after exon 1   \n",
        "MEN0009        Intron of GOSR1(+): 693bp before exon 6   \n",
        "MEN0009            Intron of NF1(+): 40bp after exon 7   \n",
        "MEN0009      Intron of FAM171A2(-): 25bp before exon 2   \n",
        "MEN0009      Intron of ARFGEF2(+): 16bp before exon 12   \n",
        "MEN0009          Intron of TCF20(-): 13Kb after exon 1   \n",
        "MEN0009         Intron of ACCN1(-): 277Kb after exon 1   \n",
        "MEN0009        Intron of ZDHHC21(-): 8Kb before exon 9   \n",
        "MEN0011        Intron of LRP1B(-): 455bp after exon 71   \n",
        "MEN0015                    IGR: 2Mb before MGC45800(-)   \n",
        "MEN0017                    IGR: 307Kb before BARHL2(-)   \n",
        "MEN0017         Intron of ALG14(-): 100bp after exon 1   \n",
        "MEN0017                    IGR: 10Kb before METTL13(+)   \n",
        "MEN0017            Intron of PIGK(-): 7Kb after exon 9   \n",
        "MEN0017                   IGR: 752Kb before PLA2G4A(+)   \n",
        "MEN0017                   IGR: 480Kb before PLA2G4A(+)   \n",
        "MEN0017  5'-UTR of FAM46C(+): 10Kb before coding start   \n",
        "MEN0017   5'-UTR of FRRS1(-): 11Kb before coding start   \n",
        "MEN0017          Intron of DNM3(+): 27Kb before exon 3   \n",
        "MEN0017           Intron of DNM3(+): 28Kb after exon 2   \n",
        "MEN0017        Intron of NPHS2(-): 422bp before exon 5   \n",
        "MEN0017           Intron of DNM3(+): 28Kb after exon 2   \n",
        "MEN0017          Intron of TDRD5(+): 3Kb after exon 16   \n",
        "...                                                ...   \n",
        "MEN0018                      IGR: 41Kb before FIBIN(+)   \n",
        "MEN0018                      IGR: 47Kb before FIBIN(+)   \n",
        "MEN0018                     IGR: 63Kb before PTGFRN(+)   \n",
        "MEN0018                     IGR: 439Kb before TMTC1(-)   \n",
        "MEN0018          Intron of PPP1CB(+): 2Kb after exon 6   \n",
        "MEN0018                   IGR: 34Kb before SLC22A15(+)   \n",
        "MEN0018      Intron of C12orf35(+): 383bp after exon 4   \n",
        "MEN0018                       IGR: 89Kb before IPO8(-)   \n",
        "MEN0018                        IGR: 6Kb after ADAM2(-)   \n",
        "MEN0018                      IGR: 109Kb before IPO8(-)   \n",
        "MEN0018                 IGR: 241Kb before LOC152225(+)   \n",
        "MEN0018         Intron of PDE4D(-): 57Kb before exon 2   \n",
        "MEN0018                      IGR: 75Kb before FIBIN(+)   \n",
        "MEN0018                   IGR: 33Kb before SLC22A15(+)   \n",
        "MEN0018       Intron of UBE2D1(+): 570bp before exon 4   \n",
        "MEN0018                     IGR: 111Kb before FOXL2(-)   \n",
        "MEN0018                     IGR: 13Kb before ZNF443(-)   \n",
        "MEN0018                     IGR: 65Kb before PTGFRN(+)   \n",
        "MEN0018        Intron of DENND5B(-): 6Kb before exon 3   \n",
        "MEN0018          Intron of WDR3(+): 1Kb before exon 12   \n",
        "MEN0018                       IGR: 43Kb after VTCN1(-)   \n",
        "MEN0018                   IGR: 33Kb before C1orf161(+)   \n",
        "MEN0020          Intron of BEND2(-): 64bp after exon 4   \n",
        "MEN0020         Intron of EPHA6(+): 10Kb before exon 3   \n",
        "MEN0021                    IGR: 2Mb before MGC45800(-)   \n",
        "MEN0021                   IGR: 955Kb before SLITRK6(-)   \n",
        "MEN0021         Intron of FOSB(+): 772bp before exon 2   \n",
        "MEN0022                     IGR: 142Kb before PVRL3(+)   \n",
        "MEN0041                             Exon 3 of FCRL4(-)   \n",
        "MEN0048         Intron of CHEK2(-): 48bp after exon 11   \n",
        "\n",
        "                                                 fusion  quality  score BP  \n",
        "Sample                                                                      \n",
        "MEN0009                                               -     1.00  44.00  +  \n",
        "MEN0009                                               -     1.00  42.00  +  \n",
        "MEN0009       Protein fusion: out of frame (CDK14-UBR3)     1.00  42.00  +  \n",
        "MEN0009    Protein fusion: out of frame (UGGT2-ARFGEF2)     1.00  40.84  +  \n",
        "MEN0009                                               -     1.00  38.00  +  \n",
        "MEN0009       Protein fusion: out of frame (UBR3-CDK14)     1.00  36.00  +  \n",
        "MEN0009                                               -     0.76  35.91  +  \n",
        "MEN0009            Protein fusion: in frame (NF2-TCF20)     0.91  31.88  +  \n",
        "MEN0009                                               -     0.96  31.60  +  \n",
        "MEN0009                                               -     1.00  26.00  +  \n",
        "MEN0009                                               -     1.00  26.00  +  \n",
        "MEN0009                                Antisense fusion     0.41  12.80  +  \n",
        "MEN0009            Protein fusion: in frame (TCF20-NF2)     0.36  10.77  +  \n",
        "MEN0009          Protein fusion: in frame (ACCN1-PFDN2)     0.70   8.44  +  \n",
        "MEN0009                                               -     1.00   8.00  +  \n",
        "MEN0011                  Deletion of 10 exons: in frame     1.00  54.00  +  \n",
        "MEN0015                                               -     1.00  24.00  +  \n",
        "MEN0017                                               -     0.99  67.65  +  \n",
        "MEN0017                                               -     1.00  32.00  +  \n",
        "MEN0017                                               -     0.96  31.71  +  \n",
        "MEN0017      Protein fusion: out of frame (PIGK-SPATA6)     0.88  31.65  +  \n",
        "MEN0017                                               -     1.00  30.00  +  \n",
        "MEN0017                                               -     0.98  29.31  +  \n",
        "MEN0017                 Transcript fusion (DAB1-FAM46C)     1.00  27.00  +  \n",
        "MEN0017                                               -     1.00  27.00  +  \n",
        "MEN0017                                               -     1.00  25.00  +  \n",
        "MEN0017                                               -     1.00  24.97  +  \n",
        "MEN0017                                Antisense fusion     0.99  24.82  +  \n",
        "MEN0017                                               -     1.00  24.00  +  \n",
        "MEN0017                                               -     1.00  24.00  +  \n",
        "...                                                 ...      ...    ... ..  \n",
        "MEN0018                                               -     1.00  86.00  +  \n",
        "MEN0018                                               -     1.00  82.00  +  \n",
        "MEN0018                                               -     1.00  79.95  +  \n",
        "MEN0018                                               -     1.00  75.00  +  \n",
        "MEN0018                                               -     1.00  75.00  +  \n",
        "MEN0018                                               -     1.00  73.00  +  \n",
        "MEN0018  Protein fusion: out of frame (TSPAN2-C12orf35)     1.00  73.00  +  \n",
        "MEN0018                                               -     1.00  71.93  +  \n",
        "MEN0018                                               -     1.00  70.00  +  \n",
        "MEN0018                                               -     1.00  68.00  +  \n",
        "MEN0018                                               -     0.99  64.30  +  \n",
        "MEN0018                                               -     0.97  63.87  +  \n",
        "MEN0018                                               -     1.00  59.00  +  \n",
        "MEN0018                                               -     1.00  51.88  +  \n",
        "MEN0018                      Deletion within transcript     1.00  51.00  +  \n",
        "MEN0018                                               -     1.00  49.00  +  \n",
        "MEN0018                                               -     0.75  23.31  +  \n",
        "MEN0018                                               -     0.98  18.71  +  \n",
        "MEN0018                                               -     1.00  18.00  +  \n",
        "MEN0018                                               -     1.00  17.00  +  \n",
        "MEN0018                                               -     1.00  16.00  +  \n",
        "MEN0018                                               -     0.97  10.66  +  \n",
        "MEN0020                                               -     1.00  67.00  +  \n",
        "MEN0020                                               -     1.00  15.00  +  \n",
        "MEN0021                                               -     1.00  71.00  +  \n",
        "MEN0021                                               -     1.00  70.00  +  \n",
        "MEN0021            Protein fusion: in frame (LMNA-FOSB)     0.82  42.60  +  \n",
        "MEN0022                                               -     1.00  46.00  +  \n",
        "MEN0041         Protein fusion: mid-exon (FCRL4-IQGAP3)     1.00  17.00  +  \n",
        "MEN0048                                Antisense fusion     1.00  14.00  +  \n",
        "\n",
        "[95 rows x 13 columns]"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# This is ranked, do not do possibly\n",
      "mutdata = pd.read_csv(PATH + 'ng.2526-S4.csv', index_col=0)\n",
      "mutdata"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Gene</th>\n",
        "      <th>Description</th>\n",
        "      <th>N</th>\n",
        "      <th>non-synonymous mutations</th>\n",
        "      <th>number of patients</th>\n",
        "      <th>unique sites</th>\n",
        "      <th>silent</th>\n",
        "      <th>*CpG-&gt;T</th>\n",
        "      <th>*Cp(A/C/T) -&gt;T</th>\n",
        "      <th>A-&gt;G</th>\n",
        "      <th>Trans-version</th>\n",
        "      <th>Indel / null</th>\n",
        "      <th>p</th>\n",
        "      <th>q</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Rank</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>1  </th>\n",
        "      <td>      NF2</td>\n",
        "      <td>                          neurofibromin 2 (merlin)</td>\n",
        "      <td>  102744</td>\n",
        "      <td> 27</td>\n",
        "      <td> 27</td>\n",
        "      <td> 24</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 27</td>\n",
        "      <td> &lt;1.00e-15</td>\n",
        "      <td> &lt;1.21e-11</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2  </th>\n",
        "      <td>     AKT1</td>\n",
        "      <td>     v-akt murine thymoma viral oncogene homolog 1</td>\n",
        "      <td>   74302</td>\n",
        "      <td>  5</td>\n",
        "      <td>  5</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 5</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>  1.58E-09</td>\n",
        "      <td>  9.60E-06</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3  </th>\n",
        "      <td>   CD300C</td>\n",
        "      <td>                            CD300c molecule (CLM6)</td>\n",
        "      <td>   11619</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  2</td>\n",
        "      <td>  9.63E-08</td>\n",
        "      <td>  3.90E-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4  </th>\n",
        "      <td>    RGPD3</td>\n",
        "      <td>          RANBP2-like and GRIP domain containing 3</td>\n",
        "      <td>  206520</td>\n",
        "      <td>  7</td>\n",
        "      <td>  4</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 3</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 4</td>\n",
        "      <td>  0</td>\n",
        "      <td>  1.53E-06</td>\n",
        "      <td>    0.0046</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5  </th>\n",
        "      <td>    KDM5C</td>\n",
        "      <td>  Lysine-specific histone demethylase 5C (JARID1C)</td>\n",
        "      <td>  283219</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 2</td>\n",
        "      <td>  1</td>\n",
        "      <td>  3.40E-05</td>\n",
        "      <td>     0.082</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6  </th>\n",
        "      <td>      SMO</td>\n",
        "      <td>                                smoothened homolog</td>\n",
        "      <td>  109996</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 2</td>\n",
        "      <td>  0</td>\n",
        "      <td>  4.50E-05</td>\n",
        "      <td>     0.092</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7  </th>\n",
        "      <td>   FANCD2</td>\n",
        "      <td>          Fanconi anemia, complementation group D2</td>\n",
        "      <td>  293230</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 1</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>  1.20E-04</td>\n",
        "      <td>      0.21</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8  </th>\n",
        "      <td>  ALOX5AP</td>\n",
        "      <td>    arachidonate 5-lipoxygenase-activating protein</td>\n",
        "      <td>    8595</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>  3.30E-04</td>\n",
        "      <td>      0.46</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9  </th>\n",
        "      <td>     IL32</td>\n",
        "      <td>                                    interleukin 32</td>\n",
        "      <td>    9996</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>  3.40E-04</td>\n",
        "      <td>      0.46</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10 </th>\n",
        "      <td>   ANP32C</td>\n",
        "      <td>  acidic (leucine-rich) nuclear phosphoprotein 32C</td>\n",
        "      <td>   11975</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>  4.70E-04</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11 </th>\n",
        "      <td> TMEM176B</td>\n",
        "      <td>                        transmembrane protein 176B</td>\n",
        "      <td>   14141</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>  5.00E-04</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12 </th>\n",
        "      <td>   CD300A</td>\n",
        "      <td>                            CD300a molecule (CLM8)</td>\n",
        "      <td>   15492</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>   0.00054</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>13 </th>\n",
        "      <td> KIAA1191</td>\n",
        "      <td>                                          KIAA1191</td>\n",
        "      <td>   16003</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>   0.00056</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>14 </th>\n",
        "      <td>     SIP1</td>\n",
        "      <td> survival of motor neuron protein interacting p...</td>\n",
        "      <td>   14783</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>   0.00061</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15 </th>\n",
        "      <td>   ZDHHC7</td>\n",
        "      <td>               zinc finger, DHHC-type containing 7</td>\n",
        "      <td>   17722</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>   0.00066</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16 </th>\n",
        "      <td>   JARID2</td>\n",
        "      <td>             jumonji, AT rich interactive domain 2</td>\n",
        "      <td>   62335</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>   0.00066</td>\n",
        "      <td>       0.5</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17 </th>\n",
        "      <td>      TBP</td>\n",
        "      <td>                          TATA box binding protein</td>\n",
        "      <td>   17195</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>   0.00093</td>\n",
        "      <td>      0.66</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18 </th>\n",
        "      <td>    LMOD1</td>\n",
        "      <td>                                       leiomodin 1</td>\n",
        "      <td>   30678</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0011</td>\n",
        "      <td>      0.71</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19 </th>\n",
        "      <td>   TAS2R7</td>\n",
        "      <td>                  taste receptor, type 2, member 7</td>\n",
        "      <td>   16293</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0011</td>\n",
        "      <td>      0.71</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>20 </th>\n",
        "      <td>     FSCB</td>\n",
        "      <td>              fibrous sheath CABYR binding protein</td>\n",
        "      <td>  159030</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>    0.0012</td>\n",
        "      <td>      0.71</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>21 </th>\n",
        "      <td>   CDKN2C</td>\n",
        "      <td>              cyclin-dependent kinase inhibitor 2C</td>\n",
        "      <td>   33435</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0013</td>\n",
        "      <td>      0.74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22 </th>\n",
        "      <td>  TBC1D29</td>\n",
        "      <td>                     TBC1 domain family, member 29</td>\n",
        "      <td>   19534</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0013</td>\n",
        "      <td>      0.74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23 </th>\n",
        "      <td>    PHF15</td>\n",
        "      <td>                             PHD finger protein 15</td>\n",
        "      <td>   39862</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0014</td>\n",
        "      <td>      0.74</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>24 </th>\n",
        "      <td>   ZNF660</td>\n",
        "      <td>                           zinc finger protein 660</td>\n",
        "      <td>   17000</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>    0.0015</td>\n",
        "      <td>      0.75</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25 </th>\n",
        "      <td> HLA-DRB1</td>\n",
        "      <td> major histocompatibility complex, class II, DR...</td>\n",
        "      <td>   41083</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0018</td>\n",
        "      <td>      0.87</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>26 </th>\n",
        "      <td>    TAOK3</td>\n",
        "      <td>                                      TAO kinase 3</td>\n",
        "      <td>   46669</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0021</td>\n",
        "      <td>      0.93</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>27 </th>\n",
        "      <td>    MERTK</td>\n",
        "      <td>              c-mer proto-oncogene tyrosine kinase</td>\n",
        "      <td>   51713</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0023</td>\n",
        "      <td>      0.93</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>28 </th>\n",
        "      <td>    ATG13</td>\n",
        "      <td>                              autophagy-related 13</td>\n",
        "      <td>   27384</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>    0.0023</td>\n",
        "      <td>      0.93</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>29 </th>\n",
        "      <td>   ENTHD1</td>\n",
        "      <td>                          ENTH domain containing 1</td>\n",
        "      <td>   31269</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>    0.0024</td>\n",
        "      <td>      0.93</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>30 </th>\n",
        "      <td>     CST8</td>\n",
        "      <td>                                        cystatin 8</td>\n",
        "      <td>    7497</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>    0.0024</td>\n",
        "      <td>      0.93</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>162</th>\n",
        "      <td>     LRP2</td>\n",
        "      <td>         low density lipoprotein-related protein 2</td>\n",
        "      <td>  916954</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>     0.046</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>163</th>\n",
        "      <td>   DICER1</td>\n",
        "      <td>                    dicer 1, ribonuclease type III</td>\n",
        "      <td>  381462</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>     0.047</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>164</th>\n",
        "      <td>  NCKIPSD</td>\n",
        "      <td>           NCK interacting protein with SH3 domain</td>\n",
        "      <td>  115119</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.049</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>165</th>\n",
        "      <td>  C2orf16</td>\n",
        "      <td>                chromosome 2 open reading frame 16</td>\n",
        "      <td>  101237</td>\n",
        "      <td>  2</td>\n",
        "      <td>  1</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.05</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>166</th>\n",
        "      <td>    ERBB3</td>\n",
        "      <td>  erythroblastic leukemia viral oncogene homolog 3</td>\n",
        "      <td>  272037</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.06</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>167</th>\n",
        "      <td>    PCSK7</td>\n",
        "      <td>     proprotein convertase subtilisin/kexin type 7</td>\n",
        "      <td>  130743</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.061</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>168</th>\n",
        "      <td>    EP300</td>\n",
        "      <td>                          E1A binding protein p300</td>\n",
        "      <td>  472127</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.061</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>169</th>\n",
        "      <td>    ZBED4</td>\n",
        "      <td>                zinc finger, BED-type containing 4</td>\n",
        "      <td>  224226</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.061</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>170</th>\n",
        "      <td>   NOTCH2</td>\n",
        "      <td>                                   Notch homolog 2</td>\n",
        "      <td>  484771</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.064</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>171</th>\n",
        "      <td>    MYH13</td>\n",
        "      <td>                            myosin, heavy chain 13</td>\n",
        "      <td>  195024</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.066</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>172</th>\n",
        "      <td>      APC</td>\n",
        "      <td>                        adenomatous polyposis coli</td>\n",
        "      <td>  557766</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  1</td>\n",
        "      <td>     0.067</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>173</th>\n",
        "      <td>    NTRK1</td>\n",
        "      <td>    neurotrophic tyrosine kinase, receptor, type 1</td>\n",
        "      <td>  124970</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.075</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>174</th>\n",
        "      <td>    ACSL6</td>\n",
        "      <td>    acyl-CoA synthetase long-chain family member 6</td>\n",
        "      <td>  143693</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 1</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.075</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>175</th>\n",
        "      <td>     TET1</td>\n",
        "      <td>                                    tet oncogene 1</td>\n",
        "      <td>  418705</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.079</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>176</th>\n",
        "      <td>       AR</td>\n",
        "      <td>                                 androgen receptor</td>\n",
        "      <td>  150931</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.085</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>177</th>\n",
        "      <td>    SYNE1</td>\n",
        "      <td>    spectrin repeat containing, nuclear envelope 1</td>\n",
        "      <td> 1753496</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 2</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.086</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>178</th>\n",
        "      <td>    EPHA5</td>\n",
        "      <td>                                   EPH receptor A5</td>\n",
        "      <td>  196037</td>\n",
        "      <td>  2</td>\n",
        "      <td>  1</td>\n",
        "      <td>  2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 2</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>     0.096</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>179</th>\n",
        "      <td>    NLRP2</td>\n",
        "      <td>             NLR family, pyrin domain containing 2</td>\n",
        "      <td>  204161</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.11</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>180</th>\n",
        "      <td>   ATP8B1</td>\n",
        "      <td>                ATPase, class I, type 8B, member 1</td>\n",
        "      <td>  234775</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.12</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>181</th>\n",
        "      <td>    LRP1B</td>\n",
        "      <td>        low density lipoprotein-related protein 1B</td>\n",
        "      <td>  913631</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.13</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>182</th>\n",
        "      <td>    MAGI2</td>\n",
        "      <td>            membrane associated guanylate kinase 2</td>\n",
        "      <td>  257957</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.13</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>183</th>\n",
        "      <td>     MTOR</td>\n",
        "      <td>      mechanistic target of rapamycin (S/T kinase)</td>\n",
        "      <td>  500566</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.13</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>184</th>\n",
        "      <td>     TET3</td>\n",
        "      <td>                      tet oncogene family member 3</td>\n",
        "      <td>  284531</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.15</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>185</th>\n",
        "      <td>    MAGI1</td>\n",
        "      <td>            membrane associated guanylate kinase 1</td>\n",
        "      <td>  292771</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.16</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>186</th>\n",
        "      <td>     MYH9</td>\n",
        "      <td>                             myosin, heavy chain 9</td>\n",
        "      <td>  350150</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.17</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>187</th>\n",
        "      <td>     MYH1</td>\n",
        "      <td>                             myosin, heavy chain 1</td>\n",
        "      <td>  387446</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.18</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>188</th>\n",
        "      <td> KIAA1549</td>\n",
        "      <td>                                          KIAA1549</td>\n",
        "      <td>  343097</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.19</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>189</th>\n",
        "      <td>     MLL3</td>\n",
        "      <td>      myeloid/lymphoid or mixed-lineage leukemia 3</td>\n",
        "      <td>  963663</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.19</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>190</th>\n",
        "      <td>     ROS1</td>\n",
        "      <td>       c-ros oncogene 1 , receptor tyrosine kinase</td>\n",
        "      <td>  468296</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td>  0</td>\n",
        "      <td>       0.2</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>191</th>\n",
        "      <td>      TTN</td>\n",
        "      <td>                                             titin</td>\n",
        "      <td> 1884193</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td>  1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>      0.31</td>\n",
        "      <td>         1</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>191 rows \u00d7 14 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 5,
       "text": [
        "          Gene                                        Description        N  \\\n",
        "Rank                                                                         \n",
        "1          NF2                           neurofibromin 2 (merlin)   102744   \n",
        "2         AKT1      v-akt murine thymoma viral oncogene homolog 1    74302   \n",
        "3       CD300C                             CD300c molecule (CLM6)    11619   \n",
        "4        RGPD3           RANBP2-like and GRIP domain containing 3   206520   \n",
        "5        KDM5C   Lysine-specific histone demethylase 5C (JARID1C)   283219   \n",
        "6          SMO                                 smoothened homolog   109996   \n",
        "7       FANCD2           Fanconi anemia, complementation group D2   293230   \n",
        "8      ALOX5AP     arachidonate 5-lipoxygenase-activating protein     8595   \n",
        "9         IL32                                     interleukin 32     9996   \n",
        "10      ANP32C   acidic (leucine-rich) nuclear phosphoprotein 32C    11975   \n",
        "11    TMEM176B                         transmembrane protein 176B    14141   \n",
        "12      CD300A                             CD300a molecule (CLM8)    15492   \n",
        "13    KIAA1191                                           KIAA1191    16003   \n",
        "14        SIP1  survival of motor neuron protein interacting p...    14783   \n",
        "15      ZDHHC7                zinc finger, DHHC-type containing 7    17722   \n",
        "16      JARID2              jumonji, AT rich interactive domain 2    62335   \n",
        "17         TBP                           TATA box binding protein    17195   \n",
        "18       LMOD1                                        leiomodin 1    30678   \n",
        "19      TAS2R7                   taste receptor, type 2, member 7    16293   \n",
        "20        FSCB               fibrous sheath CABYR binding protein   159030   \n",
        "21      CDKN2C               cyclin-dependent kinase inhibitor 2C    33435   \n",
        "22     TBC1D29                      TBC1 domain family, member 29    19534   \n",
        "23       PHF15                              PHD finger protein 15    39862   \n",
        "24      ZNF660                            zinc finger protein 660    17000   \n",
        "25    HLA-DRB1  major histocompatibility complex, class II, DR...    41083   \n",
        "26       TAOK3                                       TAO kinase 3    46669   \n",
        "27       MERTK               c-mer proto-oncogene tyrosine kinase    51713   \n",
        "28       ATG13                               autophagy-related 13    27384   \n",
        "29      ENTHD1                           ENTH domain containing 1    31269   \n",
        "30        CST8                                         cystatin 8     7497   \n",
        "...        ...                                                ...      ...   \n",
        "162       LRP2          low density lipoprotein-related protein 2   916954   \n",
        "163     DICER1                     dicer 1, ribonuclease type III   381462   \n",
        "164    NCKIPSD            NCK interacting protein with SH3 domain   115119   \n",
        "165    C2orf16                 chromosome 2 open reading frame 16   101237   \n",
        "166      ERBB3   erythroblastic leukemia viral oncogene homolog 3   272037   \n",
        "167      PCSK7      proprotein convertase subtilisin/kexin type 7   130743   \n",
        "168      EP300                           E1A binding protein p300   472127   \n",
        "169      ZBED4                 zinc finger, BED-type containing 4   224226   \n",
        "170     NOTCH2                                    Notch homolog 2   484771   \n",
        "171      MYH13                             myosin, heavy chain 13   195024   \n",
        "172        APC                         adenomatous polyposis coli   557766   \n",
        "173      NTRK1     neurotrophic tyrosine kinase, receptor, type 1   124970   \n",
        "174      ACSL6     acyl-CoA synthetase long-chain family member 6   143693   \n",
        "175       TET1                                     tet oncogene 1   418705   \n",
        "176         AR                                  androgen receptor   150931   \n",
        "177      SYNE1     spectrin repeat containing, nuclear envelope 1  1753496   \n",
        "178      EPHA5                                    EPH receptor A5   196037   \n",
        "179      NLRP2              NLR family, pyrin domain containing 2   204161   \n",
        "180     ATP8B1                 ATPase, class I, type 8B, member 1   234775   \n",
        "181      LRP1B         low density lipoprotein-related protein 1B   913631   \n",
        "182      MAGI2             membrane associated guanylate kinase 2   257957   \n",
        "183       MTOR       mechanistic target of rapamycin (S/T kinase)   500566   \n",
        "184       TET3                       tet oncogene family member 3   284531   \n",
        "185      MAGI1             membrane associated guanylate kinase 1   292771   \n",
        "186       MYH9                              myosin, heavy chain 9   350150   \n",
        "187       MYH1                              myosin, heavy chain 1   387446   \n",
        "188   KIAA1549                                           KIAA1549   343097   \n",
        "189       MLL3       myeloid/lymphoid or mixed-lineage leukemia 3   963663   \n",
        "190       ROS1        c-ros oncogene 1 , receptor tyrosine kinase   468296   \n",
        "191        TTN                                              titin  1884193   \n",
        "\n",
        "      non-synonymous mutations  number of patients  unique sites  silent  \\\n",
        "Rank                                                                       \n",
        "1                           27                  27            24       1   \n",
        "2                            5                   5             1       0   \n",
        "3                            2                   2             1       0   \n",
        "4                            7                   4             2       0   \n",
        "5                            3                   3             3       0   \n",
        "6                            3                   3             2       0   \n",
        "7                            3                   3             3       0   \n",
        "8                            1                   1             1       0   \n",
        "9                            1                   1             1       0   \n",
        "10                           1                   1             1       0   \n",
        "11                           1                   1             1       0   \n",
        "12                           1                   1             1       0   \n",
        "13                           1                   1             1       0   \n",
        "14                           1                   1             1       0   \n",
        "15                           1                   1             1       0   \n",
        "16                           2                   2             2       0   \n",
        "17                           1                   1             1       0   \n",
        "18                           1                   1             1       0   \n",
        "19                           1                   1             1       0   \n",
        "20                           2                   2             2       0   \n",
        "21                           1                   1             1       0   \n",
        "22                           1                   1             1       0   \n",
        "23                           1                   1             1       0   \n",
        "24                           1                   1             1       0   \n",
        "25                           1                   1             1       0   \n",
        "26                           1                   1             1       0   \n",
        "27                           1                   1             1       0   \n",
        "28                           1                   1             1       0   \n",
        "29                           1                   1             1       0   \n",
        "30                           1                   1             1       0   \n",
        "...                        ...                 ...           ...     ...   \n",
        "162                          2                   2             2       0   \n",
        "163                          1                   1             1       0   \n",
        "164                          1                   1             1       0   \n",
        "165                          2                   1             2       0   \n",
        "166                          1                   1             1       0   \n",
        "167                          1                   1             1       0   \n",
        "168                          1                   1             1       0   \n",
        "169                          1                   1             1       0   \n",
        "170                          1                   1             1       1   \n",
        "171                          1                   1             1       0   \n",
        "172                          1                   1             1       0   \n",
        "173                          1                   1             1       0   \n",
        "174                          1                   1             1       1   \n",
        "175                          1                   1             1       0   \n",
        "176                          1                   1             1       0   \n",
        "177                          2                   2             2       0   \n",
        "178                          2                   1             2       0   \n",
        "179                          1                   1             1       0   \n",
        "180                          1                   1             1       0   \n",
        "181                          1                   1             1       0   \n",
        "182                          1                   1             1       0   \n",
        "183                          1                   1             1       0   \n",
        "184                          1                   1             1       0   \n",
        "185                          1                   1             1       0   \n",
        "186                          1                   1             1       0   \n",
        "187                          1                   1             1       0   \n",
        "188                          1                   1             1       0   \n",
        "189                          1                   1             1       0   \n",
        "190                          1                   1             1       0   \n",
        "191                          1                   1             1       0   \n",
        "\n",
        "      *CpG->T  *Cp(A/C/T) ->T  A->G  Trans-version  Indel / null          p  \\\n",
        "Rank                                                                          \n",
        "1           0               0     0              0            27  <1.00e-15   \n",
        "2           0               5     0              0             0   1.58E-09   \n",
        "3           0               0     0              0             2   9.63E-08   \n",
        "4           3               0     0              4             0   1.53E-06   \n",
        "5           0               0     0              2             1   3.40E-05   \n",
        "6           0               1     0              2             0   4.50E-05   \n",
        "7           0               1     1              1             0   1.20E-04   \n",
        "8           0               0     0              0             1   3.30E-04   \n",
        "9           0               0     0              0             1   3.40E-04   \n",
        "10          0               0     0              0             1   4.70E-04   \n",
        "11          0               0     0              0             1   5.00E-04   \n",
        "12          0               0     0              0             1    0.00054   \n",
        "13          0               0     0              0             1    0.00056   \n",
        "14          0               0     0              0             1    0.00061   \n",
        "15          0               0     0              0             1    0.00066   \n",
        "16          1               0     0              1             0    0.00066   \n",
        "17          0               0     0              0             1    0.00093   \n",
        "18          0               0     0              0             1     0.0011   \n",
        "19          0               0     0              0             1     0.0011   \n",
        "20          0               1     0              1             0     0.0012   \n",
        "21          0               0     0              0             1     0.0013   \n",
        "22          0               0     0              0             1     0.0013   \n",
        "23          0               0     0              0             1     0.0014   \n",
        "24          0               0     0              1             0     0.0015   \n",
        "25          0               0     0              0             1     0.0018   \n",
        "26          0               0     0              0             1     0.0021   \n",
        "27          0               0     0              0             1     0.0023   \n",
        "28          0               0     0              0             1     0.0023   \n",
        "29          0               0     0              1             0     0.0024   \n",
        "30          0               0     1              0             0     0.0024   \n",
        "...       ...             ...   ...            ...           ...        ...   \n",
        "162         1               0     0              0             1      0.046   \n",
        "163         0               0     0              0             1      0.047   \n",
        "164         0               1     0              0             0      0.049   \n",
        "165         1               0     0              1             0       0.05   \n",
        "166         0               0     0              1             0       0.06   \n",
        "167         0               0     1              0             0      0.061   \n",
        "168         0               0     0              1             0      0.061   \n",
        "169         0               0     0              1             0      0.061   \n",
        "170         0               0     0              1             0      0.064   \n",
        "171         1               0     0              0             0      0.066   \n",
        "172         0               0     0              0             1      0.067   \n",
        "173         1               0     0              0             0      0.075   \n",
        "174         1               0     0              0             0      0.075   \n",
        "175         0               0     1              0             0      0.079   \n",
        "176         0               1     0              0             0      0.085   \n",
        "177         0               0     0              2             0      0.086   \n",
        "178         2               0     0              0             0      0.096   \n",
        "179         0               1     0              0             0       0.11   \n",
        "180         0               1     0              0             0       0.12   \n",
        "181         0               0     0              1             0       0.13   \n",
        "182         1               0     0              0             0       0.13   \n",
        "183         0               0     0              1             0       0.13   \n",
        "184         0               1     0              0             0       0.15   \n",
        "185         0               1     0              0             0       0.16   \n",
        "186         0               0     1              0             0       0.17   \n",
        "187         0               1     0              0             0       0.18   \n",
        "188         1               0     0              0             0       0.19   \n",
        "189         0               0     1              0             0       0.19   \n",
        "190         1               0     0              0             0        0.2   \n",
        "191         0               0     0              1             0       0.31   \n",
        "\n",
        "              q  \n",
        "Rank             \n",
        "1     <1.21e-11  \n",
        "2      9.60E-06  \n",
        "3      3.90E-04  \n",
        "4        0.0046  \n",
        "5         0.082  \n",
        "6         0.092  \n",
        "7          0.21  \n",
        "8          0.46  \n",
        "9          0.46  \n",
        "10          0.5  \n",
        "11          0.5  \n",
        "12          0.5  \n",
        "13          0.5  \n",
        "14          0.5  \n",
        "15          0.5  \n",
        "16          0.5  \n",
        "17         0.66  \n",
        "18         0.71  \n",
        "19         0.71  \n",
        "20         0.71  \n",
        "21         0.74  \n",
        "22         0.74  \n",
        "23         0.74  \n",
        "24         0.75  \n",
        "25         0.87  \n",
        "26         0.93  \n",
        "27         0.93  \n",
        "28         0.93  \n",
        "29         0.93  \n",
        "30         0.93  \n",
        "...         ...  \n",
        "162           1  \n",
        "163           1  \n",
        "164           1  \n",
        "165           1  \n",
        "166           1  \n",
        "167           1  \n",
        "168           1  \n",
        "169           1  \n",
        "170           1  \n",
        "171           1  \n",
        "172           1  \n",
        "173           1  \n",
        "174           1  \n",
        "175           1  \n",
        "176           1  \n",
        "177           1  \n",
        "178           1  \n",
        "179           1  \n",
        "180           1  \n",
        "181           1  \n",
        "182           1  \n",
        "183           1  \n",
        "184           1  \n",
        "185           1  \n",
        "186           1  \n",
        "187           1  \n",
        "188           1  \n",
        "189           1  \n",
        "190           1  \n",
        "191           1  \n",
        "\n",
        "[191 rows x 14 columns]"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "patdata[patdata.Grade]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "KeyError",
       "evalue": "\"['I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I'\\n 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I'\\n 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'II' 'II' 'II' 'II' 'II' 'II'\\n 'II' 'II' 'II' 'II' 'II' 'II' 'II' 'II' 'II' 'III' 'III' 'III'] not in index\"",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-4-fb4734191ec8>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mpatdata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mpatdata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mGrade\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;32m/cellar/users/agross/anaconda2/lib/python2.7/site-packages/pandas-0.14.1.dev-py2.7-linux-x86_64.egg/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   1671\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mSeries\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlist\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1672\u001b[0m             \u001b[1;31m# either boolean or fancy integer index\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1673\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_array\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1674\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1675\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_frame\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/cellar/users/agross/anaconda2/lib/python2.7/site-packages/pandas-0.14.1.dev-py2.7-linux-x86_64.egg/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m_getitem_array\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   1715\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtake\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mconvert\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1716\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1717\u001b[1;33m             \u001b[0mindexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mix\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_convert_to_indexer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1718\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtake\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mconvert\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1719\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/cellar/users/agross/anaconda2/lib/python2.7/site-packages/pandas-0.14.1.dev-py2.7-linux-x86_64.egg/pandas/core/indexing.pyc\u001b[0m in \u001b[0;36m_convert_to_indexer\u001b[1;34m(self, obj, axis, is_setter)\u001b[0m\n\u001b[0;32m   1085\u001b[0m                     \u001b[1;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mis_setter\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1086\u001b[0m                         \u001b[1;32mreturn\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;34m'key'\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mobj\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1087\u001b[1;33m                     \u001b[1;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'%s not in index'\u001b[0m \u001b[1;33m%\u001b[0m \u001b[0mobjarr\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mmask\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1088\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1089\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mindexer\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mKeyError\u001b[0m: \"['I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I'\\n 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I'\\n 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'I' 'II' 'II' 'II' 'II' 'II' 'II'\\n 'II' 'II' 'II' 'II' 'II' 'II' 'II' 'II' 'II' 'III' 'III' 'III'] not in index\""
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "patdata[patdata['Histology']  == 'Transitional'].T"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th>Columns</th>\n",
        "      <th>Discovery</th>\n",
        "      <th>Discovery</th>\n",
        "      <th>Discovery</th>\n",
        "      <th>Discovery</th>\n",
        "      <th>Validation</th>\n",
        "      <th>Validation</th>\n",
        "      <th>Validation</th>\n",
        "      <th>Validation</th>\n",
        "      <th>Validation</th>\n",
        "      <th>Validation</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Sample ID</th>\n",
        "      <td>        0011-MEN</td>\n",
        "      <td>                            0017-MEN</td>\n",
        "      <td>                             0018-MEN</td>\n",
        "      <td>      0021-MEN</td>\n",
        "      <td>      0043-MEN</td>\n",
        "      <td>           0056-MEN</td>\n",
        "      <td>      0062-MEN</td>\n",
        "      <td>      0068-MEN</td>\n",
        "      <td>           0079-MEN</td>\n",
        "      <td>       0085-MEN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Grade</th>\n",
        "      <td>               I</td>\n",
        "      <td>                                   I</td>\n",
        "      <td>                                    I</td>\n",
        "      <td>             I</td>\n",
        "      <td>             I</td>\n",
        "      <td>                  I</td>\n",
        "      <td>             I</td>\n",
        "      <td>             I</td>\n",
        "      <td>                  I</td>\n",
        "      <td>              I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Histology</th>\n",
        "      <td>    Transitional</td>\n",
        "      <td>                        Transitional</td>\n",
        "      <td>                         Transitional</td>\n",
        "      <td>  Transitional</td>\n",
        "      <td>  Transitional</td>\n",
        "      <td>       Transitional</td>\n",
        "      <td>  Transitional</td>\n",
        "      <td>  Transitional</td>\n",
        "      <td>       Transitional</td>\n",
        "      <td>   Transitional</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Age</th>\n",
        "      <td>              55</td>\n",
        "      <td>                                  71</td>\n",
        "      <td>                                   69</td>\n",
        "      <td>            31</td>\n",
        "      <td>            45</td>\n",
        "      <td>                 40</td>\n",
        "      <td>            69</td>\n",
        "      <td>            71</td>\n",
        "      <td>                 48</td>\n",
        "      <td>             70</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Sex</th>\n",
        "      <td>               F</td>\n",
        "      <td>                                   F</td>\n",
        "      <td>                                    F</td>\n",
        "      <td>             F</td>\n",
        "      <td>             M</td>\n",
        "      <td>                  F</td>\n",
        "      <td>             F</td>\n",
        "      <td>             F</td>\n",
        "      <td>                  F</td>\n",
        "      <td>              M</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Location</th>\n",
        "      <td> Right occipital</td>\n",
        "      <td>                  Right parasagittal</td>\n",
        "      <td>                         Left Frontal</td>\n",
        "      <td> Right frontal</td>\n",
        "      <td> Right frontal</td>\n",
        "      <td> Right Petrous apex</td>\n",
        "      <td> Left temporal</td>\n",
        "      <td> Right Frontal</td>\n",
        "      <td> Planum sphenoidale</td>\n",
        "      <td>  Left Parietal</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Location Category</th>\n",
        "      <td>       Convexity</td>\n",
        "      <td>                      Falcotentorial</td>\n",
        "      <td>                            Convexity</td>\n",
        "      <td>     Convexity</td>\n",
        "      <td>     Convexity</td>\n",
        "      <td>         Skull Base</td>\n",
        "      <td>     Convexity</td>\n",
        "      <td>     Convexity</td>\n",
        "      <td>         Skull Base</td>\n",
        "      <td>      Convexity</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Recurrent</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                                 NaN</td>\n",
        "      <td>                                  NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Rad. treated</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                                 NaN</td>\n",
        "      <td>                                  NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Rad. induced</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                                 NaN</td>\n",
        "      <td>                                  NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>             X</td>\n",
        "      <td>                  X</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Germline NF2</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                                 NaN</td>\n",
        "      <td>                                  NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NF2</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                            p.K227fs</td>\n",
        "      <td>                              p.Y144*</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>      p.S444fs</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>         p.R57*</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Chr22</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                                loss</td>\n",
        "      <td>                                 loss</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>          loss</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>          loss</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AKT1</th>\n",
        "      <td>          p.E17K</td>\n",
        "      <td>                                 NaN</td>\n",
        "      <td>                                  NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>        p.E17K</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SMO</th>\n",
        "      <td>             NaN</td>\n",
        "      <td>                                 NaN</td>\n",
        "      <td>                                  NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>            NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Other events</th>\n",
        "      <td>             NaN</td>\n",
        "      <td> chromothripsis, NEGR1 translocation</td>\n",
        "      <td> JARID2 mutation, NEGR1 translocation</td>\n",
        "      <td>       6q loss</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>           NaN</td>\n",
        "      <td>                NaN</td>\n",
        "      <td> KDM5C mutation</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "Columns                  Discovery                            Discovery  \\\n",
        "Sample ID                 0011-MEN                             0017-MEN   \n",
        "Grade                            I                                    I   \n",
        "Histology             Transitional                         Transitional   \n",
        "Age                             55                                   71   \n",
        "Sex                              F                                    F   \n",
        "Location           Right occipital                   Right parasagittal   \n",
        "Location Category        Convexity                       Falcotentorial   \n",
        "Recurrent                      NaN                                  NaN   \n",
        "Rad. treated                   NaN                                  NaN   \n",
        "Rad. induced                   NaN                                  NaN   \n",
        "Germline NF2                   NaN                                  NaN   \n",
        "NF2                            NaN                             p.K227fs   \n",
        "Chr22                          NaN                                 loss   \n",
        "AKT1                        p.E17K                                  NaN   \n",
        "SMO                            NaN                                  NaN   \n",
        "Other events                   NaN  chromothripsis, NEGR1 translocation   \n",
        "\n",
        "Columns                                       Discovery      Discovery  \\\n",
        "Sample ID                                      0018-MEN       0021-MEN   \n",
        "Grade                                                 I              I   \n",
        "Histology                                  Transitional   Transitional   \n",
        "Age                                                  69             31   \n",
        "Sex                                                   F              F   \n",
        "Location                                   Left Frontal  Right frontal   \n",
        "Location Category                             Convexity      Convexity   \n",
        "Recurrent                                           NaN            NaN   \n",
        "Rad. treated                                        NaN            NaN   \n",
        "Rad. induced                                        NaN            NaN   \n",
        "Germline NF2                                        NaN            NaN   \n",
        "NF2                                             p.Y144*            NaN   \n",
        "Chr22                                              loss            NaN   \n",
        "AKT1                                                NaN            NaN   \n",
        "SMO                                                 NaN            NaN   \n",
        "Other events       JARID2 mutation, NEGR1 translocation        6q loss   \n",
        "\n",
        "Columns               Validation          Validation     Validation  \\\n",
        "Sample ID               0043-MEN            0056-MEN       0062-MEN   \n",
        "Grade                          I                   I              I   \n",
        "Histology           Transitional        Transitional   Transitional   \n",
        "Age                           45                  40             69   \n",
        "Sex                            M                   F              F   \n",
        "Location           Right frontal  Right Petrous apex  Left temporal   \n",
        "Location Category      Convexity          Skull Base      Convexity   \n",
        "Recurrent                    NaN                 NaN            NaN   \n",
        "Rad. treated                 NaN                 NaN            NaN   \n",
        "Rad. induced                   X                   X            NaN   \n",
        "Germline NF2                 NaN                 NaN            NaN   \n",
        "NF2                          NaN                 NaN       p.S444fs   \n",
        "Chr22                       loss                 NaN           loss   \n",
        "AKT1                         NaN                 NaN            NaN   \n",
        "SMO                          NaN                 NaN            NaN   \n",
        "Other events                 NaN                 NaN            NaN   \n",
        "\n",
        "Columns               Validation          Validation      Validation  \n",
        "Sample ID               0068-MEN            0079-MEN        0085-MEN  \n",
        "Grade                          I                   I               I  \n",
        "Histology           Transitional        Transitional    Transitional  \n",
        "Age                           71                  48              70  \n",
        "Sex                            F                   F               M  \n",
        "Location           Right Frontal  Planum sphenoidale   Left Parietal  \n",
        "Location Category      Convexity          Skull Base       Convexity  \n",
        "Recurrent                    NaN                 NaN             NaN  \n",
        "Rad. treated                 NaN                 NaN             NaN  \n",
        "Rad. induced                 NaN                 NaN             NaN  \n",
        "Germline NF2                 NaN                 NaN             NaN  \n",
        "NF2                          NaN                 NaN          p.R57*  \n",
        "Chr22                        NaN                 NaN             NaN  \n",
        "AKT1                      p.E17K                 NaN             NaN  \n",
        "SMO                          NaN                 NaN             NaN  \n",
        "Other events                 NaN                 NaN  KDM5C mutation  "
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "y = patdata[['Age','Grade']]\n",
      "y\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Age</th>\n",
        "      <th>Grade</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Columns</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 65</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 29</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 55</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 41</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 47</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 71</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 69</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 51</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 48</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 31</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 61</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 61</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 54</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 35</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 61</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 46</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Discovery</th>\n",
        "      <td> 28</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 45</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 67</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 62</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 49</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 40</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 60</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 47</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 57</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 69</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 42</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 59</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 77</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 45</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 86</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 28</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 56</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 54</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 64</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 79</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 48</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 52</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 63</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 63</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 56</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Validation</th>\n",
        "      <td> 70</td>\n",
        "      <td>   I</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 47</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 45</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 61</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 67</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 79</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 57</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 84</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 65</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 75</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 40</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 42</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 85</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 60</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 66</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 40</td>\n",
        "      <td>  II</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 69</td>\n",
        "      <td> III</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 46</td>\n",
        "      <td> III</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Extrapolation</th>\n",
        "      <td> 58</td>\n",
        "      <td> III</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>65 rows \u00d7 2 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 5,
       "text": [
        "               Age Grade\n",
        "Columns                 \n",
        "Discovery       65     I\n",
        "Discovery       29     I\n",
        "Discovery       55     I\n",
        "Discovery       41     I\n",
        "Discovery       47     I\n",
        "Discovery       71     I\n",
        "Discovery       69     I\n",
        "Discovery       51     I\n",
        "Discovery       48     I\n",
        "Discovery       31     I\n",
        "Discovery       61     I\n",
        "Discovery       61     I\n",
        "Discovery       54     I\n",
        "Discovery       35     I\n",
        "Discovery       61     I\n",
        "Discovery       46     I\n",
        "Discovery       28     I\n",
        "Validation      45     I\n",
        "Validation      67     I\n",
        "Validation      62     I\n",
        "Validation      49     I\n",
        "Validation      40     I\n",
        "Validation      60     I\n",
        "Validation      47     I\n",
        "Validation      57     I\n",
        "Validation      69     I\n",
        "Validation      42     I\n",
        "Validation      59     I\n",
        "Validation      77     I\n",
        "Validation      45     I\n",
        "...            ...   ...\n",
        "Validation      86     I\n",
        "Validation      28     I\n",
        "Validation      56     I\n",
        "Validation      54     I\n",
        "Validation      64     I\n",
        "Validation      79     I\n",
        "Validation      48     I\n",
        "Validation      52     I\n",
        "Validation      63     I\n",
        "Validation      63     I\n",
        "Validation      56     I\n",
        "Validation      70     I\n",
        "Extrapolation   47    II\n",
        "Extrapolation   45    II\n",
        "Extrapolation   61    II\n",
        "Extrapolation   67    II\n",
        "Extrapolation   79    II\n",
        "Extrapolation   57    II\n",
        "Extrapolation   84    II\n",
        "Extrapolation   65    II\n",
        "Extrapolation   75    II\n",
        "Extrapolation   40    II\n",
        "Extrapolation   42    II\n",
        "Extrapolation   85    II\n",
        "Extrapolation   60    II\n",
        "Extrapolation   66    II\n",
        "Extrapolation   40    II\n",
        "Extrapolation   69   III\n",
        "Extrapolation   46   III\n",
        "Extrapolation   58   III\n",
        "\n",
        "[65 rows x 2 columns]"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "patdata[patdata.Grade == 'II'].max()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 190,
       "text": [
        "Age    85\n",
        "dtype: int64"
       ]
      }
     ],
     "prompt_number": 190
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "y= patdata.set_index(['Grade','Histology'])\n",
      "y.Location.describe()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 91,
       "text": [
        "count               65\n",
        "unique              42\n",
        "top       Left Frontal\n",
        "freq                 8\n",
        "dtype: object"
       ]
      }
     ],
     "prompt_number": 91
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "\n",
      "b= mutdata.set_index('Gene')\n",
      "b.loc['NF2':'SMO']\n",
      "b['NF2':'SMO']\n",
      "b[b['unique sites'] > 2]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Description</th>\n",
        "      <th>N</th>\n",
        "      <th>non-synonymous mutations</th>\n",
        "      <th>number of patients</th>\n",
        "      <th>unique sites</th>\n",
        "      <th>silent</th>\n",
        "      <th>*CpG-&gt;T</th>\n",
        "      <th>*Cp(A/C/T) -&gt;T</th>\n",
        "      <th>A-&gt;G</th>\n",
        "      <th>Trans-version</th>\n",
        "      <th>Indel / null</th>\n",
        "      <th>p</th>\n",
        "      <th>q</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Gene</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>NF2</th>\n",
        "      <td>                         neurofibromin 2 (merlin)</td>\n",
        "      <td> 102744</td>\n",
        "      <td> 27</td>\n",
        "      <td> 27</td>\n",
        "      <td> 24</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 27</td>\n",
        "      <td> &lt;1.00e-15</td>\n",
        "      <td> &lt;1.21e-11</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>KDM5C</th>\n",
        "      <td> Lysine-specific histone demethylase 5C (JARID1C)</td>\n",
        "      <td> 283219</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 2</td>\n",
        "      <td>  1</td>\n",
        "      <td>  3.40E-05</td>\n",
        "      <td>     0.082</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FANCD2</th>\n",
        "      <td>         Fanconi anemia, complementation group D2</td>\n",
        "      <td> 293230</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td>  3</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1</td>\n",
        "      <td> 1</td>\n",
        "      <td> 1</td>\n",
        "      <td>  0</td>\n",
        "      <td>  1.20E-04</td>\n",
        "      <td>      0.21</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 72,
       "text": [
        "                                             Description       N  \\\n",
        "Gene                                                               \n",
        "NF2                             neurofibromin 2 (merlin)  102744   \n",
        "KDM5C   Lysine-specific histone demethylase 5C (JARID1C)  283219   \n",
        "FANCD2          Fanconi anemia, complementation group D2  293230   \n",
        "\n",
        "        non-synonymous mutations  number of patients  unique sites  silent  \\\n",
        "Gene                                                                         \n",
        "NF2                           27                  27            24       1   \n",
        "KDM5C                          3                   3             3       0   \n",
        "FANCD2                         3                   3             3       0   \n",
        "\n",
        "        *CpG->T  *Cp(A/C/T) ->T  A->G  Trans-version  Indel / null          p  \\\n",
        "Gene                                                                            \n",
        "NF2           0               0     0              0            27  <1.00e-15   \n",
        "KDM5C         0               0     0              2             1   3.40E-05   \n",
        "FANCD2        0               1     1              1             0   1.20E-04   \n",
        "\n",
        "                q  \n",
        "Gene               \n",
        "NF2     <1.21e-11  \n",
        "KDM5C       0.082  \n",
        "FANCD2       0.21  "
       ]
      }
     ],
     "prompt_number": 72
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "b.ix['NF2', 'N']\n",
      "`"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 55,
       "text": [
        "102744"
       ]
      }
     ],
     "prompt_number": 55
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "b.var()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 73,
       "text": [
        "N                           5.385434e+10\n",
        "non-synonymous mutations    3.872858e+00\n",
        "number of patients          3.731166e+00\n",
        "unique sites                2.844420e+00\n",
        "silent                      3.549187e-02\n",
        "*CpG->T                     1.991182e-01\n",
        "*Cp(A/C/T) ->T              2.852576e-01\n",
        "A->G                        8.151006e-02\n",
        "Trans-version               3.169468e-01\n",
        "Indel / null                3.964067e+00\n",
        "dtype: float64"
       ]
      }
     ],
     "prompt_number": 73
    }
   ],
   "metadata": {}
  }
 ]
}