{
 "metadata": {
  "name": "",
  "signature": "sha256:25ca1c964dd576343569d8c3fe18b84f24ee8516cebcb79b4ff2631d62bee5de"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%pylab inline\n",
      "import pandas as pd\n",
      "import seaborn as sns\n",
      "PATH = '/cellar/users/talejand/Documents/IdekerLab/CancerPapers/WGS and WES of bladder cancer/'\n",
      "MUT_PATH='/cellar/users/talejand/Documents/IdekerLab/CancerPapers/Pat-Mut_Data/'\n",
      "MERGED_PATH = '/cellar/users/talejand/Documents/IdekerLab/CancerPapers/Merged/'"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Populating the interactive namespace from numpy and matplotlib\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Whole-genome and whole-exome sequencing of bladder cancer identifies frequent alterations in genes involved in sister chromatid cohesion and segregation"
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "TCC (transitional cell carcinoma), most prominent of Bladder Cancers"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Bladder Cancer</b>: Heterogeneous Disease w/ 2 Subtypes\n",
      "\n",
      "1. Superficial\n",
      "\n",
      "2. Invasive"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>WGS/WES:</b> 99 patients"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Determinations</b>:\n",
      "\n",
      "* Confirmed recurring mutations and identified additional altered genes/pathways\n",
      "\n",
      "\n",
      "* <b>Frequent Alterations</b>: STAG2 and ESPL1\n",
      "\n",
      "     * genes involved in sister chromatic cohesion and segregation (SCCS)\n",
      "\n",
      "\n",
      "\n",
      "* <b>Recurrent Fusion:</b> FGFR3 and TACC3\n",
      "     * detected by transcriptome seq of 42 DNA seq tumors\n",
      "\n",
      "\n",
      "* 32% (32/99) of tumors had genetic alterations in the SCSSS process\n",
      "\n",
      "* Analysis provides evidence that these type of alterations may be involved in bladder tumorigenesis"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Somatic Mutations</b> (candidate): 11,240\n",
      "\n",
      "validation experiments on 1,119 predicted somatic substitutions and 91 indels confirmed 1,023 (91%) and 67 (74%), respectively, by Sanger sequencing\n",
      "\n",
      "* WGS detected CNAs and obtained fourfold mean haploid coverage for each sample\n",
      "\n",
      "* RNA-seq on a subset of 42 DNA-sequenced tumors, including 16 for which matched, morpho- logically normal bladder tissue was available"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Significantly Mutated Genes</b> <i>(SMGs)</i>: 37\n",
      "\n",
      "* TP53, HRAS, FGFR3, PIK3CA, RB1, KRAS, and TSCI\n",
      "* 8 chromatin Remodeling Genes: UTX, ARID1A, MLL-MLL3, CREBBP- EP300, NCOR1 and CHD6\n",
      "* 13 prev. unreported, most important STAG2 (11%?), then ERCC2, TRRAP and FAT4\n",
      "    * 9/11 mutations were truncating ( 3 small frameshift indels, 4 nonsense, 2 splice-site)\n",
      "    * STAG2 Genomic Deletions in 5/99 tumors\n",
      "        * By screening all exons of STAG2 in an additional 50 tumor-normal pairs form individuals with TCC by Sanger sequencing, we iden- tified 5 somatic mutations in 4 tumors \n",
      "        * We also examined the methylation status of the STAG2 promoter in 30 TCCs by bisulfite sequencing and found STAG2 promoter hyper- methylation in 7 tumors (23%) relative to matched normal samples"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Individuals w/ STAG2 somatic alt. have sig. worse prognosis</b>\n",
      "    * suggests STAG2 is an independent predictor of worse outcomes in TCC"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Frequent mutations in multiple tumors</b> <i>in other chromatin- remodeling genes</i>, including: \n",
      "* the histone demethylase genes KDM6A(UTX) and UTY (30%), the chromatin-remodeler genes ARID1A and ARID4A (17%), the histone lysine methyltransferase genes KMT2A (MLL), KMT2C (MLL3) and KMT2E (MLL5) (16%), the histone acetyltransferase genes EP300 and EP400 (15%), the SWI/SNF complex\u2013related genes SMARCA4 and SMARAC1 (7%) and the his- tone demethylase genes KDM5A (JARID1A) and KDM5B (JARID1B) (6%). \n",
      "\n",
      "* Total: 57 of the 99 cases (58%) harbored somaticmutations in chromatin-remodeling genes, further indicating that altered epigenetic regulation of chromatin and chromatin post- translational modifications may be a major driver mechanism in TCC3."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Evaluation of aneuploidy:</b>\n",
      "\n",
      "* quantifying the extent of copy number changes on chromosome arms and found that tumors with genetic alterations in STAG2 had more aneuploidy (P = 0.01) than tumors with wild- type SCCS genes (Supplementary Fig. 7), which is consistent with a published study10.\n",
      "\n",
      "* In contrast with the observation in solid tumors, many STAG2-mutated primary leukemias have completely normal karyotypes11,12. \n",
      "\n",
      "* These contradictory observations in different types of cancer suggest that further comprehensive investigation of STAG2"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Profiling of 99 tumors for CNAs useing WGS</b>\n",
      "\n",
      "* abnormalities:\n",
      "    1. Gain: 5p, 8q, 13p, 20p, 20q\n",
      "    2. Loss: 8p, 9p, 11p, 14p, 15p, 17p, and 21p "
     ]
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Other"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "applied GSITIC14 analysis to whole-genome sequencing data to identify recurrent focal CNAs, and this analysis yielded 84 regions of focal amplification (Supplementary Fig. 12 and Supplementary Table 9), several of which included CNAs previously detected in bladder cancer13, which encompassed genes such as TRIO, MDM2, MYC, E2F3, CCND1 and ERBB2 "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "amplifications defined regions of CNAs reported for the first time, to our knowledge, in bladder cancer, including CCNE1, CEBPA, E2F1 and MUC1"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "An interesting finding was frequent amplification of DHFR, encoding dihydrofolate reductase, which is a target of many anticancer agents15, at 5q (frequently loss in our cohort) in 14 tumors (14%) (Supplementary Figs. 16 and 17 and Supplementary Note).\n",
      "\n",
      "identified 80 recurrent focal deletion regions, including RB1 and CREBBP, which were also fre- quently truncated in TCCs. \n",
      "\n",
      "One of the most common focal deletions, detected in 50 tumors (50%), was a deleted region at 9p21 containing CDKN2A and CDKN2B (Supplementary Fig. 18) and is a widely reported genomic alteration in bladder cancer16."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Notable</b> \n",
      "\n",
      "* Highly frequent genomic alterations in genes involved in the SCCS process, including:\n",
      "    * alterations in STAG2, ESPL1 and NIPBL, and a recurrent FGFR3-TACC3 fusion in TCC.\n",
      "    * In contrast to other cancers in which only rare or low-frequency alterations in SCCS genes have been observed10,22,23, bladder can- cer is the first type of cancer, to our knowledge, with predominant genetic lesions in genes involved in the SCCS process. \n",
      "    \n",
      "* Although we do not yet understand the detailed mechanisms by which STAG2 may contribute to TCC pathogenesis, the discovery of highly recur- rent alterations in the genes involved in the SCCS process in TCC identifies yet another pathway that is undoubtedly relevant for TCC. Our analysis suggests the necessity of comprehensive investigation of SCCS genes to elucidate their exact roles in causing aneuploidy or bladder tumorigenesis."
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "<b>Mutations</b>\n",
      "\n",
      "    * Mutated Genes: 6,737\n",
      "    * Mutations in Genes: 11,240\n",
      "        * Synonymous in coding Sequences: 3,009\n",
      "        * Splice Site: 192\n",
      "        * Non-synonymous in coding sequences:\n",
      "            1. Missense: 6,839\n",
      "            2. Nonsense: 632\n",
      "            3. Insertions: 170\n",
      "            4. Deletions: 398"
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Hard Data"
     ]
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Patient Data:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "#Patient Data\n",
      "\n",
      "#read in csv \n",
      "patdata = pd.read_csv(PATH + 'clinicaldata.csv')\n",
      "#Drop unwanted columns\n",
      "patdata.drop(['Superficial/Invasive','RNA-Seq'], 1, inplace=True)\n",
      "#rename columns to format\n",
      "patdata.rename(columns={'TNM':'Tumor_ID','Primary/Relapsed': 'Status','Case_ID':'Patient_ID','Sex':'Gender'}, inplace=True)\n",
      "\n",
      "patdata['Patient_ID']=patdata['Patient_ID'].map(str.strip)\n",
      "#write to csv in PatMut folder\n",
      "patdata.to_csv(MUT_PATH+'patcharWGSWESbladder.csv')\n",
      "\n",
      "patdata\n",
      "#.query('Patient_ID == \"B80-1\"')\n",
      "            "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Patient_ID</th>\n",
        "      <th>Gender</th>\n",
        "      <th>Age</th>\n",
        "      <th>Tumor_ID</th>\n",
        "      <th>Grade</th>\n",
        "      <th>Status</th>\n",
        "      <th>WES</th>\n",
        "      <th>WGS</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0 </th>\n",
        "      <td>    B10</td>\n",
        "      <td>  M </td>\n",
        "      <td> 52</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1 </th>\n",
        "      <td>   B100</td>\n",
        "      <td>  M </td>\n",
        "      <td> 25</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2 </th>\n",
        "      <td>   B101</td>\n",
        "      <td>  F </td>\n",
        "      <td> 69</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3 </th>\n",
        "      <td>   B102</td>\n",
        "      <td>  M </td>\n",
        "      <td> 45</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4 </th>\n",
        "      <td>   B103</td>\n",
        "      <td>  M </td>\n",
        "      <td> 54</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5 </th>\n",
        "      <td>   B104</td>\n",
        "      <td>  M </td>\n",
        "      <td> 52</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6 </th>\n",
        "      <td> B104-0</td>\n",
        "      <td>  M </td>\n",
        "      <td> 72</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7 </th>\n",
        "      <td>   B105</td>\n",
        "      <td>  M </td>\n",
        "      <td> 62</td>\n",
        "      <td>  T2NxMx </td>\n",
        "      <td> 1</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8 </th>\n",
        "      <td> B105-0</td>\n",
        "      <td>  M </td>\n",
        "      <td> 60</td>\n",
        "      <td>  T2NxMx </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9 </th>\n",
        "      <td> B105-1</td>\n",
        "      <td>  M </td>\n",
        "      <td> 64</td>\n",
        "      <td>  T2NxMx </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10</th>\n",
        "      <td>   B106</td>\n",
        "      <td>  M </td>\n",
        "      <td> 56</td>\n",
        "      <td>  T2NxMx </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11</th>\n",
        "      <td>   B107</td>\n",
        "      <td>  M </td>\n",
        "      <td> 68</td>\n",
        "      <td>  T2NxMx </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12</th>\n",
        "      <td>   B109</td>\n",
        "      <td>  M </td>\n",
        "      <td> 46</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>13</th>\n",
        "      <td>    B11</td>\n",
        "      <td>  M </td>\n",
        "      <td> 80</td>\n",
        "      <td>  T4N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>14</th>\n",
        "      <td>   B110</td>\n",
        "      <td>  M </td>\n",
        "      <td> 54</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15</th>\n",
        "      <td>   B111</td>\n",
        "      <td>  M </td>\n",
        "      <td> 45</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16</th>\n",
        "      <td>   B112</td>\n",
        "      <td>  M </td>\n",
        "      <td> 62</td>\n",
        "      <td>  T3N1M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17</th>\n",
        "      <td>   B114</td>\n",
        "      <td>  M </td>\n",
        "      <td> 73</td>\n",
        "      <td> T1bN0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18</th>\n",
        "      <td>    B13</td>\n",
        "      <td>  M </td>\n",
        "      <td> 64</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19</th>\n",
        "      <td>    B14</td>\n",
        "      <td>  M </td>\n",
        "      <td> 57</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>20</th>\n",
        "      <td>    B15</td>\n",
        "      <td>  M </td>\n",
        "      <td> 40</td>\n",
        "      <td>  T4N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>21</th>\n",
        "      <td>    B16</td>\n",
        "      <td>  M </td>\n",
        "      <td> 72</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22</th>\n",
        "      <td>    B18</td>\n",
        "      <td>  M </td>\n",
        "      <td> 66</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23</th>\n",
        "      <td>    B19</td>\n",
        "      <td>  M </td>\n",
        "      <td> 69</td>\n",
        "      <td>  T4NXMX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>24</th>\n",
        "      <td>    B21</td>\n",
        "      <td>  M </td>\n",
        "      <td> 61</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25</th>\n",
        "      <td>    B22</td>\n",
        "      <td>  M </td>\n",
        "      <td> 79</td>\n",
        "      <td>  T4N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>26</th>\n",
        "      <td>    B23</td>\n",
        "      <td>  M </td>\n",
        "      <td> 85</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>27</th>\n",
        "      <td>    B24</td>\n",
        "      <td>  M </td>\n",
        "      <td> 53</td>\n",
        "      <td>  T4N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>28</th>\n",
        "      <td>    B25</td>\n",
        "      <td>  M </td>\n",
        "      <td> 65</td>\n",
        "      <td> T4aN0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>29</th>\n",
        "      <td>    B34</td>\n",
        "      <td>  M </td>\n",
        "      <td> 66</td>\n",
        "      <td> T3aN0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>69</th>\n",
        "      <td>  B80-3</td>\n",
        "      <td>  M </td>\n",
        "      <td> 51</td>\n",
        "      <td>  T4N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>70</th>\n",
        "      <td>  B80-4</td>\n",
        "      <td>  M </td>\n",
        "      <td> 54</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>71</th>\n",
        "      <td>  B80-5</td>\n",
        "      <td>  M </td>\n",
        "      <td> 73</td>\n",
        "      <td>  TaN0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>72</th>\n",
        "      <td>  B80-7</td>\n",
        "      <td>  M </td>\n",
        "      <td> 58</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>73</th>\n",
        "      <td>  B80-8</td>\n",
        "      <td>  F </td>\n",
        "      <td> 63</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>74</th>\n",
        "      <td>    B81</td>\n",
        "      <td>  M </td>\n",
        "      <td> 63</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>75</th>\n",
        "      <td>  B81-1</td>\n",
        "      <td>  M </td>\n",
        "      <td> 62</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>76</th>\n",
        "      <td>  B81-2</td>\n",
        "      <td>  M </td>\n",
        "      <td> 61</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>77</th>\n",
        "      <td>    B82</td>\n",
        "      <td>  M </td>\n",
        "      <td> 56</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>78</th>\n",
        "      <td>    B83</td>\n",
        "      <td>  F </td>\n",
        "      <td> 84</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>79</th>\n",
        "      <td>    B84</td>\n",
        "      <td>  M </td>\n",
        "      <td> 64</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>80</th>\n",
        "      <td>  B85-0</td>\n",
        "      <td>  M </td>\n",
        "      <td> 76</td>\n",
        "      <td>  TaN0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>81</th>\n",
        "      <td>  B85-2</td>\n",
        "      <td>  M </td>\n",
        "      <td> 74</td>\n",
        "      <td>  TaN0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>82</th>\n",
        "      <td>    B86</td>\n",
        "      <td>  M </td>\n",
        "      <td> 48</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>83</th>\n",
        "      <td>    B87</td>\n",
        "      <td>  M </td>\n",
        "      <td> 57</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>84</th>\n",
        "      <td>    B88</td>\n",
        "      <td>  M </td>\n",
        "      <td> 56</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>85</th>\n",
        "      <td>  B89-1</td>\n",
        "      <td>  M </td>\n",
        "      <td> 65</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>86</th>\n",
        "      <td> B89-10</td>\n",
        "      <td>  M </td>\n",
        "      <td> 65</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>87</th>\n",
        "      <td> B89-11</td>\n",
        "      <td>  F </td>\n",
        "      <td> 38</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>88</th>\n",
        "      <td> B89-12</td>\n",
        "      <td>  F </td>\n",
        "      <td> 72</td>\n",
        "      <td>  T2N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>89</th>\n",
        "      <td> B89-16</td>\n",
        "      <td>  M </td>\n",
        "      <td> 68</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>90</th>\n",
        "      <td>  B89-3</td>\n",
        "      <td>  M </td>\n",
        "      <td> 65</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>91</th>\n",
        "      <td>  B89-4</td>\n",
        "      <td>  M </td>\n",
        "      <td> 71</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>92</th>\n",
        "      <td>  B89-5</td>\n",
        "      <td>  M </td>\n",
        "      <td> 50</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>93</th>\n",
        "      <td>     B9</td>\n",
        "      <td>  M </td>\n",
        "      <td> 81</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>94</th>\n",
        "      <td>    B90</td>\n",
        "      <td>  M </td>\n",
        "      <td> 84</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>95</th>\n",
        "      <td>    B92</td>\n",
        "      <td> M  </td>\n",
        "      <td> 53</td>\n",
        "      <td> T3bN1Mx </td>\n",
        "      <td> 3</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>96</th>\n",
        "      <td>    B96</td>\n",
        "      <td>  M </td>\n",
        "      <td> 69</td>\n",
        "      <td>  T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>97</th>\n",
        "      <td>    B98</td>\n",
        "      <td>  M </td>\n",
        "      <td> 63</td>\n",
        "      <td>  T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>98</th>\n",
        "      <td>    B99</td>\n",
        "      <td>  F </td>\n",
        "      <td> 37</td>\n",
        "      <td>  T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>99 rows \u00d7 8 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 2,
       "text": [
        "   Patient_ID Gender  Age  Tumor_ID  Grade     Status   WES   WGS\n",
        "0         B10     M    52   T2N0M0       2   Primary   Yes   Yes \n",
        "1        B100     M    25   T1N0M0       1   Primary   Yes   Yes \n",
        "2        B101     F    69   T1N0M0       1   Primary   Yes   Yes \n",
        "3        B102     M    45   T1N0M0       1   Primary   Yes   Yes \n",
        "4        B103     M    54   T1N0M0       1   Primary   Yes   Yes \n",
        "5        B104     M    52   T2N0M0       1   Primary   Yes   Yes \n",
        "6      B104-0     M    72   T1N0M0       1   Primary   Yes   Yes \n",
        "7        B105     M    62   T2NxMx       1  Relapsed   Yes   Yes \n",
        "8      B105-0     M    60   T2NxMx       1   Primary   Yes   Yes \n",
        "9      B105-1     M    64   T2NxMx       2   Primary   Yes   Yes \n",
        "10       B106     M    56   T2NxMx       1   Primary   Yes   Yes \n",
        "11       B107     M    68   T2NxMx       3   Primary   Yes   Yes \n",
        "12       B109     M    46   T1N0M0       1  Relapsed   Yes   Yes \n",
        "13        B11     M    80   T4N0M0       3  Relapsed   Yes   Yes \n",
        "14       B110     M    54   T1N0M0       1   Primary   Yes   Yes \n",
        "15       B111     M    45   T1N0M0       1   Primary   Yes   Yes \n",
        "16       B112     M    62   T3N1M0       1  Relapsed   Yes   Yes \n",
        "17       B114     M    73  T1bN0M0       1   Primary   Yes   Yes \n",
        "18        B13     M    64   T2N0M0       2   Primary   Yes   Yes \n",
        "19        B14     M    57   T2N0M0       2   Primary   Yes   Yes \n",
        "20        B15     M    40   T4N0M0       2   Primary   Yes   Yes \n",
        "21        B16     M    72   T1N0M0       3   Primary   Yes   Yes \n",
        "22        B18     M    66   T1N0M0       2   Primary   Yes   Yes \n",
        "23        B19     M    69   T4NXMX       3  Relapsed   Yes   Yes \n",
        "24        B21     M    61   T2N0M0       1   Primary   Yes   Yes \n",
        "25        B22     M    79   T4N0M0       2   Primary   Yes   Yes \n",
        "26        B23     M    85   T1N0M0       3   Primary   Yes   Yes \n",
        "27        B24     M    53   T4N0M0       1   Primary   Yes   Yes \n",
        "28        B25     M    65  T4aN0M0       3  Relapsed   Yes   Yes \n",
        "29        B34     M    66  T3aN0M0       3   Primary   Yes   Yes \n",
        "..        ...    ...  ...       ...    ...        ...   ...   ...\n",
        "69      B80-3     M    51   T4N0M0       2   Primary   Yes   Yes \n",
        "70      B80-4     M    54   T1N0M0       1   Primary   Yes   Yes \n",
        "71      B80-5     M    73   TaN0M0       2  Relapsed   Yes   Yes \n",
        "72      B80-7     M    58   T3N0M0       2   Primary   Yes   Yes \n",
        "73      B80-8     F    63   T2N0M0       3   Primary   Yes   Yes \n",
        "74        B81     M    63   T2N0M0       3   Primary   Yes   Yes \n",
        "75      B81-1     M    62   T2N0M0       1   Primary   Yes   Yes \n",
        "76      B81-2     M    61   T2N0M0       2   Primary   Yes   Yes \n",
        "77        B82     M    56   T3N0M0       2   Primary   Yes   Yes \n",
        "78        B83     F    84   T3N0M0       1   Primary   Yes   Yes \n",
        "79        B84     M    64   T1N0M0       1   Primary   Yes   Yes \n",
        "80      B85-0     M    76   TaN0M0       1   Primary   Yes   Yes \n",
        "81      B85-2     M    74   TaN0M0       1   Primary   Yes   Yes \n",
        "82        B86     M    48   T2N0M0       1   Primary   Yes   Yes \n",
        "83        B87     M    57   T1N0M0       2   Primary   Yes   Yes \n",
        "84        B88     M    56   T3N0M0       2   Primary   Yes   Yes \n",
        "85      B89-1     M    65   T2N0M0       2   Primary   Yes   Yes \n",
        "86     B89-10     M    65   T1N0M0       1   Primary   Yes   Yes \n",
        "87     B89-11     F    38   T1N0M0       1   Primary   Yes   Yes \n",
        "88     B89-12     F    72   T2N0M0       3   Primary   Yes   Yes \n",
        "89     B89-16     M    68   T1N0M0       1   Primary   Yes   Yes \n",
        "90      B89-3     M    65   T3N0M0       1   Primary   Yes   Yes \n",
        "91      B89-4     M    71   T1N0M0       1   Primary   Yes   Yes \n",
        "92      B89-5     M    50   T1N0M0       1   Primary   Yes   Yes \n",
        "93         B9     M    81   T3N0M0       3   Primary   Yes   Yes \n",
        "94        B90     M    84   T1N0M0       1   Primary   Yes   Yes \n",
        "95        B92    M     53  T3bN1Mx       3   Primary   Yes   Yes \n",
        "96        B96     M    69   T3N0M0       2  Relapsed   Yes   Yes \n",
        "97        B98     M    63   T3N1MX       3  Relapsed   Yes   Yes \n",
        "98        B99     F    37   T1N0M0       1   Primary   Yes   Yes \n",
        "\n",
        "[99 rows x 8 columns]"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Mutation Data:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "# Mutation Data\n",
      "\n",
      "#to get the chromosome from genome change.. how?\n",
      "## can I take the characters in each string between chr and : and strip it out to a new chr column????\n",
      "\n",
      "\n",
      "#read in csv\n",
      "mut = pd.read_csv(PATH + 'ng.2798-S3.csv')\n",
      "#drop unwanted columns\n",
      "mut.drop(['Codon change','Nucleotide (cDNA)$','Amino Acid'], 1, inplace=True)\n",
      "#rename columns to format\n",
      "mut.rename(columns={'Gene Symbol':'Gene','Tumor ID':'Patient_ID','Nucleotide (genomic)#':'Genome_Change','Mutation Type':'Mutation_Type'}, inplace=True)\n",
      "\n",
      "mut['Patient_ID'] = mut['Patient_ID'].map(str.strip)\n",
      "#write to csv in Mutdata folder\n",
      "mut.to_csv(MUT_PATH+ 'mutdataWGSWESbladder.csv')\n",
      "mut\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Gene</th>\n",
        "      <th>Patient_ID</th>\n",
        "      <th>Genome_Change</th>\n",
        "      <th>Protein_Change</th>\n",
        "      <th>Mutation_Type</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0   </th>\n",
        "      <td>    ABCA8</td>\n",
        "      <td>    B65</td>\n",
        "      <td>  g.chr17:64393053T&gt;A</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1   </th>\n",
        "      <td>   ACAD10</td>\n",
        "      <td>  B81-1</td>\n",
        "      <td> g.chr12:110624480A&gt;T</td>\n",
        "      <td>  p.R107*</td>\n",
        "      <td>          nonsense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2   </th>\n",
        "      <td>    ACAT2</td>\n",
        "      <td>  B85-0</td>\n",
        "      <td>  g.chr6:160119679A&gt;T</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3   </th>\n",
        "      <td>    ACIN1</td>\n",
        "      <td> B105-0</td>\n",
        "      <td>  g.chr14:22617904C&gt;T</td>\n",
        "      <td>  p.E716K</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4   </th>\n",
        "      <td>    ACIN1</td>\n",
        "      <td>    B22</td>\n",
        "      <td>  g.chr14:22618068G&gt;A</td>\n",
        "      <td>  p.S661F</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5   </th>\n",
        "      <td>    ACIN1</td>\n",
        "      <td>    B45</td>\n",
        "      <td>  g.chr14:22603490G&gt;A</td>\n",
        "      <td>  p.Q844*</td>\n",
        "      <td>          nonsense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6   </th>\n",
        "      <td>    ACIN1</td>\n",
        "      <td>    B52</td>\n",
        "      <td>  g.chr14:22634096C&gt;T</td>\n",
        "      <td>   p.L80L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7   </th>\n",
        "      <td>    ACIN1</td>\n",
        "      <td> B80-13</td>\n",
        "      <td>  g.chr14:22602557C&gt;T</td>\n",
        "      <td>  p.E947K</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8   </th>\n",
        "      <td>     ACTB</td>\n",
        "      <td>   B101</td>\n",
        "      <td>    g.chr7:5534867C&gt;T</td>\n",
        "      <td>  p.E125K</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9   </th>\n",
        "      <td>     ACTB</td>\n",
        "      <td> B105-0</td>\n",
        "      <td>    g.chr7:5534806G&gt;A</td>\n",
        "      <td>  p.S145F</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10  </th>\n",
        "      <td>     ACTB</td>\n",
        "      <td>  B59-3</td>\n",
        "      <td>    g.chr7:5534768C&gt;T</td>\n",
        "      <td>  p.G158R</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11  </th>\n",
        "      <td>     ACTB</td>\n",
        "      <td>    B64</td>\n",
        "      <td>    g.chr7:5533990G&gt;A</td>\n",
        "      <td>  p.S348L</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12  </th>\n",
        "      <td>    ACTG1</td>\n",
        "      <td>   B101</td>\n",
        "      <td>  g.chr17:77093710C&gt;T</td>\n",
        "      <td>   p.Q59Q</td>\n",
        "      <td> synonymous-coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>13  </th>\n",
        "      <td> ADAMTSL3</td>\n",
        "      <td>     B5</td>\n",
        "      <td>  g.chr15:82491167C&gt;G</td>\n",
        "      <td> p.S1578C</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>14  </th>\n",
        "      <td> ADAMTSL3</td>\n",
        "      <td>    B79</td>\n",
        "      <td>  g.chr15:82442799C&gt;T</td>\n",
        "      <td> p.R1139W</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15  </th>\n",
        "      <td> ADAMTSL3</td>\n",
        "      <td>    B83</td>\n",
        "      <td>  g.chr15:82383690A&gt;G</td>\n",
        "      <td>  p.H673R</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16  </th>\n",
        "      <td>    ADCY2</td>\n",
        "      <td>   B106</td>\n",
        "      <td>    g.chr5:7842879C&gt;G</td>\n",
        "      <td>  p.A865G</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17  </th>\n",
        "      <td>    ADCY2</td>\n",
        "      <td>    B47</td>\n",
        "      <td>    g.chr5:7679420A&gt;G</td>\n",
        "      <td>  p.K237K</td>\n",
        "      <td> synonymous-coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18  </th>\n",
        "      <td>    ADCY2</td>\n",
        "      <td>     B5</td>\n",
        "      <td>    g.chr5:7762361G&gt;A</td>\n",
        "      <td>  p.R480H</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19  </th>\n",
        "      <td>    ADCY8</td>\n",
        "      <td>   B114</td>\n",
        "      <td>  g.chr8:131917703A&gt;T</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>20  </th>\n",
        "      <td>    ADCY8</td>\n",
        "      <td>    B59</td>\n",
        "      <td>  g.chr8:131866873C&gt;T</td>\n",
        "      <td> p.E1031K</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>21  </th>\n",
        "      <td>    AHNAK</td>\n",
        "      <td>  B59-0</td>\n",
        "      <td>  g.chr11:62042315T&gt;C</td>\n",
        "      <td> p.K5384E</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22  </th>\n",
        "      <td>    AHNAK</td>\n",
        "      <td>    B63</td>\n",
        "      <td>  g.chr11:62051062T&gt;C</td>\n",
        "      <td> p.K2468R</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23  </th>\n",
        "      <td>    AHNAK</td>\n",
        "      <td>    B78</td>\n",
        "      <td>  g.chr11:62048829C&gt;T</td>\n",
        "      <td> p.M3212I</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>24  </th>\n",
        "      <td>    AHNAK</td>\n",
        "      <td>  B81-2</td>\n",
        "      <td>  g.chr11:62042503C&gt;T</td>\n",
        "      <td> p.S5321N</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25  </th>\n",
        "      <td>    AHNAK</td>\n",
        "      <td> B89-12</td>\n",
        "      <td>  g.chr11:62056821T&gt;A</td>\n",
        "      <td>  p.L548L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>26  </th>\n",
        "      <td>    AHNAK</td>\n",
        "      <td> B89-16</td>\n",
        "      <td>  g.chr11:62054251T&gt;C</td>\n",
        "      <td> p.K1405R</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>27  </th>\n",
        "      <td>     AIRE</td>\n",
        "      <td>   B111</td>\n",
        "      <td>  g.chr21:44537326C&gt;T</td>\n",
        "      <td>  p.A176V</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>28  </th>\n",
        "      <td>     AIRE</td>\n",
        "      <td>   B111</td>\n",
        "      <td>  g.chr21:44537355G&gt;A</td>\n",
        "      <td>  p.E186K</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>29  </th>\n",
        "      <td>   AKAP13</td>\n",
        "      <td>    B23</td>\n",
        "      <td>  g.chr15:83878113A&gt;T</td>\n",
        "      <td>  p.H159L</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1032</th>\n",
        "      <td>     WEE1</td>\n",
        "      <td>  B89-4</td>\n",
        "      <td>   g.chr11:9554010G&gt;T</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1033</th>\n",
        "      <td>  WHSC1L1</td>\n",
        "      <td>    B74</td>\n",
        "      <td>   g.chr8:38291449C&gt;T</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1034</th>\n",
        "      <td>   WNT10B</td>\n",
        "      <td> B104-0</td>\n",
        "      <td>  g.chr12:47650545G&gt;A</td>\n",
        "      <td>   p.S12L</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1035</th>\n",
        "      <td>   WNT10B</td>\n",
        "      <td>  B89-4</td>\n",
        "      <td>  g.chr12:47648000C&gt;T</td>\n",
        "      <td>  p.R236H</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1036</th>\n",
        "      <td>     WNT2</td>\n",
        "      <td>    B22</td>\n",
        "      <td>  g.chr7:116742602G&gt;A</td>\n",
        "      <td>  p.S116L</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1037</th>\n",
        "      <td>     WNT2</td>\n",
        "      <td>  B89-4</td>\n",
        "      <td>  g.chr7:116725165G&gt;A</td>\n",
        "      <td>  p.A197V</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1038</th>\n",
        "      <td>     XPO4</td>\n",
        "      <td>    B54</td>\n",
        "      <td>  g.chr13:20294018C&gt;G</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1039</th>\n",
        "      <td>    ZFHX3</td>\n",
        "      <td>    B15</td>\n",
        "      <td>  g.chr16:71379700C&gt;T</td>\n",
        "      <td> p.A3326T</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1040</th>\n",
        "      <td>    ZFHX3</td>\n",
        "      <td>    B60</td>\n",
        "      <td>  g.chr16:71550398G&gt;A</td>\n",
        "      <td>  p.S383F</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1041</th>\n",
        "      <td>    ZFHX3</td>\n",
        "      <td>  B80-0</td>\n",
        "      <td>  g.chr16:71384976G&gt;A</td>\n",
        "      <td> p.R3036W</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1042</th>\n",
        "      <td>    ZFHX3</td>\n",
        "      <td>    B80</td>\n",
        "      <td>  g.chr16:71549289T&gt;C</td>\n",
        "      <td>  p.N753D</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1043</th>\n",
        "      <td>     ZIK1</td>\n",
        "      <td>     B5</td>\n",
        "      <td>  g.chr19:62793764A&gt;T</td>\n",
        "      <td>  p.Q258L</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1044</th>\n",
        "      <td>    ZMYM3</td>\n",
        "      <td>    B59</td>\n",
        "      <td>   g.chrX:70378945C&gt;T</td>\n",
        "      <td> p.G1201E</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1045</th>\n",
        "      <td>    ZMYM3</td>\n",
        "      <td>    B82</td>\n",
        "      <td>   g.chrX:70386653G&gt;C</td>\n",
        "      <td>  p.S400C</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1046</th>\n",
        "      <td>  ZMYND10</td>\n",
        "      <td>     B5</td>\n",
        "      <td>   g.chr3:50357582C&gt;G</td>\n",
        "      <td>   p.E60Q</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1047</th>\n",
        "      <td>   ZMYND8</td>\n",
        "      <td>    B21</td>\n",
        "      <td>  g.chr20:45338770T&gt;G</td>\n",
        "      <td>  p.N392T</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1048</th>\n",
        "      <td>   ZMYND8</td>\n",
        "      <td>    B66</td>\n",
        "      <td>  g.chr20:45311519G&gt;A</td>\n",
        "      <td>  p.L565F</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1049</th>\n",
        "      <td>   ZMYND8</td>\n",
        "      <td>  B89-3</td>\n",
        "      <td>  g.chr20:45356915C&gt;G</td>\n",
        "      <td>  p.E157Q</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1050</th>\n",
        "      <td>   ZNF174</td>\n",
        "      <td>    B13</td>\n",
        "      <td>   g.chr16:3398555A&gt;T</td>\n",
        "      <td>  p.S287C</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1051</th>\n",
        "      <td>   ZNF174</td>\n",
        "      <td>    B65</td>\n",
        "      <td>   g.chr16:3398475C&gt;T</td>\n",
        "      <td>  p.S260L</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1052</th>\n",
        "      <td>   ZNF295</td>\n",
        "      <td>     B5</td>\n",
        "      <td>  g.chr21:42287117C&gt;T</td>\n",
        "      <td>   p.E53K</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1053</th>\n",
        "      <td>   ZNF471</td>\n",
        "      <td>    B15</td>\n",
        "      <td>  g.chr19:61719525C&gt;G</td>\n",
        "      <td>   p.Q35E</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1054</th>\n",
        "      <td>   ZNF536</td>\n",
        "      <td>    B15</td>\n",
        "      <td>  g.chr19:35717711C&gt;T</td>\n",
        "      <td>  p.S763F</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1055</th>\n",
        "      <td>   ZNF623</td>\n",
        "      <td>   B106</td>\n",
        "      <td>  g.chr8:144804645A&gt;G</td>\n",
        "      <td>  p.Y487C</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1056</th>\n",
        "      <td>   ZNF623</td>\n",
        "      <td>     B5</td>\n",
        "      <td>  g.chr8:144804284A&gt;T</td>\n",
        "      <td>  p.R367*</td>\n",
        "      <td>          nonsense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1057</th>\n",
        "      <td>    ZNF80</td>\n",
        "      <td>   B106</td>\n",
        "      <td>  g.chr3:115438210C&gt;G</td>\n",
        "      <td>  p.E134D</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1058</th>\n",
        "      <td>    ZNF83</td>\n",
        "      <td>    B78</td>\n",
        "      <td>  g.chr19:57808279G&gt;A</td>\n",
        "      <td>  p.H451Y</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1059</th>\n",
        "      <td>    ZNF83</td>\n",
        "      <td>  B89-5</td>\n",
        "      <td>  g.chr19:57808751C&gt;T</td>\n",
        "      <td>  p.E293E</td>\n",
        "      <td> synonymous-coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1060</th>\n",
        "      <td>    ZNF83</td>\n",
        "      <td>  B89-5</td>\n",
        "      <td>  g.chr19:57808752T&gt;A</td>\n",
        "      <td>  p.E293V</td>\n",
        "      <td>          missense</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1061</th>\n",
        "      <td>    ZZEF1</td>\n",
        "      <td>    B59</td>\n",
        "      <td>   g.chr17:3912961C&gt;G</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>1062 rows \u00d7 5 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 3,
       "text": [
        "          Gene Patient_ID         Genome_Change Protein_Change  \\\n",
        "0        ABCA8        B65   g.chr17:64393053T>A              -   \n",
        "1       ACAD10      B81-1  g.chr12:110624480A>T        p.R107*   \n",
        "2        ACAT2      B85-0   g.chr6:160119679A>T              -   \n",
        "3        ACIN1     B105-0   g.chr14:22617904C>T        p.E716K   \n",
        "4        ACIN1        B22   g.chr14:22618068G>A        p.S661F   \n",
        "5        ACIN1        B45   g.chr14:22603490G>A        p.Q844*   \n",
        "6        ACIN1        B52   g.chr14:22634096C>T         p.L80L   \n",
        "7        ACIN1     B80-13   g.chr14:22602557C>T        p.E947K   \n",
        "8         ACTB       B101     g.chr7:5534867C>T        p.E125K   \n",
        "9         ACTB     B105-0     g.chr7:5534806G>A        p.S145F   \n",
        "10        ACTB      B59-3     g.chr7:5534768C>T        p.G158R   \n",
        "11        ACTB        B64     g.chr7:5533990G>A        p.S348L   \n",
        "12       ACTG1       B101   g.chr17:77093710C>T         p.Q59Q   \n",
        "13    ADAMTSL3         B5   g.chr15:82491167C>G       p.S1578C   \n",
        "14    ADAMTSL3        B79   g.chr15:82442799C>T       p.R1139W   \n",
        "15    ADAMTSL3        B83   g.chr15:82383690A>G        p.H673R   \n",
        "16       ADCY2       B106     g.chr5:7842879C>G        p.A865G   \n",
        "17       ADCY2        B47     g.chr5:7679420A>G        p.K237K   \n",
        "18       ADCY2         B5     g.chr5:7762361G>A        p.R480H   \n",
        "19       ADCY8       B114   g.chr8:131917703A>T              -   \n",
        "20       ADCY8        B59   g.chr8:131866873C>T       p.E1031K   \n",
        "21       AHNAK      B59-0   g.chr11:62042315T>C       p.K5384E   \n",
        "22       AHNAK        B63   g.chr11:62051062T>C       p.K2468R   \n",
        "23       AHNAK        B78   g.chr11:62048829C>T       p.M3212I   \n",
        "24       AHNAK      B81-2   g.chr11:62042503C>T       p.S5321N   \n",
        "25       AHNAK     B89-12   g.chr11:62056821T>A        p.L548L   \n",
        "26       AHNAK     B89-16   g.chr11:62054251T>C       p.K1405R   \n",
        "27        AIRE       B111   g.chr21:44537326C>T        p.A176V   \n",
        "28        AIRE       B111   g.chr21:44537355G>A        p.E186K   \n",
        "29      AKAP13        B23   g.chr15:83878113A>T        p.H159L   \n",
        "...        ...        ...                   ...            ...   \n",
        "1032      WEE1      B89-4    g.chr11:9554010G>T              -   \n",
        "1033   WHSC1L1        B74    g.chr8:38291449C>T              -   \n",
        "1034    WNT10B     B104-0   g.chr12:47650545G>A         p.S12L   \n",
        "1035    WNT10B      B89-4   g.chr12:47648000C>T        p.R236H   \n",
        "1036      WNT2        B22   g.chr7:116742602G>A        p.S116L   \n",
        "1037      WNT2      B89-4   g.chr7:116725165G>A        p.A197V   \n",
        "1038      XPO4        B54   g.chr13:20294018C>G              -   \n",
        "1039     ZFHX3        B15   g.chr16:71379700C>T       p.A3326T   \n",
        "1040     ZFHX3        B60   g.chr16:71550398G>A        p.S383F   \n",
        "1041     ZFHX3      B80-0   g.chr16:71384976G>A       p.R3036W   \n",
        "1042     ZFHX3        B80   g.chr16:71549289T>C        p.N753D   \n",
        "1043      ZIK1         B5   g.chr19:62793764A>T        p.Q258L   \n",
        "1044     ZMYM3        B59    g.chrX:70378945C>T       p.G1201E   \n",
        "1045     ZMYM3        B82    g.chrX:70386653G>C        p.S400C   \n",
        "1046   ZMYND10         B5    g.chr3:50357582C>G         p.E60Q   \n",
        "1047    ZMYND8        B21   g.chr20:45338770T>G        p.N392T   \n",
        "1048    ZMYND8        B66   g.chr20:45311519G>A        p.L565F   \n",
        "1049    ZMYND8      B89-3   g.chr20:45356915C>G        p.E157Q   \n",
        "1050    ZNF174        B13    g.chr16:3398555A>T        p.S287C   \n",
        "1051    ZNF174        B65    g.chr16:3398475C>T        p.S260L   \n",
        "1052    ZNF295         B5   g.chr21:42287117C>T         p.E53K   \n",
        "1053    ZNF471        B15   g.chr19:61719525C>G         p.Q35E   \n",
        "1054    ZNF536        B15   g.chr19:35717711C>T        p.S763F   \n",
        "1055    ZNF623       B106   g.chr8:144804645A>G        p.Y487C   \n",
        "1056    ZNF623         B5   g.chr8:144804284A>T        p.R367*   \n",
        "1057     ZNF80       B106   g.chr3:115438210C>G        p.E134D   \n",
        "1058     ZNF83        B78   g.chr19:57808279G>A        p.H451Y   \n",
        "1059     ZNF83      B89-5   g.chr19:57808751C>T        p.E293E   \n",
        "1060     ZNF83      B89-5   g.chr19:57808752T>A        p.E293V   \n",
        "1061     ZZEF1        B59    g.chr17:3912961C>G              -   \n",
        "\n",
        "          Mutation_Type  \n",
        "0            spliceSite  \n",
        "1              nonsense  \n",
        "2            spliceSite  \n",
        "3              missense  \n",
        "4              missense  \n",
        "5              nonsense  \n",
        "6     synonymous-coding  \n",
        "7              missense  \n",
        "8              missense  \n",
        "9              missense  \n",
        "10             missense  \n",
        "11             missense  \n",
        "12    synonymous-coding  \n",
        "13             missense  \n",
        "14             missense  \n",
        "15             missense  \n",
        "16             missense  \n",
        "17    synonymous-coding  \n",
        "18             missense  \n",
        "19           spliceSite  \n",
        "20             missense  \n",
        "21             missense  \n",
        "22             missense  \n",
        "23             missense  \n",
        "24             missense  \n",
        "25    synonymous-coding  \n",
        "26             missense  \n",
        "27             missense  \n",
        "28             missense  \n",
        "29             missense  \n",
        "...                 ...  \n",
        "1032         spliceSite  \n",
        "1033         spliceSite  \n",
        "1034           missense  \n",
        "1035           missense  \n",
        "1036           missense  \n",
        "1037           missense  \n",
        "1038         spliceSite  \n",
        "1039           missense  \n",
        "1040           missense  \n",
        "1041           missense  \n",
        "1042           missense  \n",
        "1043           missense  \n",
        "1044           missense  \n",
        "1045           missense  \n",
        "1046           missense  \n",
        "1047           missense  \n",
        "1048           missense  \n",
        "1049           missense  \n",
        "1050           missense  \n",
        "1051           missense  \n",
        "1052           missense  \n",
        "1053           missense  \n",
        "1054           missense  \n",
        "1055           missense  \n",
        "1056           nonsense  \n",
        "1057           missense  \n",
        "1058           missense  \n",
        "1059  synonymous-coding  \n",
        "1060           missense  \n",
        "1061         spliceSite  \n",
        "\n",
        "[1062 rows x 5 columns]"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Merged Data:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#I had originally merged to the outer, but there were ten more rows than genes.. therefore i merged to right which\n",
      "#is on mut data and it has now same number... need to figure out merging better..\n",
      "merged = pd.merge(patdata, mut, how='right')\n",
      "\n",
      "merged['Tissue'] = 'Bladder cancer - transitional cell carcinoma'\n",
      "\n",
      "merged.to_csv(MERGED_PATH+'mergedWGSWESbladder.csv')\n",
      "\n",
      "merged"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Patient_ID</th>\n",
        "      <th>Gender</th>\n",
        "      <th>Age</th>\n",
        "      <th>Tumor_ID</th>\n",
        "      <th>Grade</th>\n",
        "      <th>Status</th>\n",
        "      <th>WES</th>\n",
        "      <th>WGS</th>\n",
        "      <th>Gene</th>\n",
        "      <th>Genome_Change</th>\n",
        "      <th>Protein_Change</th>\n",
        "      <th>Mutation_Type</th>\n",
        "      <th>Tissue</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    ACTB</td>\n",
        "      <td>                     g.chr7:5534867C&gt;T</td>\n",
        "      <td>  p.E125K</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   ACTG1</td>\n",
        "      <td>                   g.chr17:77093710C&gt;T</td>\n",
        "      <td>   p.Q59Q</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  ATP2A3</td>\n",
        "      <td>                    g.chr17:3778304G&gt;A</td>\n",
        "      <td> p.S1016F</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   FRRS1</td>\n",
        "      <td>                    g.chr1:99979085C&gt;T</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   HUWE1</td>\n",
        "      <td>                    g.chrX:53593080G&gt;A</td>\n",
        "      <td> p.L3200L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    MLL5</td>\n",
        "      <td>                   g.chr7:104539925C&gt;A</td>\n",
        "      <td> p.Q1496K</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     NF1</td>\n",
        "      <td>                   g.chr17:26581441C&gt;T</td>\n",
        "      <td> p.Q1010*</td>\n",
        "      <td>          nonsense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     NF1</td>\n",
        "      <td>                   g.chr17:26689262G&gt;C</td>\n",
        "      <td> p.Q2266H</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     NHS</td>\n",
        "      <td>                    g.chrX:17656689G&gt;A</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9   </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   SALL4</td>\n",
        "      <td>                   g.chr20:49840373C&gt;T</td>\n",
        "      <td>  p.D686N</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  SFRS16</td>\n",
        "      <td>                   g.chr19:50247168G&gt;A</td>\n",
        "      <td>        -</td>\n",
        "      <td>        spliceSite</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>      TG</td>\n",
        "      <td>                   g.chr8:133963297G&gt;A</td>\n",
        "      <td>  p.D216N</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    TP53</td>\n",
        "      <td>                    g.chr17:7517824C&gt;G</td>\n",
        "      <td>  p.R280T</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>13  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  TRIM23</td>\n",
        "      <td>                    g.chr5:64943386C&gt;T</td>\n",
        "      <td>  p.E149K</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>14  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     UTX</td>\n",
        "      <td>                    g.chrX:44803586C&gt;T</td>\n",
        "      <td>  p.Y375Y</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     UTX</td>\n",
        "      <td> g.chrX:44807883_44807892delAAGTGGAAGT</td>\n",
        "      <td>        -</td>\n",
        "      <td> Frame-shift_indel</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16  </th>\n",
        "      <td>   B101</td>\n",
        "      <td> F </td>\n",
        "      <td> 69</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     UTX</td>\n",
        "      <td> g.chrX:44813956_44813965delTACCTCAGGT</td>\n",
        "      <td>        -</td>\n",
        "      <td> Frame-shift_indel</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17  </th>\n",
        "      <td>   B103</td>\n",
        "      <td> M </td>\n",
        "      <td> 54</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   FGFR3</td>\n",
        "      <td>                     g.chr4:1773362C&gt;T</td>\n",
        "      <td>  p.R248C</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18  </th>\n",
        "      <td>   B103</td>\n",
        "      <td> M </td>\n",
        "      <td> 54</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    LRP1</td>\n",
        "      <td>                   g.chr12:55858590G&gt;T</td>\n",
        "      <td> p.V1515L</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>19  </th>\n",
        "      <td>   B103</td>\n",
        "      <td> M </td>\n",
        "      <td> 54</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     MLL</td>\n",
        "      <td>                  g.chr11:117864574C&gt;A</td>\n",
        "      <td> p.H1456Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>20  </th>\n",
        "      <td>   B103</td>\n",
        "      <td> M </td>\n",
        "      <td> 54</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  NFE2L3</td>\n",
        "      <td>                    g.chr7:26191453G&gt;A</td>\n",
        "      <td>  p.R537H</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>21  </th>\n",
        "      <td>   B103</td>\n",
        "      <td> M </td>\n",
        "      <td> 54</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    TSC1</td>\n",
        "      <td>        g.chr9:134775785_134775785delG</td>\n",
        "      <td>        -</td>\n",
        "      <td> Frame-shift_indel</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>22  </th>\n",
        "      <td>   B103</td>\n",
        "      <td> M </td>\n",
        "      <td> 54</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     UTX</td>\n",
        "      <td>                   g.chrX:44814463insA</td>\n",
        "      <td>        -</td>\n",
        "      <td> Frame-shift_indel</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>23  </th>\n",
        "      <td>   B104</td>\n",
        "      <td> M </td>\n",
        "      <td> 52</td>\n",
        "      <td> T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  ATP2A3</td>\n",
        "      <td>                    g.chr17:3801309T&gt;C</td>\n",
        "      <td>  p.I150V</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>24  </th>\n",
        "      <td>   B104</td>\n",
        "      <td> M </td>\n",
        "      <td> 52</td>\n",
        "      <td> T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   FGFR3</td>\n",
        "      <td>                     g.chr4:1775897A&gt;G</td>\n",
        "      <td>  p.Y373C</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>25  </th>\n",
        "      <td>   B104</td>\n",
        "      <td> M </td>\n",
        "      <td> 52</td>\n",
        "      <td> T2N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   SYNE1</td>\n",
        "      <td>                   g.chr6:152788249C&gt;T</td>\n",
        "      <td> p.D1750N</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>26  </th>\n",
        "      <td> B104-0</td>\n",
        "      <td> M </td>\n",
        "      <td> 72</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    CUL5</td>\n",
        "      <td>                  g.chr11:107422280C&gt;G</td>\n",
        "      <td>   p.L70V</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>27  </th>\n",
        "      <td> B104-0</td>\n",
        "      <td> M </td>\n",
        "      <td> 72</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   DNAH8</td>\n",
        "      <td>                    g.chr6:38898765G&gt;C</td>\n",
        "      <td> p.E1016Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>28  </th>\n",
        "      <td> B104-0</td>\n",
        "      <td> M </td>\n",
        "      <td> 72</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    E4F1</td>\n",
        "      <td>                    g.chr16:2224294C&gt;G</td>\n",
        "      <td>  p.L499L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>29  </th>\n",
        "      <td> B104-0</td>\n",
        "      <td> M </td>\n",
        "      <td> 72</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   ERBB2</td>\n",
        "      <td>                   g.chr17:35136779C&gt;G</td>\n",
        "      <td> p.T1052T</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1032</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   OPCML</td>\n",
        "      <td>                  g.chr11:132032228G&gt;A</td>\n",
        "      <td>  p.H122Y</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1033</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   PARD3</td>\n",
        "      <td>                   g.chr10:34713082G&gt;C</td>\n",
        "      <td>  p.R333G</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1034</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   PDZD2</td>\n",
        "      <td>                    g.chr5:32109904G&gt;C</td>\n",
        "      <td>  p.E979Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1035</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    PFKL</td>\n",
        "      <td>                   g.chr21:44550152C&gt;A</td>\n",
        "      <td>   p.F44L</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1036</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     RB1</td>\n",
        "      <td>         g.chr13:47835030_47835030delA</td>\n",
        "      <td>        -</td>\n",
        "      <td> Frame-shift_indel</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1037</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    RXRA</td>\n",
        "      <td>                   g.chr9:136468172C&gt;T</td>\n",
        "      <td>  p.S427F</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1038</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   SCRIB</td>\n",
        "      <td>                   g.chr8:144957927G&gt;A</td>\n",
        "      <td> p.L1098L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1039</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     SP1</td>\n",
        "      <td>                   g.chr12:52063362C&gt;G</td>\n",
        "      <td>  p.T455R</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1040</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    STAM</td>\n",
        "      <td>                   g.chr10:17796768C&gt;T</td>\n",
        "      <td>  p.Q536*</td>\n",
        "      <td>          nonsense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1041</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   SYNE1</td>\n",
        "      <td>                   g.chr6:152499480C&gt;T</td>\n",
        "      <td>  p.R727Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1042</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   SYNE2</td>\n",
        "      <td>                   g.chr14:63567722A&gt;C</td>\n",
        "      <td> p.K2372T</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1043</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    TP53</td>\n",
        "      <td>                    g.chr17:7517747G&gt;A</td>\n",
        "      <td>  p.R306*</td>\n",
        "      <td>          nonsense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1044</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    UBR4</td>\n",
        "      <td>                    g.chr1:19340826C&gt;G</td>\n",
        "      <td> p.D2738H</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1045</th>\n",
        "      <td>    B96</td>\n",
        "      <td> M </td>\n",
        "      <td> 69</td>\n",
        "      <td> T3N0M0 </td>\n",
        "      <td> 2</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>     UTX</td>\n",
        "      <td>                    g.chrX:44807746C&gt;T</td>\n",
        "      <td>  p.Q555*</td>\n",
        "      <td>          nonsense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1046</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    BAI3</td>\n",
        "      <td>                    g.chr6:70122468G&gt;A</td>\n",
        "      <td> p.R1197Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1047</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    BRAF</td>\n",
        "      <td>                   g.chr7:140099623T&gt;C</td>\n",
        "      <td>  p.D594G</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1048</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   BRCA2</td>\n",
        "      <td>                   g.chr13:31810996C&gt;G</td>\n",
        "      <td> p.Q1502E</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1049</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  CABIN1</td>\n",
        "      <td>                   g.chr22:22824071C&gt;T</td>\n",
        "      <td> p.L1345L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1050</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  CREBBP</td>\n",
        "      <td>                    g.chr16:3718536C&gt;T</td>\n",
        "      <td> p.L2171L</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1051</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   DNAH8</td>\n",
        "      <td>                    g.chr6:38948800T&gt;C</td>\n",
        "      <td> p.F2243L</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1052</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td> DSCAML1</td>\n",
        "      <td>                  g.chr11:116804694C&gt;G</td>\n",
        "      <td> p.E1968Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1053</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td> IL18RAP</td>\n",
        "      <td>                   g.chr2:102430091C&gt;T</td>\n",
        "      <td>  p.T401M</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1054</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  NLRP14</td>\n",
        "      <td>                    g.chr11:7020534C&gt;G</td>\n",
        "      <td>  p.S234*</td>\n",
        "      <td>          nonsense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1055</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  PIK3CA</td>\n",
        "      <td>                   g.chr3:180418776G&gt;A</td>\n",
        "      <td>  p.E542K</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1056</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  PIK3CA</td>\n",
        "      <td>                   g.chr3:180418785G&gt;C</td>\n",
        "      <td>  p.E545Q</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1057</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  PKD1L1</td>\n",
        "      <td>                    g.chr7:47935479G&gt;A</td>\n",
        "      <td>  p.R303W</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1058</th>\n",
        "      <td>    B98</td>\n",
        "      <td> M </td>\n",
        "      <td> 63</td>\n",
        "      <td> T3N1MX </td>\n",
        "      <td> 3</td>\n",
        "      <td> Relapsed </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>   SYNE1</td>\n",
        "      <td>                   g.chr6:152630895A&gt;G</td>\n",
        "      <td> p.S6197S</td>\n",
        "      <td> synonymous-coding</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1059</th>\n",
        "      <td>    B99</td>\n",
        "      <td> F </td>\n",
        "      <td> 37</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    HRAS</td>\n",
        "      <td>                     g.chr11:523303G&gt;T</td>\n",
        "      <td>  p.P169H</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1060</th>\n",
        "      <td>    B99</td>\n",
        "      <td> F </td>\n",
        "      <td> 37</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>    HRAS</td>\n",
        "      <td>                     g.chr11:524288C&gt;T</td>\n",
        "      <td>   p.G12D</td>\n",
        "      <td>          missense</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1061</th>\n",
        "      <td>    B99</td>\n",
        "      <td> F </td>\n",
        "      <td> 37</td>\n",
        "      <td> T1N0M0 </td>\n",
        "      <td> 1</td>\n",
        "      <td>  Primary </td>\n",
        "      <td> Yes </td>\n",
        "      <td> Yes </td>\n",
        "      <td>  PDGFRB</td>\n",
        "      <td>   g.chr5:149473043_149473048delCTCTTC</td>\n",
        "      <td>        -</td>\n",
        "      <td> Frame-shift_indel</td>\n",
        "      <td> Bladder cancer - transitional cell carcinoma</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>1062 rows \u00d7 13 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 5,
       "text": [
        "     Patient_ID Gender  Age Tumor_ID  Grade     Status   WES   WGS     Gene  \\\n",
        "0          B101     F    69  T1N0M0       1   Primary   Yes   Yes      ACTB   \n",
        "1          B101     F    69  T1N0M0       1   Primary   Yes   Yes     ACTG1   \n",
        "2          B101     F    69  T1N0M0       1   Primary   Yes   Yes    ATP2A3   \n",
        "3          B101     F    69  T1N0M0       1   Primary   Yes   Yes     FRRS1   \n",
        "4          B101     F    69  T1N0M0       1   Primary   Yes   Yes     HUWE1   \n",
        "5          B101     F    69  T1N0M0       1   Primary   Yes   Yes      MLL5   \n",
        "6          B101     F    69  T1N0M0       1   Primary   Yes   Yes       NF1   \n",
        "7          B101     F    69  T1N0M0       1   Primary   Yes   Yes       NF1   \n",
        "8          B101     F    69  T1N0M0       1   Primary   Yes   Yes       NHS   \n",
        "9          B101     F    69  T1N0M0       1   Primary   Yes   Yes     SALL4   \n",
        "10         B101     F    69  T1N0M0       1   Primary   Yes   Yes    SFRS16   \n",
        "11         B101     F    69  T1N0M0       1   Primary   Yes   Yes        TG   \n",
        "12         B101     F    69  T1N0M0       1   Primary   Yes   Yes      TP53   \n",
        "13         B101     F    69  T1N0M0       1   Primary   Yes   Yes    TRIM23   \n",
        "14         B101     F    69  T1N0M0       1   Primary   Yes   Yes       UTX   \n",
        "15         B101     F    69  T1N0M0       1   Primary   Yes   Yes       UTX   \n",
        "16         B101     F    69  T1N0M0       1   Primary   Yes   Yes       UTX   \n",
        "17         B103     M    54  T1N0M0       1   Primary   Yes   Yes     FGFR3   \n",
        "18         B103     M    54  T1N0M0       1   Primary   Yes   Yes      LRP1   \n",
        "19         B103     M    54  T1N0M0       1   Primary   Yes   Yes       MLL   \n",
        "20         B103     M    54  T1N0M0       1   Primary   Yes   Yes    NFE2L3   \n",
        "21         B103     M    54  T1N0M0       1   Primary   Yes   Yes      TSC1   \n",
        "22         B103     M    54  T1N0M0       1   Primary   Yes   Yes       UTX   \n",
        "23         B104     M    52  T2N0M0       1   Primary   Yes   Yes    ATP2A3   \n",
        "24         B104     M    52  T2N0M0       1   Primary   Yes   Yes     FGFR3   \n",
        "25         B104     M    52  T2N0M0       1   Primary   Yes   Yes     SYNE1   \n",
        "26       B104-0     M    72  T1N0M0       1   Primary   Yes   Yes      CUL5   \n",
        "27       B104-0     M    72  T1N0M0       1   Primary   Yes   Yes     DNAH8   \n",
        "28       B104-0     M    72  T1N0M0       1   Primary   Yes   Yes      E4F1   \n",
        "29       B104-0     M    72  T1N0M0       1   Primary   Yes   Yes     ERBB2   \n",
        "...         ...    ...  ...      ...    ...        ...   ...   ...      ...   \n",
        "1032        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes     OPCML   \n",
        "1033        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes     PARD3   \n",
        "1034        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes     PDZD2   \n",
        "1035        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes      PFKL   \n",
        "1036        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes       RB1   \n",
        "1037        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes      RXRA   \n",
        "1038        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes     SCRIB   \n",
        "1039        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes       SP1   \n",
        "1040        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes      STAM   \n",
        "1041        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes     SYNE1   \n",
        "1042        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes     SYNE2   \n",
        "1043        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes      TP53   \n",
        "1044        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes      UBR4   \n",
        "1045        B96     M    69  T3N0M0       2  Relapsed   Yes   Yes       UTX   \n",
        "1046        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes      BAI3   \n",
        "1047        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes      BRAF   \n",
        "1048        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes     BRCA2   \n",
        "1049        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes    CABIN1   \n",
        "1050        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes    CREBBP   \n",
        "1051        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes     DNAH8   \n",
        "1052        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes   DSCAML1   \n",
        "1053        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes   IL18RAP   \n",
        "1054        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes    NLRP14   \n",
        "1055        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes    PIK3CA   \n",
        "1056        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes    PIK3CA   \n",
        "1057        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes    PKD1L1   \n",
        "1058        B98     M    63  T3N1MX       3  Relapsed   Yes   Yes     SYNE1   \n",
        "1059        B99     F    37  T1N0M0       1   Primary   Yes   Yes      HRAS   \n",
        "1060        B99     F    37  T1N0M0       1   Primary   Yes   Yes      HRAS   \n",
        "1061        B99     F    37  T1N0M0       1   Primary   Yes   Yes    PDGFRB   \n",
        "\n",
        "                              Genome_Change Protein_Change      Mutation_Type  \\\n",
        "0                         g.chr7:5534867C>T        p.E125K           missense   \n",
        "1                       g.chr17:77093710C>T         p.Q59Q  synonymous-coding   \n",
        "2                        g.chr17:3778304G>A       p.S1016F           missense   \n",
        "3                        g.chr1:99979085C>T              -         spliceSite   \n",
        "4                        g.chrX:53593080G>A       p.L3200L  synonymous-coding   \n",
        "5                       g.chr7:104539925C>A       p.Q1496K           missense   \n",
        "6                       g.chr17:26581441C>T       p.Q1010*           nonsense   \n",
        "7                       g.chr17:26689262G>C       p.Q2266H           missense   \n",
        "8                        g.chrX:17656689G>A              -         spliceSite   \n",
        "9                       g.chr20:49840373C>T        p.D686N           missense   \n",
        "10                      g.chr19:50247168G>A              -         spliceSite   \n",
        "11                      g.chr8:133963297G>A        p.D216N           missense   \n",
        "12                       g.chr17:7517824C>G        p.R280T           missense   \n",
        "13                       g.chr5:64943386C>T        p.E149K           missense   \n",
        "14                       g.chrX:44803586C>T        p.Y375Y  synonymous-coding   \n",
        "15    g.chrX:44807883_44807892delAAGTGGAAGT              -  Frame-shift_indel   \n",
        "16    g.chrX:44813956_44813965delTACCTCAGGT              -  Frame-shift_indel   \n",
        "17                        g.chr4:1773362C>T        p.R248C           missense   \n",
        "18                      g.chr12:55858590G>T       p.V1515L           missense   \n",
        "19                     g.chr11:117864574C>A       p.H1456Q           missense   \n",
        "20                       g.chr7:26191453G>A        p.R537H           missense   \n",
        "21           g.chr9:134775785_134775785delG              -  Frame-shift_indel   \n",
        "22                      g.chrX:44814463insA              -  Frame-shift_indel   \n",
        "23                       g.chr17:3801309T>C        p.I150V           missense   \n",
        "24                        g.chr4:1775897A>G        p.Y373C           missense   \n",
        "25                      g.chr6:152788249C>T       p.D1750N           missense   \n",
        "26                     g.chr11:107422280C>G         p.L70V           missense   \n",
        "27                       g.chr6:38898765G>C       p.E1016Q           missense   \n",
        "28                       g.chr16:2224294C>G        p.L499L  synonymous-coding   \n",
        "29                      g.chr17:35136779C>G       p.T1052T  synonymous-coding   \n",
        "...                                     ...            ...                ...   \n",
        "1032                   g.chr11:132032228G>A        p.H122Y           missense   \n",
        "1033                    g.chr10:34713082G>C        p.R333G           missense   \n",
        "1034                     g.chr5:32109904G>C        p.E979Q           missense   \n",
        "1035                    g.chr21:44550152C>A         p.F44L           missense   \n",
        "1036          g.chr13:47835030_47835030delA              -  Frame-shift_indel   \n",
        "1037                    g.chr9:136468172C>T        p.S427F           missense   \n",
        "1038                    g.chr8:144957927G>A       p.L1098L  synonymous-coding   \n",
        "1039                    g.chr12:52063362C>G        p.T455R           missense   \n",
        "1040                    g.chr10:17796768C>T        p.Q536*           nonsense   \n",
        "1041                    g.chr6:152499480C>T        p.R727Q           missense   \n",
        "1042                    g.chr14:63567722A>C       p.K2372T           missense   \n",
        "1043                     g.chr17:7517747G>A        p.R306*           nonsense   \n",
        "1044                     g.chr1:19340826C>G       p.D2738H           missense   \n",
        "1045                     g.chrX:44807746C>T        p.Q555*           nonsense   \n",
        "1046                     g.chr6:70122468G>A       p.R1197Q           missense   \n",
        "1047                    g.chr7:140099623T>C        p.D594G           missense   \n",
        "1048                    g.chr13:31810996C>G       p.Q1502E           missense   \n",
        "1049                    g.chr22:22824071C>T       p.L1345L  synonymous-coding   \n",
        "1050                     g.chr16:3718536C>T       p.L2171L  synonymous-coding   \n",
        "1051                     g.chr6:38948800T>C       p.F2243L           missense   \n",
        "1052                   g.chr11:116804694C>G       p.E1968Q           missense   \n",
        "1053                    g.chr2:102430091C>T        p.T401M           missense   \n",
        "1054                     g.chr11:7020534C>G        p.S234*           nonsense   \n",
        "1055                    g.chr3:180418776G>A        p.E542K           missense   \n",
        "1056                    g.chr3:180418785G>C        p.E545Q           missense   \n",
        "1057                     g.chr7:47935479G>A        p.R303W           missense   \n",
        "1058                    g.chr6:152630895A>G       p.S6197S  synonymous-coding   \n",
        "1059                      g.chr11:523303G>T        p.P169H           missense   \n",
        "1060                      g.chr11:524288C>T         p.G12D           missense   \n",
        "1061    g.chr5:149473043_149473048delCTCTTC              -  Frame-shift_indel   \n",
        "\n",
        "                                            Tissue  \n",
        "0     Bladder cancer - transitional cell carcinoma  \n",
        "1     Bladder cancer - transitional cell carcinoma  \n",
        "2     Bladder cancer - transitional cell carcinoma  \n",
        "3     Bladder cancer - transitional cell carcinoma  \n",
        "4     Bladder cancer - transitional cell carcinoma  \n",
        "5     Bladder cancer - transitional cell carcinoma  \n",
        "6     Bladder cancer - transitional cell carcinoma  \n",
        "7     Bladder cancer - transitional cell carcinoma  \n",
        "8     Bladder cancer - transitional cell carcinoma  \n",
        "9     Bladder cancer - transitional cell carcinoma  \n",
        "10    Bladder cancer - transitional cell carcinoma  \n",
        "11    Bladder cancer - transitional cell carcinoma  \n",
        "12    Bladder cancer - transitional cell carcinoma  \n",
        "13    Bladder cancer - transitional cell carcinoma  \n",
        "14    Bladder cancer - transitional cell carcinoma  \n",
        "15    Bladder cancer - transitional cell carcinoma  \n",
        "16    Bladder cancer - transitional cell carcinoma  \n",
        "17    Bladder cancer - transitional cell carcinoma  \n",
        "18    Bladder cancer - transitional cell carcinoma  \n",
        "19    Bladder cancer - transitional cell carcinoma  \n",
        "20    Bladder cancer - transitional cell carcinoma  \n",
        "21    Bladder cancer - transitional cell carcinoma  \n",
        "22    Bladder cancer - transitional cell carcinoma  \n",
        "23    Bladder cancer - transitional cell carcinoma  \n",
        "24    Bladder cancer - transitional cell carcinoma  \n",
        "25    Bladder cancer - transitional cell carcinoma  \n",
        "26    Bladder cancer - transitional cell carcinoma  \n",
        "27    Bladder cancer - transitional cell carcinoma  \n",
        "28    Bladder cancer - transitional cell carcinoma  \n",
        "29    Bladder cancer - transitional cell carcinoma  \n",
        "...                                            ...  \n",
        "1032  Bladder cancer - transitional cell carcinoma  \n",
        "1033  Bladder cancer - transitional cell carcinoma  \n",
        "1034  Bladder cancer - transitional cell carcinoma  \n",
        "1035  Bladder cancer - transitional cell carcinoma  \n",
        "1036  Bladder cancer - transitional cell carcinoma  \n",
        "1037  Bladder cancer - transitional cell carcinoma  \n",
        "1038  Bladder cancer - transitional cell carcinoma  \n",
        "1039  Bladder cancer - transitional cell carcinoma  \n",
        "1040  Bladder cancer - transitional cell carcinoma  \n",
        "1041  Bladder cancer - transitional cell carcinoma  \n",
        "1042  Bladder cancer - transitional cell carcinoma  \n",
        "1043  Bladder cancer - transitional cell carcinoma  \n",
        "1044  Bladder cancer - transitional cell carcinoma  \n",
        "1045  Bladder cancer - transitional cell carcinoma  \n",
        "1046  Bladder cancer - transitional cell carcinoma  \n",
        "1047  Bladder cancer - transitional cell carcinoma  \n",
        "1048  Bladder cancer - transitional cell carcinoma  \n",
        "1049  Bladder cancer - transitional cell carcinoma  \n",
        "1050  Bladder cancer - transitional cell carcinoma  \n",
        "1051  Bladder cancer - transitional cell carcinoma  \n",
        "1052  Bladder cancer - transitional cell carcinoma  \n",
        "1053  Bladder cancer - transitional cell carcinoma  \n",
        "1054  Bladder cancer - transitional cell carcinoma  \n",
        "1055  Bladder cancer - transitional cell carcinoma  \n",
        "1056  Bladder cancer - transitional cell carcinoma  \n",
        "1057  Bladder cancer - transitional cell carcinoma  \n",
        "1058  Bladder cancer - transitional cell carcinoma  \n",
        "1059  Bladder cancer - transitional cell carcinoma  \n",
        "1060  Bladder cancer - transitional cell carcinoma  \n",
        "1061  Bladder cancer - transitional cell carcinoma  \n",
        "\n",
        "[1062 rows x 13 columns]"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Copy Number Data:"
     ]
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Other Data:"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "sigmutgene = pd.read_csv(PATH + 'sigmutgenese.csv', index_col = 0)\n",
      "sigmutgene.ix[['STAG2 ']]\n",
      "\n",
      "# c=b.query('Gender != [\"M\"]').dropna(subset=['Gender'])\n",
      "# d=c.sort('Age', ascending=False)\n",
      "# d.sort('Site').query('Site != [\"Small intestine\",\"Lymph node\"]')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Missense</th>\n",
        "      <th>Nonsense,splice_site,or_indel</th>\n",
        "      <th>Synnonymous_mutations</th>\n",
        "      <th>Total_Mutations</th>\n",
        "      <th>N:S_Ratio</th>\n",
        "      <th>Number(%)_of_subjects_harboring_non-silent_mutations</th>\n",
        "      <th>P-Value</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td> 2</td>\n",
        "      <td> 9</td>\n",
        "      <td> 0</td>\n",
        "      <td> 11</td>\n",
        "      <td> - </td>\n",
        "      <td> 11</td>\n",
        "      <td> 3.830000e-14</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 98,
       "text": [
        "        Missense  Nonsense,splice_site,or_indel  Synnonymous_mutations  \\\n",
        "STAG2          2                              9                      0   \n",
        "\n",
        "        Total_Mutations N:S_Ratio  \\\n",
        "STAG2                11        -    \n",
        "\n",
        "        Number(%)_of_subjects_harboring_non-silent_mutations       P-Value  \n",
        "STAG2                                                  11     3.830000e-14  "
       ]
      }
     ],
     "prompt_number": 98
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "#STAG2 mutations detected  in discovery(99) and prevalence(50) cohorts\n",
      "\n",
      "stag2muts = pd.read_csv(PATH + 'stag2muts.csv', index_col = 0)\n",
      "stag2muts.drop(['Nucleotide(cDNA)','Codon_Change','Amino_Acid_Change'], 1, inplace=True)\n",
      "stag2muts"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Tumor_ID</th>\n",
        "      <th>Genome_Change</th>\n",
        "      <th>Protein_Change</th>\n",
        "      <th>Mutation_Type</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Gene</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>   B105 </td>\n",
        "      <td>             g.chrX:123048109C&gt;T </td>\n",
        "      <td> p.Q1029* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>   B109 </td>\n",
        "      <td>             g.chrX:123037893T&gt;C </td>\n",
        "      <td>  p.I855T </td>\n",
        "      <td>          missense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>   B114 </td>\n",
        "      <td>             g.chrX:123027705G&gt;A </td>\n",
        "      <td>        - </td>\n",
        "      <td>        spliceSite </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>    B18 </td>\n",
        "      <td>             g.chrX:123045061C&gt;T </td>\n",
        "      <td> p.R1012* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>    B35 </td>\n",
        "      <td>             g.chrX:123012651G&gt;A </td>\n",
        "      <td>        - </td>\n",
        "      <td>        spliceSite </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>    B57 </td>\n",
        "      <td>             g.chrX:122999075G&gt;A </td>\n",
        "      <td>  p.W102* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>  B80-3 </td>\n",
        "      <td>             g.chrX:123024692C&gt;T </td>\n",
        "      <td>  p.Q593* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>  B89-4 </td>\n",
        "      <td>             g.chrX:123011737G&gt;T </td>\n",
        "      <td>  p.R305L </td>\n",
        "      <td>          missense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>   B110 </td>\n",
        "      <td>  g.chrX:123030112_123030112delA </td>\n",
        "      <td>   L761fs </td>\n",
        "      <td> Frame-shift_indel </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>    B87 </td>\n",
        "      <td>            g.chrX:123006776insA </td>\n",
        "      <td>   L182fs </td>\n",
        "      <td> Frame-shift_indel </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td> B89-16 </td>\n",
        "      <td> g.chrX:123024495_123024496delTG </td>\n",
        "      <td>   F567fs </td>\n",
        "      <td> Frame-shift_indel </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>     267</td>\n",
        "      <td>             g.chrX:123006731C&gt;T </td>\n",
        "      <td>  p.Q167* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>     267</td>\n",
        "      <td>             g.chrX:123006861C&gt;G </td>\n",
        "      <td>  p.S210* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>     237</td>\n",
        "      <td>             g.chrX:123030188G&gt;T </td>\n",
        "      <td>        - </td>\n",
        "      <td>        spliceSite </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>      71</td>\n",
        "      <td>             g.chrX:123052177C&gt;T </td>\n",
        "      <td> p.Q1117* </td>\n",
        "      <td>          nonsense </td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>STAG2 </th>\n",
        "      <td>     137</td>\n",
        "      <td>         g.chrX:123037883insTCAT </td>\n",
        "      <td>   A852fs </td>\n",
        "      <td> Frame-shift_indel </td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 102,
       "text": [
        "       Tumor_ID                     Genome_Change Protein_Change  \\\n",
        "Gene                                                               \n",
        "STAG2     B105               g.chrX:123048109C>T       p.Q1029*    \n",
        "STAG2     B109               g.chrX:123037893T>C        p.I855T    \n",
        "STAG2     B114               g.chrX:123027705G>A              -    \n",
        "STAG2      B18               g.chrX:123045061C>T       p.R1012*    \n",
        "STAG2      B35               g.chrX:123012651G>A              -    \n",
        "STAG2      B57               g.chrX:122999075G>A        p.W102*    \n",
        "STAG2    B80-3               g.chrX:123024692C>T        p.Q593*    \n",
        "STAG2    B89-4               g.chrX:123011737G>T        p.R305L    \n",
        "STAG2     B110    g.chrX:123030112_123030112delA         L761fs    \n",
        "STAG2      B87              g.chrX:123006776insA         L182fs    \n",
        "STAG2   B89-16   g.chrX:123024495_123024496delTG         F567fs    \n",
        "STAG2       267              g.chrX:123006731C>T        p.Q167*    \n",
        "STAG2       267              g.chrX:123006861C>G        p.S210*    \n",
        "STAG2       237              g.chrX:123030188G>T              -    \n",
        "STAG2        71              g.chrX:123052177C>T       p.Q1117*    \n",
        "STAG2       137          g.chrX:123037883insTCAT         A852fs    \n",
        "\n",
        "             Mutation_Type  \n",
        "Gene                        \n",
        "STAG2            nonsense   \n",
        "STAG2            missense   \n",
        "STAG2          spliceSite   \n",
        "STAG2            nonsense   \n",
        "STAG2          spliceSite   \n",
        "STAG2            nonsense   \n",
        "STAG2            nonsense   \n",
        "STAG2            missense   \n",
        "STAG2   Frame-shift_indel   \n",
        "STAG2   Frame-shift_indel   \n",
        "STAG2   Frame-shift_indel   \n",
        "STAG2            nonsense   \n",
        "STAG2            nonsense   \n",
        "STAG2          spliceSite   \n",
        "STAG2            nonsense   \n",
        "STAG2   Frame-shift_indel   "
       ]
      }
     ],
     "prompt_number": 102
    },
    {
     "cell_type": "code",
     "collapsed": true,
     "input": [
      "# Not needed: Significantly Mutated Paths\n",
      "\n",
      "sigmutpaths = pd.read_csv(PATH + 'sigmutpaths.csv', index_col = 0)\n",
      "sigmutpaths"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>#_of_reference_genes_in_category</th>\n",
        "      <th>#_of_genes_in_gene_set_and_also_in_category</th>\n",
        "      <th>Expected_#_in_category</th>\n",
        "      <th>Ratio_of_enrichment</th>\n",
        "      <th>P-Value(hypergeometric_test)</th>\n",
        "      <th>FDR</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Pathway</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>MAPK signaling pathway </th>\n",
        "      <td> 269</td>\n",
        "      <td> 31</td>\n",
        "      <td> 6.05</td>\n",
        "      <td>  5.12</td>\n",
        "      <td> 1.380000e-13</td>\n",
        "      <td> 1.920000e-12</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Cell cycle </th>\n",
        "      <td> 128</td>\n",
        "      <td> 21</td>\n",
        "      <td> 2.88</td>\n",
        "      <td>  7.29</td>\n",
        "      <td> 1.290000e-12</td>\n",
        "      <td> 1.630000e-11</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ErbB signaling pathway </th>\n",
        "      <td>  87</td>\n",
        "      <td> 17</td>\n",
        "      <td> 1.96</td>\n",
        "      <td>  8.69</td>\n",
        "      <td> 9.610000e-12</td>\n",
        "      <td> 1.110000e-10</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ECM-receptor interaction </th>\n",
        "      <td>  84</td>\n",
        "      <td> 16</td>\n",
        "      <td> 1.89</td>\n",
        "      <td>  8.47</td>\n",
        "      <td> 5.860000e-11</td>\n",
        "      <td> 6.010000e-10</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Calcium signaling pathway </th>\n",
        "      <td> 178</td>\n",
        "      <td> 21</td>\n",
        "      <td> 4.00</td>\n",
        "      <td>  5.24</td>\n",
        "      <td> 7.530000e-10</td>\n",
        "      <td> 6.160000e-09</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>egulation of actin cytoskeleton </th>\n",
        "      <td> 216</td>\n",
        "      <td> 23</td>\n",
        "      <td> 4.86</td>\n",
        "      <td>  4.73</td>\n",
        "      <td> 9.180000e-10</td>\n",
        "      <td> 7.090000e-09</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Insulin signaling pathway </th>\n",
        "      <td> 137</td>\n",
        "      <td> 18</td>\n",
        "      <td> 3.08</td>\n",
        "      <td>  5.84</td>\n",
        "      <td> 2.140000e-09</td>\n",
        "      <td> 1.490000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>eurotrophin signaling pathway </th>\n",
        "      <td> 126</td>\n",
        "      <td> 17</td>\n",
        "      <td> 2.83</td>\n",
        "      <td>  6.00</td>\n",
        "      <td> 3.950000e-09</td>\n",
        "      <td> 2.610000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Long-term potentiation </th>\n",
        "      <td>  70</td>\n",
        "      <td> 13</td>\n",
        "      <td> 1.57</td>\n",
        "      <td>  8.26</td>\n",
        "      <td> 5.090000e-09</td>\n",
        "      <td> 3.220000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Axon guidance </th>\n",
        "      <td> 129</td>\n",
        "      <td> 17</td>\n",
        "      <td> 2.90</td>\n",
        "      <td>  5.86</td>\n",
        "      <td> 5.690000e-09</td>\n",
        "      <td> 3.410000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Notch signaling pathway </th>\n",
        "      <td>  47</td>\n",
        "      <td> 11</td>\n",
        "      <td> 1.06</td>\n",
        "      <td> 10.40</td>\n",
        "      <td> 5.890000e-09</td>\n",
        "      <td> 3.410000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>GnRH signaling pathway </th>\n",
        "      <td> 101</td>\n",
        "      <td> 15</td>\n",
        "      <td> 2.27</td>\n",
        "      <td>  6.60</td>\n",
        "      <td> 8.500000e-09</td>\n",
        "      <td> 4.730000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TGF-beta signaling pathway </th>\n",
        "      <td>  87</td>\n",
        "      <td> 14</td>\n",
        "      <td> 1.96</td>\n",
        "      <td>  7.15</td>\n",
        "      <td> 9.190000e-09</td>\n",
        "      <td> 4.910000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Endocytosis </th>\n",
        "      <td> 187</td>\n",
        "      <td> 20</td>\n",
        "      <td> 4.21</td>\n",
        "      <td>  4.75</td>\n",
        "      <td> 1.050000e-08</td>\n",
        "      <td> 5.410000e-08</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Purine metabolism </th>\n",
        "      <td> 151</td>\n",
        "      <td> 16</td>\n",
        "      <td> 3.40</td>\n",
        "      <td>  4.71</td>\n",
        "      <td> 3.460000e-07</td>\n",
        "      <td> 1.500000e-06</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>p53 signaling pathway </th>\n",
        "      <td>  69</td>\n",
        "      <td> 11</td>\n",
        "      <td> 1.55</td>\n",
        "      <td>  7.09</td>\n",
        "      <td> 3.930000e-07</td>\n",
        "      <td> 1.660000e-06</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Adherens junction </th>\n",
        "      <td>  77</td>\n",
        "      <td> 11</td>\n",
        "      <td> 1.73</td>\n",
        "      <td>  6.35</td>\n",
        "      <td> 1.220000e-06</td>\n",
        "      <td> 4.990000e-06</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Vascular smooth muscle </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>contraction </th>\n",
        "      <td> 115</td>\n",
        "      <td> 13</td>\n",
        "      <td> 2.59</td>\n",
        "      <td>  5.02</td>\n",
        "      <td> 2.070000e-06</td>\n",
        "      <td> 7.990000e-06</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RNA polymerase </th>\n",
        "      <td>  29</td>\n",
        "      <td>  7</td>\n",
        "      <td> 0.65</td>\n",
        "      <td> 10.73</td>\n",
        "      <td> 2.890000e-06</td>\n",
        "      <td> 1.080000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Ubiquitin mediated proteolysis </th>\n",
        "      <td> 138</td>\n",
        "      <td> 14</td>\n",
        "      <td> 3.10</td>\n",
        "      <td>  4.51</td>\n",
        "      <td> 3.100000e-06</td>\n",
        "      <td> 1.100000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Long-term depression </th>\n",
        "      <td>  70</td>\n",
        "      <td> 10</td>\n",
        "      <td> 1.57</td>\n",
        "      <td>  6.35</td>\n",
        "      <td> 3.720000e-06</td>\n",
        "      <td> 1.290000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Phosphatidylinositol signaling </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>system </th>\n",
        "      <td>  76</td>\n",
        "      <td> 10</td>\n",
        "      <td> 1.71</td>\n",
        "      <td>  5.85</td>\n",
        "      <td> 7.920000e-06</td>\n",
        "      <td> 2.620000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Wnt signaling pathway </th>\n",
        "      <td> 151</td>\n",
        "      <td> 14</td>\n",
        "      <td> 3.40</td>\n",
        "      <td>  4.12</td>\n",
        "      <td> 8.890000e-06</td>\n",
        "      <td> 2.870000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Oocyte meiosis </th>\n",
        "      <td> 114</td>\n",
        "      <td> 12</td>\n",
        "      <td> 2.56</td>\n",
        "      <td>  4.68</td>\n",
        "      <td> 1.060000e-05</td>\n",
        "      <td> 3.350000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Pyrimidine metabolism </th>\n",
        "      <td>  98</td>\n",
        "      <td> 11</td>\n",
        "      <td> 2.20</td>\n",
        "      <td>  4.99</td>\n",
        "      <td> 1.340000e-05</td>\n",
        "      <td> 4.140000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mTOR signaling pathway </th>\n",
        "      <td>  52</td>\n",
        "      <td>  8</td>\n",
        "      <td> 1.17</td>\n",
        "      <td>  6.84</td>\n",
        "      <td> 2.000000e-05</td>\n",
        "      <td> 5.910000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Chemokine signaling pathway </th>\n",
        "      <td> 190</td>\n",
        "      <td> 15</td>\n",
        "      <td> 4.27</td>\n",
        "      <td>  3.51</td>\n",
        "      <td> 2.950000e-05</td>\n",
        "      <td> 8.200000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Gap junction </th>\n",
        "      <td>  90</td>\n",
        "      <td> 10</td>\n",
        "      <td> 2.02</td>\n",
        "      <td>  4.94</td>\n",
        "      <td> 3.580000e-05</td>\n",
        "      <td> 9.760000e-05</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>VEGF signaling pathway </th>\n",
        "      <td>  76</td>\n",
        "      <td>  9</td>\n",
        "      <td> 1.71</td>\n",
        "      <td>  5.26</td>\n",
        "      <td> 5.270000e-05</td>\n",
        "      <td> 1.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>lycosaminoglycan degradation </th>\n",
        "      <td>  21</td>\n",
        "      <td>  5</td>\n",
        "      <td> 0.47</td>\n",
        "      <td> 10.58</td>\n",
        "      <td> 8.600000e-05</td>\n",
        "      <td> 2.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Progesterone-mediated oocyte </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>maturation </th>\n",
        "      <td>  86</td>\n",
        "      <td>  9</td>\n",
        "      <td> 1.93</td>\n",
        "      <td>  4.65</td>\n",
        "      <td> 1.000000e-04</td>\n",
        "      <td> 2.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Tight junction </th>\n",
        "      <td> 134</td>\n",
        "      <td> 11</td>\n",
        "      <td> 3.01</td>\n",
        "      <td>  3.65</td>\n",
        "      <td> 2.000000e-04</td>\n",
        "      <td> 5.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Leukocyte transendothelial </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>migration </th>\n",
        "      <td> 118</td>\n",
        "      <td> 10</td>\n",
        "      <td> 2.65</td>\n",
        "      <td>  3.77</td>\n",
        "      <td> 3.000000e-04</td>\n",
        "      <td> 7.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>B cell receptor signaling </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>pathway </th>\n",
        "      <td>  75</td>\n",
        "      <td>  8</td>\n",
        "      <td> 1.69</td>\n",
        "      <td>  4.74</td>\n",
        "      <td> 3.000000e-04</td>\n",
        "      <td> 7.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Lysine degradation </th>\n",
        "      <td>  44</td>\n",
        "      <td>  6</td>\n",
        "      <td> 0.99</td>\n",
        "      <td>  6.06</td>\n",
        "      <td> 4.000000e-04</td>\n",
        "      <td> 9.000000e-04</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NOD-like receptor signaling </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>pathway </th>\n",
        "      <td>  62</td>\n",
        "      <td>  7</td>\n",
        "      <td> 1.39</td>\n",
        "      <td>  5.02</td>\n",
        "      <td> 5.000000e-04</td>\n",
        "      <td> 1.100000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>PPAR signaling pathway </th>\n",
        "      <td>  69</td>\n",
        "      <td>  7</td>\n",
        "      <td> 1.55</td>\n",
        "      <td>  4.51</td>\n",
        "      <td> 9.000000e-04</td>\n",
        "      <td> 2.000000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>RIG-I-like receptor signaling </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>pathway </th>\n",
        "      <td>  71</td>\n",
        "      <td>  7</td>\n",
        "      <td> 1.60</td>\n",
        "      <td>  4.38</td>\n",
        "      <td> 1.100000e-03</td>\n",
        "      <td> 2.400000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Basal transcription factors </th>\n",
        "      <td>  36</td>\n",
        "      <td>  5</td>\n",
        "      <td> 0.81</td>\n",
        "      <td>  6.17</td>\n",
        "      <td> 1.200000e-03</td>\n",
        "      <td> 2.600000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Lysosome </th>\n",
        "      <td> 117</td>\n",
        "      <td>  9</td>\n",
        "      <td> 2.63</td>\n",
        "      <td>  3.42</td>\n",
        "      <td> 1.400000e-03</td>\n",
        "      <td> 2.900000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Jak-STAT signaling pathway </th>\n",
        "      <td> 155</td>\n",
        "      <td> 10</td>\n",
        "      <td> 3.49</td>\n",
        "      <td>  2.87</td>\n",
        "      <td> 2.800000e-03</td>\n",
        "      <td> 5.600000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Nucleotide excision repair </th>\n",
        "      <td>  44</td>\n",
        "      <td>  5</td>\n",
        "      <td> 0.99</td>\n",
        "      <td>  5.05</td>\n",
        "      <td> 3.000000e-03</td>\n",
        "      <td> 5.900000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>T cell receptor signaling </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>pathway </th>\n",
        "      <td> 108</td>\n",
        "      <td>  8</td>\n",
        "      <td> 2.43</td>\n",
        "      <td>  3.29</td>\n",
        "      <td> 3.100000e-03</td>\n",
        "      <td> 6.000000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Tyrosine metabolism </th>\n",
        "      <td>  46</td>\n",
        "      <td>  5</td>\n",
        "      <td> 1.03</td>\n",
        "      <td>  4.83</td>\n",
        "      <td> 3.600000e-03</td>\n",
        "      <td> 6.900000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Apoptosis </th>\n",
        "      <td>  88</td>\n",
        "      <td>  7</td>\n",
        "      <td> 1.98</td>\n",
        "      <td>  3.54</td>\n",
        "      <td> 3.700000e-03</td>\n",
        "      <td> 7.000000e-03</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Natural killer cell mediated </th>\n",
        "      <td> NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>          NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>cytotoxicity </th>\n",
        "      <td> 137</td>\n",
        "      <td>  9</td>\n",
        "      <td> 3.08</td>\n",
        "      <td>  2.92</td>\n",
        "      <td> 4.000000e-03</td>\n",
        "      <td> 7.400000e-03</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 71,
       "text": [
        "                                  #_of_reference_genes_in_category  \\\n",
        "Pathway                                                              \n",
        "MAPK signaling pathway                                         269   \n",
        "Cell cycle                                                     128   \n",
        "ErbB signaling pathway                                          87   \n",
        "ECM-receptor interaction                                        84   \n",
        "Calcium signaling pathway                                      178   \n",
        "egulation of actin cytoskeleton                                216   \n",
        "Insulin signaling pathway                                      137   \n",
        "eurotrophin signaling pathway                                  126   \n",
        "Long-term potentiation                                          70   \n",
        "Axon guidance                                                  129   \n",
        "Notch signaling pathway                                         47   \n",
        "GnRH signaling pathway                                         101   \n",
        "TGF-beta signaling pathway                                      87   \n",
        "Endocytosis                                                    187   \n",
        "Purine metabolism                                              151   \n",
        "p53 signaling pathway                                           69   \n",
        "Adherens junction                                               77   \n",
        "Vascular smooth muscle                                         NaN   \n",
        "contraction                                                    115   \n",
        "RNA polymerase                                                  29   \n",
        "Ubiquitin mediated proteolysis                                 138   \n",
        "Long-term depression                                            70   \n",
        "Phosphatidylinositol signaling                                 NaN   \n",
        "system                                                          76   \n",
        "Wnt signaling pathway                                          151   \n",
        "Oocyte meiosis                                                 114   \n",
        "Pyrimidine metabolism                                           98   \n",
        "mTOR signaling pathway                                          52   \n",
        "Chemokine signaling pathway                                    190   \n",
        "Gap junction                                                    90   \n",
        "VEGF signaling pathway                                          76   \n",
        "lycosaminoglycan degradation                                    21   \n",
        "Progesterone-mediated oocyte                                   NaN   \n",
        "maturation                                                      86   \n",
        "Tight junction                                                 134   \n",
        "Leukocyte transendothelial                                     NaN   \n",
        "migration                                                      118   \n",
        "B cell receptor signaling                                      NaN   \n",
        "pathway                                                         75   \n",
        "Lysine degradation                                              44   \n",
        "NOD-like receptor signaling                                    NaN   \n",
        "pathway                                                         62   \n",
        "PPAR signaling pathway                                          69   \n",
        "RIG-I-like receptor signaling                                  NaN   \n",
        "pathway                                                         71   \n",
        "Basal transcription factors                                     36   \n",
        "Lysosome                                                       117   \n",
        "Jak-STAT signaling pathway                                     155   \n",
        "Nucleotide excision repair                                      44   \n",
        "T cell receptor signaling                                      NaN   \n",
        "pathway                                                        108   \n",
        "Tyrosine metabolism                                             46   \n",
        "Apoptosis                                                       88   \n",
        "Natural killer cell mediated                                   NaN   \n",
        "cytotoxicity                                                   137   \n",
        "\n",
        "                                  #_of_genes_in_gene_set_and_also_in_category  \\\n",
        "Pathway                                                                         \n",
        "MAPK signaling pathway                                                     31   \n",
        "Cell cycle                                                                 21   \n",
        "ErbB signaling pathway                                                     17   \n",
        "ECM-receptor interaction                                                   16   \n",
        "Calcium signaling pathway                                                  21   \n",
        "egulation of actin cytoskeleton                                            23   \n",
        "Insulin signaling pathway                                                  18   \n",
        "eurotrophin signaling pathway                                              17   \n",
        "Long-term potentiation                                                     13   \n",
        "Axon guidance                                                              17   \n",
        "Notch signaling pathway                                                    11   \n",
        "GnRH signaling pathway                                                     15   \n",
        "TGF-beta signaling pathway                                                 14   \n",
        "Endocytosis                                                                20   \n",
        "Purine metabolism                                                          16   \n",
        "p53 signaling pathway                                                      11   \n",
        "Adherens junction                                                          11   \n",
        "Vascular smooth muscle                                                    NaN   \n",
        "contraction                                                                13   \n",
        "RNA polymerase                                                              7   \n",
        "Ubiquitin mediated proteolysis                                             14   \n",
        "Long-term depression                                                       10   \n",
        "Phosphatidylinositol signaling                                            NaN   \n",
        "system                                                                     10   \n",
        "Wnt signaling pathway                                                      14   \n",
        "Oocyte meiosis                                                             12   \n",
        "Pyrimidine metabolism                                                      11   \n",
        "mTOR signaling pathway                                                      8   \n",
        "Chemokine signaling pathway                                                15   \n",
        "Gap junction                                                               10   \n",
        "VEGF signaling pathway                                                      9   \n",
        "lycosaminoglycan degradation                                                5   \n",
        "Progesterone-mediated oocyte                                              NaN   \n",
        "maturation                                                                  9   \n",
        "Tight junction                                                             11   \n",
        "Leukocyte transendothelial                                                NaN   \n",
        "migration                                                                  10   \n",
        "B cell receptor signaling                                                 NaN   \n",
        "pathway                                                                     8   \n",
        "Lysine degradation                                                          6   \n",
        "NOD-like receptor signaling                                               NaN   \n",
        "pathway                                                                     7   \n",
        "PPAR signaling pathway                                                      7   \n",
        "RIG-I-like receptor signaling                                             NaN   \n",
        "pathway                                                                     7   \n",
        "Basal transcription factors                                                 5   \n",
        "Lysosome                                                                    9   \n",
        "Jak-STAT signaling pathway                                                 10   \n",
        "Nucleotide excision repair                                                  5   \n",
        "T cell receptor signaling                                                 NaN   \n",
        "pathway                                                                     8   \n",
        "Tyrosine metabolism                                                         5   \n",
        "Apoptosis                                                                   7   \n",
        "Natural killer cell mediated                                              NaN   \n",
        "cytotoxicity                                                                9   \n",
        "\n",
        "                                  Expected_#_in_category  Ratio_of_enrichment  \\\n",
        "Pathway                                                                         \n",
        "MAPK signaling pathway                              6.05                 5.12   \n",
        "Cell cycle                                          2.88                 7.29   \n",
        "ErbB signaling pathway                              1.96                 8.69   \n",
        "ECM-receptor interaction                            1.89                 8.47   \n",
        "Calcium signaling pathway                           4.00                 5.24   \n",
        "egulation of actin cytoskeleton                     4.86                 4.73   \n",
        "Insulin signaling pathway                           3.08                 5.84   \n",
        "eurotrophin signaling pathway                       2.83                 6.00   \n",
        "Long-term potentiation                              1.57                 8.26   \n",
        "Axon guidance                                       2.90                 5.86   \n",
        "Notch signaling pathway                             1.06                10.40   \n",
        "GnRH signaling pathway                              2.27                 6.60   \n",
        "TGF-beta signaling pathway                          1.96                 7.15   \n",
        "Endocytosis                                         4.21                 4.75   \n",
        "Purine metabolism                                   3.40                 4.71   \n",
        "p53 signaling pathway                               1.55                 7.09   \n",
        "Adherens junction                                   1.73                 6.35   \n",
        "Vascular smooth muscle                               NaN                  NaN   \n",
        "contraction                                         2.59                 5.02   \n",
        "RNA polymerase                                      0.65                10.73   \n",
        "Ubiquitin mediated proteolysis                      3.10                 4.51   \n",
        "Long-term depression                                1.57                 6.35   \n",
        "Phosphatidylinositol signaling                       NaN                  NaN   \n",
        "system                                              1.71                 5.85   \n",
        "Wnt signaling pathway                               3.40                 4.12   \n",
        "Oocyte meiosis                                      2.56                 4.68   \n",
        "Pyrimidine metabolism                               2.20                 4.99   \n",
        "mTOR signaling pathway                              1.17                 6.84   \n",
        "Chemokine signaling pathway                         4.27                 3.51   \n",
        "Gap junction                                        2.02                 4.94   \n",
        "VEGF signaling pathway                              1.71                 5.26   \n",
        "lycosaminoglycan degradation                        0.47                10.58   \n",
        "Progesterone-mediated oocyte                         NaN                  NaN   \n",
        "maturation                                          1.93                 4.65   \n",
        "Tight junction                                      3.01                 3.65   \n",
        "Leukocyte transendothelial                           NaN                  NaN   \n",
        "migration                                           2.65                 3.77   \n",
        "B cell receptor signaling                            NaN                  NaN   \n",
        "pathway                                             1.69                 4.74   \n",
        "Lysine degradation                                  0.99                 6.06   \n",
        "NOD-like receptor signaling                          NaN                  NaN   \n",
        "pathway                                             1.39                 5.02   \n",
        "PPAR signaling pathway                              1.55                 4.51   \n",
        "RIG-I-like receptor signaling                        NaN                  NaN   \n",
        "pathway                                             1.60                 4.38   \n",
        "Basal transcription factors                         0.81                 6.17   \n",
        "Lysosome                                            2.63                 3.42   \n",
        "Jak-STAT signaling pathway                          3.49                 2.87   \n",
        "Nucleotide excision repair                          0.99                 5.05   \n",
        "T cell receptor signaling                            NaN                  NaN   \n",
        "pathway                                             2.43                 3.29   \n",
        "Tyrosine metabolism                                 1.03                 4.83   \n",
        "Apoptosis                                           1.98                 3.54   \n",
        "Natural killer cell mediated                         NaN                  NaN   \n",
        "cytotoxicity                                        3.08                 2.92   \n",
        "\n",
        "                                  P-Value(hypergeometric_test)           FDR  \n",
        "Pathway                                                                       \n",
        "MAPK signaling pathway                            1.380000e-13  1.920000e-12  \n",
        "Cell cycle                                        1.290000e-12  1.630000e-11  \n",
        "ErbB signaling pathway                            9.610000e-12  1.110000e-10  \n",
        "ECM-receptor interaction                          5.860000e-11  6.010000e-10  \n",
        "Calcium signaling pathway                         7.530000e-10  6.160000e-09  \n",
        "egulation of actin cytoskeleton                   9.180000e-10  7.090000e-09  \n",
        "Insulin signaling pathway                         2.140000e-09  1.490000e-08  \n",
        "eurotrophin signaling pathway                     3.950000e-09  2.610000e-08  \n",
        "Long-term potentiation                            5.090000e-09  3.220000e-08  \n",
        "Axon guidance                                     5.690000e-09  3.410000e-08  \n",
        "Notch signaling pathway                           5.890000e-09  3.410000e-08  \n",
        "GnRH signaling pathway                            8.500000e-09  4.730000e-08  \n",
        "TGF-beta signaling pathway                        9.190000e-09  4.910000e-08  \n",
        "Endocytosis                                       1.050000e-08  5.410000e-08  \n",
        "Purine metabolism                                 3.460000e-07  1.500000e-06  \n",
        "p53 signaling pathway                             3.930000e-07  1.660000e-06  \n",
        "Adherens junction                                 1.220000e-06  4.990000e-06  \n",
        "Vascular smooth muscle                                     NaN           NaN  \n",
        "contraction                                       2.070000e-06  7.990000e-06  \n",
        "RNA polymerase                                    2.890000e-06  1.080000e-05  \n",
        "Ubiquitin mediated proteolysis                    3.100000e-06  1.100000e-05  \n",
        "Long-term depression                              3.720000e-06  1.290000e-05  \n",
        "Phosphatidylinositol signaling                             NaN           NaN  \n",
        "system                                            7.920000e-06  2.620000e-05  \n",
        "Wnt signaling pathway                             8.890000e-06  2.870000e-05  \n",
        "Oocyte meiosis                                    1.060000e-05  3.350000e-05  \n",
        "Pyrimidine metabolism                             1.340000e-05  4.140000e-05  \n",
        "mTOR signaling pathway                            2.000000e-05  5.910000e-05  \n",
        "Chemokine signaling pathway                       2.950000e-05  8.200000e-05  \n",
        "Gap junction                                      3.580000e-05  9.760000e-05  \n",
        "VEGF signaling pathway                            5.270000e-05  1.000000e-04  \n",
        "lycosaminoglycan degradation                      8.600000e-05  2.000000e-04  \n",
        "Progesterone-mediated oocyte                               NaN           NaN  \n",
        "maturation                                        1.000000e-04  2.000000e-04  \n",
        "Tight junction                                    2.000000e-04  5.000000e-04  \n",
        "Leukocyte transendothelial                                 NaN           NaN  \n",
        "migration                                         3.000000e-04  7.000000e-04  \n",
        "B cell receptor signaling                                  NaN           NaN  \n",
        "pathway                                           3.000000e-04  7.000000e-04  \n",
        "Lysine degradation                                4.000000e-04  9.000000e-04  \n",
        "NOD-like receptor signaling                                NaN           NaN  \n",
        "pathway                                           5.000000e-04  1.100000e-03  \n",
        "PPAR signaling pathway                            9.000000e-04  2.000000e-03  \n",
        "RIG-I-like receptor signaling                              NaN           NaN  \n",
        "pathway                                           1.100000e-03  2.400000e-03  \n",
        "Basal transcription factors                       1.200000e-03  2.600000e-03  \n",
        "Lysosome                                          1.400000e-03  2.900000e-03  \n",
        "Jak-STAT signaling pathway                        2.800000e-03  5.600000e-03  \n",
        "Nucleotide excision repair                        3.000000e-03  5.900000e-03  \n",
        "T cell receptor signaling                                  NaN           NaN  \n",
        "pathway                                           3.100000e-03  6.000000e-03  \n",
        "Tyrosine metabolism                               3.600000e-03  6.900000e-03  \n",
        "Apoptosis                                         3.700000e-03  7.000000e-03  \n",
        "Natural killer cell mediated                               NaN           NaN  \n",
        "cytotoxicity                                      4.000000e-03  7.400000e-03  "
       ]
      }
     ],
     "prompt_number": 71
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%pylab inline\n",
      "patdata['Age'] = np.arange(99)\n",
      "patdata['Diagnosis'] = patdata.Age == 3\n",
      "patdata.Diagnosis.value_counts()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Populating the interactive namespace from numpy and matplotlib\n"
       ]
      },
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 55,
       "text": [
        "False    98\n",
        "True      1\n",
        "dtype: int64"
       ]
      }
     ],
     "prompt_number": 55
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patdata['Man'] = patdata['Gender'] == 'M '\n",
      "patdata['Gender'] = patdata['Gender'].map(str.strip)\n",
      "patdata['Man fo real'] = patdata['Gender'] == 'M'\n",
      "patdata.query('Gender == \"M\"')\n",
      "a = patdata.head()\n",
      "#a\n",
      "del a['Man']\n",
      "'Gender' in a.columns\n",
      "'Gender' in a.index\n",
      "b = a.set_index('Status')\n",
      "#a.drop(['Tumor_ID','WES'], axis =1)\n",
      "b < 5"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Patient_ID</th>\n",
        "      <th>Gender</th>\n",
        "      <th>Age</th>\n",
        "      <th>Tumor_ID</th>\n",
        "      <th>Grade</th>\n",
        "      <th>WES</th>\n",
        "      <th>WGS</th>\n",
        "      <th>Diagnosis</th>\n",
        "      <th>Status is now Good</th>\n",
        "      <th>Status is now bad</th>\n",
        "      <th>Man fo real</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Status</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Primary </th>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Primary </th>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Primary </th>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Primary </th>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Primary </th>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> False</td>\n",
        "      <td> False</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "      <td> True</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 117,
       "text": [
        "         Patient_ID Gender   Age Tumor_ID Grade    WES    WGS Diagnosis  \\\n",
        "Status                                                                    \n",
        "Primary       False  False  True    False  True  False  False      True   \n",
        "Primary       False  False  True    False  True  False  False      True   \n",
        "Primary       False  False  True    False  True  False  False      True   \n",
        "Primary       False  False  True    False  True  False  False      True   \n",
        "Primary       False  False  True    False  True  False  False      True   \n",
        "\n",
        "         Status is now Good Status is now bad Man fo real  \n",
        "Status                                                     \n",
        "Primary                True              True        True  \n",
        "Primary                True              True        True  \n",
        "Primary                True              True        True  \n",
        "Primary                True              True        True  \n",
        "Primary                True              True        True  "
       ]
      }
     ],
     "prompt_number": 117
    }
   ],
   "metadata": {}
  }
 ]
}